<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xml:lang="en" xmlns="http://www.w3.org/1999/xhtml" lang="en">
  <head>
    <title>bioruby's bioruby - Class: Bio::PAML::Codeml - GitHub Documentation</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta name="description" content="">

<link rel="search" type="application/opensearchdescription+xml" href="http://github.com/opensearch.xml" title="GitHub">
<link rel="fluid-icon" href="http://github.com/fluidicon.png" title="GitHub">
<link href="http://assets0.github.com/stylesheets/bundle.css" media="screen" rel="stylesheet" type="text/css">
<link href="http://github.com/feeds/bioruby/commits/bioruby/master" rel="alternate" title="Recent Commits to bioruby:master" type="application/atom+xml">

<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js"></script>
<script type="text/javascript" src="http://assets3.github.com/javascripts/bundle.js"></script>              
<script src="../../app.js" type="text/javascript" charset="utf-8"></script>

<script type="text/javascript" src="http://docs.github.com/javascripts/md5.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/repos.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/commits.js"></script>
<script type="text/javascript">
  $(document).ready(function(){
    if (typeof skipHeaders != 'undefined') return;
    
    GitHubRepo.init("bioruby", "bioruby", function(){ 
      GitHubRepo.links = " <a id='namespaces_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/namespaces_button.png' alt='Modules and Classes'/></a> ";
      GitHubRepo.links += " <a id='methods_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/methods_button.png' alt='Methods'/></a> ";
      $("#repo").html(GitHubRepo.content());
      $('#namespaces_button').click(function(){
        var url="/"+$(this).attr('rel')+"/namespaces/";
        $.gitbox(url);
        return false;
      });    
      $('#methods_button').click(function(){
        var url="/"+$(this).attr('rel')+"/methods/";
        $.gitbox(url);
        return false;
      });        
    });
    GitHubCommit.init("bioruby", "bioruby", "97b9284109c9a4431b92eab208509e1df6069b4b", function(){ 
      $("#commit").html(GitHubCommit.content());
    });
  });
</script>

    <link rel="stylesheet" href="../../style.css" type="text/css" charset="utf-8" />
  </head>
  <body>    
    
    
    
        <div id="main" style="padding-bottom:8em;">
      <div id="header">
        <div class="site">
          <div class="logo">
            <a href="http://github.com/"><img src="http://github.com/images/modules/header/logov3.png" alt="github"></a>
          </div>          
          <div class="topsearch">
            <form action="http://github.com/search" id="top_search_form" method="get">
              <input class="search" name="q" type="search"> <input value="Search" type="submit">
            </form>
            <div class="links">
              <a href="http://github.com/repositories">Browse</a> | <a href="http://github.com/guides">Guides</a> | <a href="http://github.com/search">Advanced</a>
            </div>
          </div>
          <div class="actions">
            <a href="http://github.com">Home</a>
            <a href="http://github.com/plans"><b><u>Pricing and Signup</u></b></a>
            <a href="http://docs.github.com/">Documentation</a>
            <a href="https://github.com/login">Login</a>
          </div>                  
        </div>
      </div>    
                                                  
      <div id="repo_menu">
        <div class="site">
          <ul>          
            <li class=""><a href="http://github.com/bioruby/bioruby/tree/">Source</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/commits/">Commits</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/network">Network</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/issues">Issues</a></li>
            <li class=""><a href="http://wiki.github.com/bioruby/bioruby">Wiki</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/graphs">Graphs</a></li>                    
            <li class="active"><a href="/bioruby/bioruby">Documentation</a></li>
          </ul>
        </div>
      </div>

      <div id="repo_sub_menu">
        <div class="site">
          <div class="joiner"></div>      
        </div>
      </div>

      <div class="site">  
        <div id="repos">
          <div id="repo" class="repo public">
            <div class="title">
              <div class="path">
                <a href="http://github.com/bioruby">bioruby</a> / <b><a href="http://github.com/bioruby/bioruby/tree">bioruby</a></b>        
                <a id="namespaces_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/namespaces_button.png" alt="Modules and Classes"/></a>
                <a id="methods_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/methods_button.png" alt="Methods"/></a>
              </div>
            </div>            
          </div>
          <div id="commit">
          </div> 
          <br/>                                       
          <div id="content">
            <div class="wikistyle">
  <div class="section class Bio_PAML_Codeml">
  <h1>Class: Bio::PAML::Codeml</h1>
  <div class="section inheritance">
  <ul>
  <li><a href="Common.html" title="Bio::PAML::Common">Bio::PAML::Common</a></li><ul><li>Bio::PAML::Codeml</li>
  </ul></ul>
</div><div class="section docstring">
  <h2>Description</h2>
<p>
Bio::PAML::Codeml is a wrapper for estimating evolutionary rate using the
CODEML tool. The class provides methods for generating the necessary
configuration file, and running codeml with the specified binary. Codeml
output is returned when codeml is run. Bio::PAML::Codeml::Report and
Bio::PAML::Codeml::Rates provide simple classes for parsing and accessing
the Codeml report and rates files respectively.
</p>
<h2>Examples</h2>
<p>
Example 1:
</p>
<pre class="code">
  <span class='require identifier id'>require</span> <span class='string val'>'bio'</span>
  <span class='comment val'># Reads multi-fasta formatted file and gets a Bio::Alignment object.</span>
  <span class='alignment identifier id'>alignment</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='FlatFile constant id'>FlatFile</span><span class='dot token'>.</span><span class='open identifier id'>open</span><span class='lparen token'>(</span><span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Alignment constant id'>Alignment</span><span class='colon2 op'>::</span><span class='MultiFastaFormat constant id'>MultiFastaFormat</span><span class='comma token'>,</span>
                                 <span class='string val'>'example.fst'</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='alignment identifier id'>alignment</span>
  <span class='comment val'># Reads newick tree from a file</span>
  <span class='tree identifier id'>tree</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='FlatFile constant id'>FlatFile</span><span class='dot token'>.</span><span class='open identifier id'>open</span><span class='lparen token'>(</span><span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Newick constant id'>Newick</span><span class='comma token'>,</span> <span class='string val'>'example.tree'</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='tree identifier id'>tree</span>
  <span class='comment val'># Creates a Codeml object</span>
  <span class='codeml identifier id'>codeml</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='PAML constant id'>PAML</span><span class='colon2 op'>::</span><span class='Codeml constant id'>Codeml</span><span class='dot token'>.</span><span class='new identifier id'>new</span>
  <span class='comment val'># Sets parameters</span>
  <span class='codeml identifier id'>codeml</span><span class='dot token'>.</span><span class='parameters identifier id'>parameters</span><span class='lbrack token'>[</span><span class='symbol val'>:runmode</span><span class='rbrack token'>]</span> <span class='assign token'>=</span> <span class='integer val'>0</span>
  <span class='codeml identifier id'>codeml</span><span class='dot token'>.</span><span class='parameters identifier id'>parameters</span><span class='lbrack token'>[</span><span class='symbol val'>:RateAncestor</span><span class='rbrack token'>]</span> <span class='assign token'>=</span> <span class='integer val'>1</span>
  <span class='comment val'># You can also set many parameters at a time.</span>
  <span class='codeml identifier id'>codeml</span><span class='dot token'>.</span><span class='parameters identifier id'>parameters</span><span class='dot token'>.</span><span class='update identifier id'>update</span><span class='lparen token'>(</span><span class='lbrace token'>{</span> <span class='symbol val'>:alpha</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>0</span><span class='integer val'>.5</span><span class='comma token'>,</span> <span class='symbol val'>:fix_alpha</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>0</span> <span class='rbrace token'>}</span><span class='rparen token'>)</span>
  <span class='comment val'># Executes codeml with the alignment and the tree</span>
  <span class='report identifier id'>report</span> <span class='assign token'>=</span> <span class='codeml identifier id'>codeml</span><span class='dot token'>.</span><span class='query identifier id'>query</span><span class='lparen token'>(</span><span class='alignment identifier id'>alignment</span><span class='comma token'>,</span> <span class='tree identifier id'>tree</span><span class='rparen token'>)</span>
</pre>
<p>
Example 2 (Obsolete usage):
</p>
<pre class="code">
  <span class='comment val'># Create a control file, setting some Codeml options</span>
  <span class='comment val'># Default parameters are used otherwise, see RDoc for defaults</span>
  <span class='comment val'># The names of the parameters correspond to those specified</span>
  <span class='comment val'># in the Codeml documentation</span>
  <span class='control_file identifier id'>control_file</span> <span class='assign token'>=</span> <span class='Tempfile constant id'>Tempfile</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'codeml_ctl'</span><span class='rparen token'>)</span>
  <span class='control_file identifier id'>control_file</span><span class='dot token'>.</span><span class='close identifier id'>close</span><span class='lparen token'>(</span><span class='false false kw'>false</span><span class='rparen token'>)</span>
  <span class='comment val'># Prepare output file as a temporary file</span>
  <span class='output_file identifier id'>output_file</span> <span class='assign token'>=</span> <span class='Tempfile constant id'>Tempfile</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'codeml_test'</span><span class='rparen token'>)</span>
  <span class='output_file identifier id'>output_file</span><span class='dot token'>.</span><span class='close identifier id'>close</span><span class='lparen token'>(</span><span class='false false kw'>false</span><span class='rparen token'>)</span>
  <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='PAML constant id'>PAML</span><span class='colon2 op'>::</span><span class='Codeml constant id'>Codeml</span><span class='dot token'>.</span><span class='create_control_file identifier id'>create_control_file</span><span class='lparen token'>(</span><span class='config_file identifier id'>config_file</span><span class='dot token'>.</span><span class='path identifier id'>path</span><span class='comma token'>,</span> <span class='lbrace token'>{</span>
    <span class='symbol val'>:model</span>       <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>1</span><span class='comma token'>,</span>
    <span class='symbol val'>:fix_kappa</span>   <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>1</span><span class='comma token'>,</span>
    <span class='symbol val'>:aaRatefile</span>  <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='TEST_DATA constant id'>TEST_DATA</span> <span class='plus op'>+</span> <span class='string val'>'/wag.dat'</span><span class='comma token'>,</span>
    <span class='symbol val'>:seqfile</span>     <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='TEST_DATA constant id'>TEST_DATA</span> <span class='plus op'>+</span> <span class='string val'>'/abglobin.aa'</span><span class='comma token'>,</span>
    <span class='symbol val'>:treefile</span>    <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='TEST_DATA constant id'>TEST_DATA</span> <span class='plus op'>+</span> <span class='string val'>'/abglobin.trees'</span><span class='comma token'>,</span>
    <span class='symbol val'>:outfile</span>     <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='output_file identifier id'>output_file</span><span class='dot token'>.</span><span class='path identifier id'>path</span><span class='comma token'>,</span>
  <span class='rbrace token'>}</span><span class='rparen token'>)</span>

  <span class='comment val'># Create an instance of Codeml specifying where the codeml binary is</span>
  <span class='codeml identifier id'>codeml</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='PAML constant id'>PAML</span><span class='colon2 op'>::</span><span class='Codeml constant id'>Codeml</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'/path/to/codeml'</span><span class='rparen token'>)</span>

  <span class='comment val'># Run codeml using a control file</span>
  <span class='comment val'># Returns the command line output</span>
  <span class='codeml_output identifier id'>codeml_output</span> <span class='assign token'>=</span> <span class='codeml identifier id'>codeml</span><span class='dot token'>.</span><span class='run identifier id'>run</span><span class='lparen token'>(</span><span class='control_file identifier id'>control_file</span><span class='rparen token'>)</span>
</pre>

</div><div class="section constants">
  <div class="thisclass">
  <h2>Constants</h2>
  <dl>
    
      <dt id="DEFAULT_PARAMETERS-constant">DEFAULT_PARAMETERS</dt>
      <dd>{
  # Essential argumemts
  :seqfile             =&gt; nil,
  :outfile             =&gt; nil,
  # Optional arguments
  :treefile            =&gt; nil,
  :noisy               =&gt; 0,
  :verbose             =&gt; 1,
  :runmode             =&gt; 0,
  :seqtype             =&gt; 2,
  :CodonFreq           =&gt; 2,
  :ndata               =&gt; 1,
  :clock               =&gt; 0,
  :aaDist              =&gt; 0,
  :aaRatefile          =&gt; 'wag.dat',
  :model               =&gt; 2,
  :NSsites             =&gt; 0,
  :icode               =&gt; 0,
  :Mgene               =&gt; 0,
  :fix_kappa           =&gt; 0,
  :kappa               =&gt; 2,
  :fix_omega           =&gt; 0,
  :omega               =&gt; 0.4,
  :fix_alpha           =&gt; 0,
  :alpha               =&gt; 0.0,
  :Malpha              =&gt; 0,
  :ncatG               =&gt; 3,
  :fix_rho             =&gt; 1,
  :rho                 =&gt; 0.0,
  :getSE               =&gt; 0,
  :RateAncestor        =&gt; 0,
  :Small_Diff          =&gt; 0.5e-6,
  :cleandata           =&gt; 1,
  :fix_blength         =&gt; 0,
  :method              =&gt; 0
}</dd>
    
      <dt id="DEFAULT_PROGRAM-constant">DEFAULT_PROGRAM</dt>
      <dd>'codeml'.freeze</dd>
    
  </dl>
</div>
<div class="inherited">
  
  <div>
    <h2>Constants Inherited from <a href="Common.html" title="Bio::PAML::Common">Bio::PAML::Common</a></h2>
    <p><span class='name'><a href="Common.html#DEFAULT_PARAMETERS_ORDER-constant" title="DEFAULT_PARAMETERS_ORDER">DEFAULT_PARAMETERS_ORDER</a></span></p>
  </div>
  
</div>
<div class="clear"></div>
</div><div class="section constructor">
<h2>Constructor</h2>
  <p class="inherited_msg">
    This class inherits a constructor from <a href="Common.html#initialize-instance_method" title="Bio::PAML::Common">Bio::PAML::Common</a>.
  </p>
</div>  <div class="section visibilitygroup">
      <div class="section methodsummary">
    <h2>Public Class Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#create_config_file-class_method">create_config_file</span><span class='args'>(parameters, filename)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
OBSOLETE.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#create_control_file-class_method">create_control_file</span><span class='args'>(parameters, filename)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Obsolete.
</p>

        
      </td>
    </tr>
  
</table>
  </div>
  <div class="section methodsummary">
    <h2>Public Instance Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#options-instance_method">#options</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
OBSOLETE.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#options-3D-instance_method">#options=</span><span class='args'>(hash)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
OBSOLETE.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#query-instance_method">#query</span><span class='args'>(alignment, tree = nil, aarate = nil)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Runs the program on the internal parameters with the specified sequence
alignment and tree.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#query_by_string-instance_method">#query_by_string</span><span class='args'>(alignment = nil, tree = nil, aarate = nil)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Runs the program on the internal parameters with the specified sequence
alignment data string and tree data string.
</p>

        
      </td>
    </tr>
  
</table><div class="inherited">
  
  <div>
    <h2>Public Instance Methods Inherited from <a href="Common.html" title="Bio::PAML::Common">Bio::PAML::Common</a></h2>
    <p><span class='name'><a href="Common.html#dump_parameters-instance_method" title="dump_parameters">dump_parameters</a></span>, <span class='name'><a href="Common.html#load_parameters-instance_method" title="load_parameters">load_parameters</a></span>, <span class='name'><a href="Common.html#run-instance_method" title="run">run</a></span>, <span class='name'><a href="Common.html#set_default_parameters-instance_method" title="set_default_parameters">set_default_parameters</a></span></p>
  </div>
  
</div>
<div class="clear"></div>
  </div>
<div class="section methoddetails">
  <h2>Public Class Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>create_config_file</h3>
</div><div id="create_config_file-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>create_config_file</span><span class='args'>(parameters, filename)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
OBSOLETE. This method will soon be removed. Instead, use
create_control_file(parameters, filename).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/paml/codeml.rb', line 160</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/paml/codeml.rb#L160">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">160
161
162
163</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='create_config_file identifier id'>create_config_file</span><span class='lparen token'>(</span><span class='parameters identifier id'>parameters</span><span class='comma token'>,</span> <span class='filename identifier id'>filename</span><span class='rparen token'>)</span>
  <span class='warn identifier id'>warn</span> <span class='string val'>&quot;The method Codeml.create_config_file(parameters, filename) will soon be removed. Instead, use Codeml.create_control_file(filename, parameters).&quot;</span>
  <span class='create_control_file identifier id'>create_control_file</span><span class='lparen token'>(</span><span class='parameters identifier id'>parameters</span><span class='comma token'>,</span> <span class='filename identifier id'>filename</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>create_control_file</h3>
</div><div id="create_control_file-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>create_control_file</span><span class='args'>(parameters, filename)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Obsolete. This method will be removed in the future. Helper method for
creating a codeml control file. Note that default parameters are
automatically merged.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/paml/codeml.rb', line 148</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/paml/codeml.rb#L148">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">148
149
150
151
152
153
154
155
156</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='create_control_file identifier id'>create_control_file</span><span class='lparen token'>(</span><span class='parameters identifier id'>parameters</span><span class='comma token'>,</span> <span class='filename identifier id'>filename</span><span class='rparen token'>)</span>
  <span class='parameters identifier id'>parameters</span> <span class='assign token'>=</span> <span class='DEFAULT_PARAMETERS constant id'>DEFAULT_PARAMETERS</span><span class='dot token'>.</span><span class='merge identifier id'>merge</span><span class='lparen token'>(</span><span class='parameters identifier id'>parameters</span><span class='rparen token'>)</span>
  <span class='File constant id'>File</span><span class='dot token'>.</span><span class='open identifier id'>open</span><span class='lparen token'>(</span><span class='filename identifier id'>filename</span><span class='comma token'>,</span> <span class='string val'>'w'</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='file identifier id'>file</span><span class='bitor op'>|</span>
    <span class='parameters identifier id'>parameters</span><span class='dot token'>.</span><span class='each identifier id'>each</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='key identifier id'>key</span><span class='comma token'>,</span> <span class='value identifier id'>value</span><span class='bitor op'>|</span>
      <span class='file identifier id'>file</span><span class='dot token'>.</span><span class='puts identifier id'>puts</span> <span class='dstring node'>&quot;#{key.to_s} = #{value.to_s}&quot;</span> <span class='if if_mod kw'>if</span> <span class='value identifier id'>value</span>
    <span class='end end kw'>end</span>
  <span class='end end kw'>end</span>
  <span class='filename identifier id'>filename</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div><div class="section methoddetails">
  <h2>Public Instance Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>options</h3>
</div><div id="options-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>options</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
OBSOLETE. This method should not be used. Instead, use parameters.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/paml/codeml.rb', line 133</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/paml/codeml.rb#L133">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">133
134
135
136</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='options identifier id'>options</span>
  <span class='warn identifier id'>warn</span> <span class='string val'>'The method Codeml#options will be changed to be used for command line arguments in the future. Instead, use Codeml#parameters.'</span>
  <span class='parameters identifier id'>parameters</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>options=</h3>
</div><div id="options-3D-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>options=</span><span class='args'>(hash)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
OBSOLETE. This method should not be used. Instead, use parameters=(hash).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/paml/codeml.rb', line 140</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/paml/codeml.rb#L140">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">140
141
142
143</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='options= identifier id'>options=</span><span class='lparen token'>(</span><span class='hash identifier id'>hash</span><span class='rparen token'>)</span>
  <span class='warn identifier id'>warn</span> <span class='string val'>'The method Codeml#options=() will be changed to be used for command line arguments in the future. Instead, use Codeml#parameters=().'</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='parameters identifier id'>parameters</span><span class='assign token'>=</span><span class='lparen token'>(</span><span class='hash identifier id'>hash</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>query</h3>
</div><div id="query-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>query</span><span class='args'>(alignment, tree = nil, aarate = nil)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Runs the program on the internal parameters with the specified sequence
alignment and tree.
</p>
<p>
Note that parameters[:seqfile] and parameters[:outfile] are always
modified, and parameters[:treefile] and parameters[:aaRatefile] are
modified when tree and aarate are specified respectively.
</p>
<p>
For other important information, see the document of
Bio::PAML::Common#query.
</p>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>alignment</em>: Bio::Alignment object or similar object

</li>
<li>(optional) <em>tree</em>: Bio::Tree object

</li>
<li>(optional) <em>aarate</em>: String or nil

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Report object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/paml/codeml.rb', line 183</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/paml/codeml.rb#L183">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">183
184
185
186
187
188
189
190
191</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='query identifier id'>query</span><span class='lparen token'>(</span><span class='alignment identifier id'>alignment</span><span class='comma token'>,</span> <span class='tree identifier id'>tree</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='aarate identifier id'>aarate</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='begin begin kw'>begin</span>
    <span class='aaratefile identifier id'>aaratefile</span> <span class='assign token'>=</span> <span class='prepare_aaratefile identifier id'>prepare_aaratefile</span><span class='lparen token'>(</span><span class='aarate identifier id'>aarate</span><span class='rparen token'>)</span>
    <span class='ret identifier id'>ret</span> <span class='assign token'>=</span> <span class='super super kw'>super</span><span class='lparen token'>(</span><span class='alignment identifier id'>alignment</span><span class='comma token'>,</span> <span class='tree identifier id'>tree</span><span class='rparen token'>)</span>
  <span class='ensure ensure kw'>ensure</span>
    <span class='finalize_aaratefile identifier id'>finalize_aaratefile</span><span class='lparen token'>(</span><span class='aaratefile identifier id'>aaratefile</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='ret identifier id'>ret</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>query_by_string</h3>
</div><div id="query_by_string-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>query_by_string</span><span class='args'>(alignment = nil, tree = nil, aarate = nil)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Runs the program on the internal parameters with the specified sequence
alignment data string and tree data string.
</p>
<p>
Note that parameters[:outfile] is always modified, and
parameters[:seqfile], parameters[:treefile], and parameters[:aaRatefile]
are modified when alignment, tree, and aarate are specified respectively.
</p>
<p>
It raises RuntimeError if seqfile is not specified in the argument or in
the parameter.
</p>
<p>
For other important information, see the document of query method.
</p>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(optional) <em>alignment</em>: String

</li>
<li>(optional) <em>tree</em>: String or nil

</li>
<li>(optional) <em>aarate</em>: String or nil

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>contents of output file (String)

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/paml/codeml.rb', line 212</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/paml/codeml.rb#L212">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">212
213
214
215
216
217
218
219
220</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='query_by_string identifier id'>query_by_string</span><span class='lparen token'>(</span><span class='alignment identifier id'>alignment</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='tree identifier id'>tree</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='aarate identifier id'>aarate</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='begin begin kw'>begin</span>
    <span class='aaratefile identifier id'>aaratefile</span> <span class='assign token'>=</span> <span class='prepare_aaratefile identifier id'>prepare_aaratefile</span><span class='lparen token'>(</span><span class='aarate identifier id'>aarate</span><span class='rparen token'>)</span>
    <span class='ret identifier id'>ret</span> <span class='assign token'>=</span> <span class='super super kw'>super</span><span class='lparen token'>(</span><span class='alignment identifier id'>alignment</span><span class='comma token'>,</span> <span class='tree identifier id'>tree</span><span class='rparen token'>)</span>
  <span class='ensure ensure kw'>ensure</span>
    <span class='finalize_aaratefile identifier id'>finalize_aaratefile</span><span class='lparen token'>(</span><span class='aaratefile identifier id'>aaratefile</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='ret identifier id'>ret</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div>
  </div>

</div>
</div>
          </div>
        </div>
      </div>                    
    </div>

    <div id="footer">
      <div class="site">
        <div class="info">
          <div class="links">
            <a href="http://github.com/blog/148-github-shirts-now-available">Shirts</a> |
            <a href="http://github.com/blog">Blog</a> |
            <a href="http://support.github.com/">Support</a> |
            <a href="http://github.com/training">Training</a> |
            <a href="http://github.com/contact">Contact</a> |
            <a href="http://groups.google.com/group/github/">Google Group</a> |
            <a href="http://develop.github.com/">API</a> |
            <a href="http://twitter.com/github">Status</a>
          </div>
          <div class="company">
            <span id="_rrt" title="0.64509s from xc88-s00039">GitHub</span>
            is <a href="http://logicalawesome.com/">Logical Awesome</a> ©2009 | <a href="http://github.com/site/terms">Terms of Service</a> | <a href="http://github.com/site/privacy">Privacy Policy</a>
          </div>
        </div>
        <div class="sponsor">
          <a href="http://engineyard.com/"><img src="http://github.com/images/modules/footer/engine_yard_logo.png" alt="Engine Yard"></a>
          <div>
            Hosting provided by our<br> partners at Engine Yard
          </div>
        </div>
      </div>
    </div>
    
    <div id="coming_soon" style="display: none;">
      This feature is coming soon.  Sit tight!
    </div>

    <div id="facebox" style="display: none;">       
      <div class="popup">         
        <table>           
          <tbody>             
            <tr>               
              <td class="tl"></td>
              <td class="b"></td>
              <td class="tr"></td>             
            </tr>             
          <tr>               
            <td class="b"></td>               
            <td class="body">                 
              <div class="content"></div>                 
              <div class="footer">                   
                <a href="#" class="close"><img src="http://github.com/facebox/closelabel.gif" title="close" class="close_image"></a>                 
              </div>               
            </td>               
            <td class="b"></td>             
          </tr>             
          <tr>               
            <td class="bl"></td>
            <td class="b"></td>
            <td class="br"></td>             
          </tr>           
        </tbody>         
      </table>       
    </div>     


    <script type="text/javascript">
    var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
    document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
    </script>
    <script type="text/javascript">
    try {
    var pageTracker = _gat._getTracker("UA-8689483-1");
    pageTracker._trackPageview();
    } catch(err) {}</script>

  </body>
</html>