<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xml:lang="en" xmlns="http://www.w3.org/1999/xhtml" lang="en">
  <head>
    <title>bioruby's bioruby - Module: Bio::Sequence::Common - GitHub Documentation</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta name="description" content="">

<link rel="search" type="application/opensearchdescription+xml" href="http://github.com/opensearch.xml" title="GitHub">
<link rel="fluid-icon" href="http://github.com/fluidicon.png" title="GitHub">
<link href="http://assets0.github.com/stylesheets/bundle.css" media="screen" rel="stylesheet" type="text/css">
<link href="http://github.com/feeds/bioruby/commits/bioruby/master" rel="alternate" title="Recent Commits to bioruby:master" type="application/atom+xml">

<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js"></script>
<script type="text/javascript" src="http://assets3.github.com/javascripts/bundle.js"></script>              
<script src="../../app.js" type="text/javascript" charset="utf-8"></script>

<script type="text/javascript" src="http://docs.github.com/javascripts/md5.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/repos.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/commits.js"></script>
<script type="text/javascript">
  $(document).ready(function(){
    if (typeof skipHeaders != 'undefined') return;
    
    GitHubRepo.init("bioruby", "bioruby", function(){ 
      GitHubRepo.links = " <a id='namespaces_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/namespaces_button.png' alt='Modules and Classes'/></a> ";
      GitHubRepo.links += " <a id='methods_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/methods_button.png' alt='Methods'/></a> ";
      $("#repo").html(GitHubRepo.content());
      $('#namespaces_button').click(function(){
        var url="/"+$(this).attr('rel')+"/namespaces/";
        $.gitbox(url);
        return false;
      });    
      $('#methods_button').click(function(){
        var url="/"+$(this).attr('rel')+"/methods/";
        $.gitbox(url);
        return false;
      });        
    });
    GitHubCommit.init("bioruby", "bioruby", "97b9284109c9a4431b92eab208509e1df6069b4b", function(){ 
      $("#commit").html(GitHubCommit.content());
    });
  });
</script>

    <link rel="stylesheet" href="../../style.css" type="text/css" charset="utf-8" />
  </head>
  <body>    
    
    
    
        <div id="main" style="padding-bottom:8em;">
      <div id="header">
        <div class="site">
          <div class="logo">
            <a href="http://github.com/"><img src="http://github.com/images/modules/header/logov3.png" alt="github"></a>
          </div>          
          <div class="topsearch">
            <form action="http://github.com/search" id="top_search_form" method="get">
              <input class="search" name="q" type="search"> <input value="Search" type="submit">
            </form>
            <div class="links">
              <a href="http://github.com/repositories">Browse</a> | <a href="http://github.com/guides">Guides</a> | <a href="http://github.com/search">Advanced</a>
            </div>
          </div>
          <div class="actions">
            <a href="http://github.com">Home</a>
            <a href="http://github.com/plans"><b><u>Pricing and Signup</u></b></a>
            <a href="http://docs.github.com/">Documentation</a>
            <a href="https://github.com/login">Login</a>
          </div>                  
        </div>
      </div>    
                                                  
      <div id="repo_menu">
        <div class="site">
          <ul>          
            <li class=""><a href="http://github.com/bioruby/bioruby/tree/">Source</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/commits/">Commits</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/network">Network</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/issues">Issues</a></li>
            <li class=""><a href="http://wiki.github.com/bioruby/bioruby">Wiki</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/graphs">Graphs</a></li>                    
            <li class="active"><a href="/bioruby/bioruby">Documentation</a></li>
          </ul>
        </div>
      </div>

      <div id="repo_sub_menu">
        <div class="site">
          <div class="joiner"></div>      
        </div>
      </div>

      <div class="site">  
        <div id="repos">
          <div id="repo" class="repo public">
            <div class="title">
              <div class="path">
                <a href="http://github.com/bioruby">bioruby</a> / <b><a href="http://github.com/bioruby/bioruby/tree">bioruby</a></b>        
                <a id="namespaces_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/namespaces_button.png" alt="Modules and Classes"/></a>
                <a id="methods_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/methods_button.png" alt="Methods"/></a>
              </div>
            </div>            
          </div>
          <div id="commit">
          </div> 
          <br/>                                       
          <div id="content">
            <div class="wikistyle">
  <div class="section module Bio_Sequence_Common">
  <h1>Module: Bio::Sequence::Common</h1>
  <div class="section docstring">
  <p>
Bio::Sequence::Common is a
Mixin[http://www.rubycentral.com/book/tut_modules.html] implementing
methods common to Bio::Sequence::AA and Bio::Sequence::NA. All of these
methods are available to either Amino Acid or Nucleic Acid sequences, and
by encapsulation are also available to Bio::Sequence objects.
</p>
<h1>USAGE</h1>
<pre class="code">
  <span class='comment val'># Create a sequence</span>
  <span class='dna identifier id'>dna</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcatgc'</span><span class='rparen token'>)</span>

  <span class='comment val'># Splice out a subsequence using a Genbank-style location string</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='splice identifier id'>splice</span><span class='lparen token'>(</span><span class='string val'>'complement(1..4)'</span><span class='rparen token'>)</span>

  <span class='comment val'># What is the base composition?</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>

  <span class='comment val'># Create a random sequence with the composition of a current sequence</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span>
</pre>

</div><div class="section constants">
  
</div>  <div class="section visibilitygroup">
      <div class="section methodsummary">
    <h2>Public Instance Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#-2B-instance_method">#+</span><span class='args'>(*arg)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Create a new sequence by adding to an existing sequence.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#-3C-3C-instance_method">#<<</span><span class='args'>(*arg)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        
        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#composition-instance_method">#composition</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns a hash of the occurrence counts for each residue or base.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#concat-instance_method">#concat</span><span class='args'>(*arg)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Add new data to the end of the current sequence.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#normalize-21-instance_method">#normalize!</span><span class='args'></span>
        <span class='block'></span>
        
          <span class='alias'>#seq!</span>
        
      </th>
      <td class="docstring">
        <p>
Normalize the current sequence, removing all whitespace and transforming
all positions to uppercase if the sequence is AA or transforming all
positions to lowercase if the sequence is NA.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#randomize-instance_method">#randomize</span><span class='args'>(hash = nil)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns a randomized sequence.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#seq-instance_method">#seq</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Create a new sequence based on the current sequence.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#splice-instance_method">#splice</span><span class='args'>(position)</span>
        <span class='block'></span>
        
          <span class='alias'>#splicing</span>
        
      </th>
      <td class="docstring">
        <p>
Return a new sequence extracted from the original using a GenBank style
position string.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#subseq-instance_method">#subseq</span><span class='args'>(s = 1, e = self.length)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns a new sequence containing the subsequence identified by the start
and end numbers given as parameters.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#to_fasta-instance_method">#to_fasta</span><span class='args'>(header = '', width = nil)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
<b>DEPRECIATED</b> Do not use! Use Bio::Sequence#output instead.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#to_s-instance_method">#to_s</span><span class='args'></span>
        <span class='block'></span>
        
          <span class='alias'>#to_str</span>
        
      </th>
      <td class="docstring">
        <p>
Return sequence as String[http://corelib.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#total-instance_method">#total</span><span class='args'>(hash)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns a float total value for the sequence given a hash of base or
residue values,.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#window_search-instance_method">#window_search</span><span class='args'>(window_size, step_size = 1)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
This method steps through a sequences in steps of &#8216;step_size&#8217;
by subsequences of &#8216;window_size&#8217;.
</p>

        
      </td>
    </tr>
  
</table>
  </div>
<div class="section methoddetails">
  <h2>Public Instance Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>+</h3>
</div><div id="-2B-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>+</span><span class='args'>(*arg)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Create a new sequence by adding to an existing sequence. The existing
sequence is not modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='s2 identifier id'>s2</span> <span class='assign token'>=</span> <span class='s identifier id'>s</span> <span class='plus op'>+</span> <span class='string val'>'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s2 identifier id'>s2</span>                                 <span class='comment val'>#=&gt; &quot;atgcatgc&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; &quot;atgc&quot;</span>
</pre>
<p>
The new sequence is of the same class as the existing sequence if the new
data was added to an existing sequence,
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='s2 identifier id'>s2</span><span class='dot token'>.</span><span class='class identifier id'>class</span> <span class='eq op'>==</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                <span class='comment val'>#=&gt; true</span>
</pre>
<p>
but if an existing sequence is added to a String, the result is a String
</p>
<pre class="code">
  <span class='s3 identifier id'>s3</span> <span class='assign token'>=</span> <span class='string val'>'atgc'</span> <span class='plus op'>+</span> <span class='s identifier id'>s</span>
  <span class='puts identifier id'>puts</span> <span class='s3 identifier id'>s3</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                           <span class='comment val'>#=&gt; String</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA/AA <b>or</b> String object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 121</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L121">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">121
122
123</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='plus op'>+</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='rparen token'>)</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='super super kw'>super</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='rparen token'>)</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>&lt;&lt;</h3>
</div><div id="-3C-3C-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'><<</span><span class='args'>(*arg)</span>
    <span class='block'></span>
  </tt>
</div>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 98</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L98">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">98
99
100</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='lshft op'>&lt;&lt;</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='rparen token'>)</span>
  <span class='concat identifier id'>concat</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>composition</h3>
</div><div id="composition-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>composition</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns a hash of the occurrence counts for each residue or base.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>              <span class='comment val'>#=&gt; {&quot;a&quot;=&gt;1, &quot;c&quot;=&gt;1, &quot;g&quot;=&gt;1, &quot;t&quot;=&gt;1}</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Hash object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 215</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L215">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">215
216
217
218
219
220
221</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='composition identifier id'>composition</span>
  <span class='count identifier id'>count</span> <span class='assign token'>=</span> <span class='Hash constant id'>Hash</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='integer val'>0</span><span class='rparen token'>)</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='scan identifier id'>scan</span><span class='lparen token'>(</span><span class='regexp val'>/./</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='x identifier id'>x</span><span class='bitor op'>|</span>
    <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='x identifier id'>x</span><span class='rbrack token'>]</span> <span class='opasgn op'>+=</span> <span class='integer val'>1</span>
  <span class='end end kw'>end</span>
  <span class='return return kw'>return</span> <span class='count identifier id'>count</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>concat</h3>
</div><div id="concat-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>concat</span><span class='args'>(*arg)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Add new data to the end of the current sequence. The original sequence is
modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='s identifier id'>s</span> <span class='lshft op'>&lt;&lt;</span> <span class='string val'>'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; &quot;atgcatgc&quot;</span>
  <span class='s identifier id'>s</span> <span class='lshft op'>&lt;&lt;</span> <span class='s identifier id'>s</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; &quot;atgcatgcatgcatgc&quot;</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>current Bio::Sequence::NA/AA object (modified)

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 94</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L94">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">94
95
96</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='concat identifier id'>concat</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='rparen token'>)</span>
  <span class='super super kw'>super</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='rparen token'>)</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>normalize!</h3>
</div><div id="normalize-21-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>normalize!</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>
<p class="aliases">
  <span class="aka">Also known as:</span>
  
    <tt id="seq!-instance_method">seq!</tt>
  
</p>
</div><div class="section docstring">
  <p>
Normalize the current sequence, removing all whitespace and transforming
all positions to uppercase if the sequence is AA or transforming all
positions to lowercase if the sequence is NA. The original sequence is
modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='normalize! fid id'>normalize!</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>current Bio::Sequence::NA/AA object (modified)

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 78</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L78">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">78
79
80
81</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='normalize! fid id'>normalize!</span>
  <span class='initialize identifier id'>initialize</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='rparen token'>)</span>
  <span class='self self kw'>self</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>randomize</h3>
</div><div id="randomize-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>randomize</span><span class='args'>(hash = nil)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns a randomized sequence. The default is to retain the same
base/residue composition as the original. If a hash of base/residue counts
is given, the new sequence will be based on that hash composition. If a
block is given, each new randomly selected position will be passed into the
block. In all cases, the original sequence is not modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span>                        <span class='comment val'>#=&gt; &quot;tcag&quot;  (for example)</span>

  <span class='new_composition identifier id'>new_composition</span> <span class='assign token'>=</span> <span class='lbrace token'>{</span><span class='string val'>'a'</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>2</span><span class='comma token'>,</span> <span class='string val'>'t'</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>2</span><span class='rbrace token'>}</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span><span class='lparen token'>(</span><span class='new_composition identifier id'>new_composition</span><span class='rparen token'>)</span>       <span class='comment val'>#=&gt; &quot;ttaa&quot;  (for example)</span>

  <span class='count identifier id'>count</span> <span class='assign token'>=</span> <span class='integer val'>0</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span> <span class='lbrace token'>{</span> <span class='bitor op'>|</span><span class='x identifier id'>x</span><span class='bitor op'>|</span> <span class='count identifier id'>count</span> <span class='opasgn op'>+=</span> <span class='integer val'>1</span> <span class='rbrace token'>}</span>
  <span class='puts identifier id'>puts</span> <span class='count identifier id'>count</span>                              <span class='comment val'>#=&gt; 4</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(optional) <em>hash</em>: Hash object

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA/AA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 243</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L243">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='randomize identifier id'>randomize</span><span class='lparen token'>(</span><span class='hash identifier id'>hash</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='if if kw'>if</span> <span class='hash identifier id'>hash</span>
    <span class='tmp identifier id'>tmp</span> <span class='assign token'>=</span> <span class='string val'>''</span>
    <span class='hash identifier id'>hash</span><span class='dot token'>.</span><span class='each identifier id'>each</span> <span class='lbrace token'>{</span><span class='bitor op'>|</span><span class='k identifier id'>k</span><span class='comma token'>,</span> <span class='v identifier id'>v</span><span class='bitor op'>|</span>
      <span class='tmp identifier id'>tmp</span> <span class='opasgn op'>+=</span> <span class='k identifier id'>k</span> <span class='mult op'>*</span> <span class='v identifier id'>v</span><span class='dot token'>.</span><span class='to_i identifier id'>to_i</span>
    <span class='rbrace token'>}</span>
  <span class='else else kw'>else</span>
    <span class='tmp identifier id'>tmp</span> <span class='assign token'>=</span> <span class='self self kw'>self</span>
  <span class='end end kw'>end</span>
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='tmp identifier id'>tmp</span><span class='rparen token'>)</span>
  <span class='comment val'># Reference: http://en.wikipedia.org/wiki/Fisher-Yates_shuffle</span>
  <span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='length identifier id'>length</span><span class='dot token'>.</span><span class='downto identifier id'>downto</span><span class='lparen token'>(</span><span class='integer val'>2</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='n identifier id'>n</span><span class='bitor op'>|</span>
    <span class='k identifier id'>k</span> <span class='assign token'>=</span> <span class='rand identifier id'>rand</span><span class='lparen token'>(</span><span class='n identifier id'>n</span><span class='rparen token'>)</span>
    <span class='c identifier id'>c</span> <span class='assign token'>=</span> <span class='seq identifier id'>seq</span><span class='lbrack token'>[</span><span class='n identifier id'>n</span> <span class='minus op'>-</span> <span class='integer val'>1</span><span class='rbrack token'>]</span>
    <span class='seq identifier id'>seq</span><span class='lbrack token'>[</span><span class='n identifier id'>n</span> <span class='minus op'>-</span> <span class='integer val'>1</span><span class='rbrack token'>]</span> <span class='assign token'>=</span> <span class='seq identifier id'>seq</span><span class='lbrack token'>[</span><span class='k identifier id'>k</span><span class='rbrack token'>]</span>
    <span class='seq identifier id'>seq</span><span class='lbrack token'>[</span><span class='k identifier id'>k</span><span class='rbrack token'>]</span> <span class='assign token'>=</span> <span class='c identifier id'>c</span>
  <span class='end end kw'>end</span>
  <span class='if if kw'>if</span> <span class='block_given? fid id'>block_given?</span> <span class='then then kw'>then</span>
    <span class='lparen token'>(</span><span class='integer val'>0</span><span class='dot3 op'>...</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='length identifier id'>length</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='each identifier id'>each</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='i identifier id'>i</span><span class='bitor op'>|</span>
      <span class='yield yield kw'>yield</span> <span class='seq identifier id'>seq</span><span class='lbrack token'>[</span><span class='i identifier id'>i</span><span class='comma token'>,</span> <span class='integer val'>1</span><span class='rbrack token'>]</span>
    <span class='end end kw'>end</span>
    <span class='return return kw'>return</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>''</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='return return kw'>return</span> <span class='seq identifier id'>seq</span>
  <span class='end end kw'>end</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>seq</h3>
</div><div id="seq-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>seq</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Create a new sequence based on the current sequence. The original sequence
is unchanged.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='s2 identifier id'>s2</span> <span class='assign token'>=</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span>
  <span class='puts identifier id'>puts</span> <span class='s2 identifier id'>s2</span>                                 <span class='comment val'>#=&gt; 'atgc'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA/AA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 65</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L65">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">65
66
67</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='seq identifier id'>seq</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>splice</h3>
</div><div id="splice-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>splice</span><span class='args'>(position)</span>
    <span class='block'></span>
  </tt>
</div>
<p class="aliases">
  <span class="aka">Also known as:</span>
  
    <tt id="splicing-instance_method">splicing</tt>
  
</p>
</div><div class="section docstring">
  <p>
Return a new sequence extracted from the original using a GenBank style
position string. See also documentation for the Bio::Location class.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcatgcatgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='splice identifier id'>splice</span><span class='lparen token'>(</span><span class='string val'>'1..3'</span><span class='rparen token'>)</span>                           <span class='comment val'>#=&gt; &quot;atg&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='splice identifier id'>splice</span><span class='lparen token'>(</span><span class='string val'>'join(1..3,8..10)'</span><span class='rparen token'>)</span>               <span class='comment val'>#=&gt; &quot;atgcat&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='splice identifier id'>splice</span><span class='lparen token'>(</span><span class='string val'>'complement(1..3)'</span><span class='rparen token'>)</span>               <span class='comment val'>#=&gt; &quot;cat&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='splice identifier id'>splice</span><span class='lparen token'>(</span><span class='string val'>'complement(join(1..3,8..10))'</span><span class='rparen token'>)</span>   <span class='comment val'>#=&gt; &quot;atgcat&quot;</span>
</pre>
<p>
Note that &#8216;complement&#8217;ed Genbank position strings will have no
effect on Bio::Sequence::AA objects.
</p>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>position</em>: String <b>or</b> Bio::Location object

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::NA/AA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 285</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L285">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">285
286
287
288
289
290
291
292
293
294
295
296
297
298
299
300
301
302
303</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='splice identifier id'>splice</span><span class='lparen token'>(</span><span class='position identifier id'>position</span><span class='rparen token'>)</span>
  <span class='unless unless kw'>unless</span> <span class='position identifier id'>position</span><span class='dot token'>.</span><span class='is_a? fid id'>is_a?</span><span class='lparen token'>(</span><span class='Locations constant id'>Locations</span><span class='rparen token'>)</span> <span class='then then kw'>then</span>
    <span class='position identifier id'>position</span> <span class='assign token'>=</span> <span class='Locations constant id'>Locations</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='position identifier id'>position</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='string val'>''</span>
  <span class='position identifier id'>position</span><span class='dot token'>.</span><span class='each identifier id'>each</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='location identifier id'>location</span><span class='bitor op'>|</span>
    <span class='if if kw'>if</span> <span class='location identifier id'>location</span><span class='dot token'>.</span><span class='sequence identifier id'>sequence</span>
      <span class='s identifier id'>s</span> <span class='lshft op'>&lt;&lt;</span> <span class='location identifier id'>location</span><span class='dot token'>.</span><span class='sequence identifier id'>sequence</span>
    <span class='else else kw'>else</span>
      <span class='exon identifier id'>exon</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='subseq identifier id'>subseq</span><span class='lparen token'>(</span><span class='location identifier id'>location</span><span class='dot token'>.</span><span class='from identifier id'>from</span><span class='comma token'>,</span> <span class='location identifier id'>location</span><span class='dot token'>.</span><span class='to identifier id'>to</span><span class='rparen token'>)</span>
      <span class='begin begin kw'>begin</span>
        <span class='exon identifier id'>exon</span><span class='dot token'>.</span><span class='complement! fid id'>complement!</span> <span class='if if_mod kw'>if</span> <span class='location identifier id'>location</span><span class='dot token'>.</span><span class='strand identifier id'>strand</span> <span class='lt op'>&lt;</span> <span class='integer val'>0</span>
      <span class='rescue rescue kw'>rescue</span> <span class='NameError constant id'>NameError</span>
      <span class='end end kw'>end</span>
      <span class='s identifier id'>s</span> <span class='lshft op'>&lt;&lt;</span> <span class='exon identifier id'>exon</span>
    <span class='end end kw'>end</span>
  <span class='end end kw'>end</span>
  <span class='return return kw'>return</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='s identifier id'>s</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>subseq</h3>
</div><div id="subseq-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>subseq</span><span class='args'>(s = 1, e = self.length)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns a new sequence containing the subsequence identified by the start
and end numbers given as parameters. *Important:* Biological sequence
numbering conventions (one-based) rather than ruby&#8217;s (zero-based)
numbering conventions are used.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggaatga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='subseq identifier id'>subseq</span><span class='lparen token'>(</span><span class='integer val'>1</span><span class='comma token'>,</span><span class='integer val'>3</span><span class='rparen token'>)</span>                      <span class='comment val'>#=&gt; &quot;atg&quot;</span>
</pre>
<p>
Start defaults to 1 and end defaults to the entire existing string, so
subseq called without any parameters simply returns a new sequence
identical to the existing sequence.
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='subseq identifier id'>subseq</span>                           <span class='comment val'>#=&gt; &quot;atggaatga&quot;</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(optional) <em>s</em>(start): Integer (default 1)

</li>
<li>(optional) <em>e</em>(end): Integer (default current sequence length)

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA/AA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 143</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L143">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">143
144
145
146
147
148</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='subseq identifier id'>subseq</span><span class='lparen token'>(</span><span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='integer val'>1</span><span class='comma token'>,</span> <span class='e identifier id'>e</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='length identifier id'>length</span><span class='rparen token'>)</span>
  <span class='raise identifier id'>raise</span> <span class='string val'>&quot;Error: start/end position must be a positive integer&quot;</span> <span class='unless unless_mod kw'>unless</span> <span class='s identifier id'>s</span> <span class='gt op'>&gt;</span> <span class='integer val'>0</span> <span class='and and kw'>and</span> <span class='e identifier id'>e</span> <span class='gt op'>&gt;</span> <span class='integer val'>0</span>
  <span class='s identifier id'>s</span> <span class='opasgn op'>-=</span> <span class='integer val'>1</span>
  <span class='e identifier id'>e</span> <span class='opasgn op'>-=</span> <span class='integer val'>1</span>
  <span class='self self kw'>self</span><span class='lbrack token'>[</span><span class='s identifier id'>s</span><span class='dot2 op'>..</span><span class='e identifier id'>e</span><span class='rbrack token'>]</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>to_fasta</h3>
</div><div id="to_fasta-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>to_fasta</span><span class='args'>(header = '', width = nil)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
<b>DEPRECIATED</b> Do not use! Use Bio::Sequence#output instead.
</p>
<p>
Output the FASTA format string of the sequence. The 1st argument is used as
the comment string. If the 2nd option is given, the output sequence will be
folded.
</p>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(optional) <em>header</em>: String object

</li>
<li>(optional) <em>width</em>: Fixnum object (default nil)

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>String

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/compat.rb', line 50</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/compat.rb#L50">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">50
51
52
53
54
55
56
57
58</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='to_fasta identifier id'>to_fasta</span><span class='lparen token'>(</span><span class='header identifier id'>header</span> <span class='assign token'>=</span> <span class='string val'>''</span><span class='comma token'>,</span> <span class='width identifier id'>width</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='warn identifier id'>warn</span> <span class='string val'>&quot;Bio::Sequence#to_fasta is obsolete. Use Bio::Sequence#output(:fasta) instead&quot;</span> <span class='if if_mod kw'>if</span> <span class='$DEBUG gvar id'>$DEBUG</span>
  <span class='dstring node'>&quot;&gt;#{header}\n&quot;</span> <span class='plus op'>+</span>
  <span class='if if kw'>if</span> <span class='width identifier id'>width</span>
    <span class='self self kw'>self</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span><span class='dot token'>.</span><span class='gsub identifier id'>gsub</span><span class='lparen token'>(</span><span class='Regexp constant id'>Regexp</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='dstring node'>&quot;.{1,#{width}}&quot;</span><span class='rparen token'>)</span><span class='comma token'>,</span> <span class='string val'>&quot;\\0\n&quot;</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='self self kw'>self</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span> <span class='plus op'>+</span> <span class='string val'>&quot;\n&quot;</span>
  <span class='end end kw'>end</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>to_s</h3>
</div><div id="to_s-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>to_s</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>
<p class="aliases">
  <span class="aka">Also known as:</span>
  
    <tt id="to_str-instance_method">to_str</tt>
  
</p>
</div><div class="section docstring">
  <p>
Return sequence as
String[http://corelib.rubyonrails.org/classes/String.html]. The original
sequence is unchanged.
</p>
<pre class="code">
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span>                             <span class='comment val'>#=&gt; 'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                       <span class='comment val'>#=&gt; String</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                            <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>String object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 52</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L52">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">52
53
54</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='to_s identifier id'>to_s</span>
  <span class='String constant id'>String</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>total</h3>
</div><div id="total-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>total</span><span class='args'>(hash)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns a float total value for the sequence given a hash of base or
residue values,
</p>
<pre class="code">
  <span class='values identifier id'>values</span> <span class='assign token'>=</span> <span class='lbrace token'>{</span><span class='string val'>'a'</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>0</span><span class='integer val'>.1</span><span class='comma token'>,</span> <span class='string val'>'t'</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>0</span><span class='integer val'>.2</span><span class='comma token'>,</span> <span class='string val'>'g'</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>0</span><span class='integer val'>.3</span><span class='comma token'>,</span> <span class='string val'>'c'</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='integer val'>0</span><span class='integer val'>.4</span><span class='rbrace token'>}</span>
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='total identifier id'>total</span><span class='lparen token'>(</span><span class='values identifier id'>values</span><span class='rparen token'>)</span>                    <span class='comment val'>#=&gt; 1.0</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>hash</em>: Hash object

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Float object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 198</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L198">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">198
199
200
201
202
203
204
205
206
207</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='total identifier id'>total</span><span class='lparen token'>(</span><span class='hash identifier id'>hash</span><span class='rparen token'>)</span>
  <span class='hash identifier id'>hash</span><span class='dot token'>.</span><span class='default identifier id'>default</span> <span class='assign token'>=</span> <span class='integer val'>0</span><span class='integer val'>.0</span> <span class='unless unless_mod kw'>unless</span> <span class='hash identifier id'>hash</span><span class='dot token'>.</span><span class='default identifier id'>default</span>
  <span class='sum identifier id'>sum</span> <span class='assign token'>=</span> <span class='integer val'>0</span><span class='integer val'>.0</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='each_byte identifier id'>each_byte</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='x identifier id'>x</span><span class='bitor op'>|</span>
    <span class='begin begin kw'>begin</span>
      <span class='sum identifier id'>sum</span> <span class='opasgn op'>+=</span> <span class='hash identifier id'>hash</span><span class='lbrack token'>[</span><span class='x identifier id'>x</span><span class='dot token'>.</span><span class='chr identifier id'>chr</span><span class='rbrack token'>]</span>
    <span class='end end kw'>end</span>
  <span class='end end kw'>end</span>
  <span class='return return kw'>return</span> <span class='sum identifier id'>sum</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>window_search</h3>
</div><div id="window_search-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>window_search</span><span class='args'>(window_size, step_size = 1)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
This method steps through a sequences in steps of &#8216;step_size&#8217;
by subsequences of &#8216;window_size&#8217;. Typically used with a block.
Any remaining sequence at the terminal end will be returned.
</p>
<p>
Prints average GC% on each 100bp
</p>
<pre class="code">
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='window_search identifier id'>window_search</span><span class='lparen token'>(</span><span class='integer val'>100</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='subseq identifier id'>subseq</span><span class='bitor op'>|</span>
    <span class='puts identifier id'>puts</span> <span class='subseq identifier id'>subseq</span><span class='dot token'>.</span><span class='gc identifier id'>gc</span>
  <span class='end end kw'>end</span>
</pre>
<p>
Prints every translated peptide (length 5aa) in the same frame
</p>
<pre class="code">
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='window_search identifier id'>window_search</span><span class='lparen token'>(</span><span class='integer val'>15</span><span class='comma token'>,</span> <span class='integer val'>3</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='subseq identifier id'>subseq</span><span class='bitor op'>|</span>
    <span class='puts identifier id'>puts</span> <span class='subseq identifier id'>subseq</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span>
  <span class='end end kw'>end</span>
</pre>
<p>
Split genome sequence by 10000bp with 1000bp overlap in fasta format
</p>
<pre class="code">
  <span class='i identifier id'>i</span> <span class='assign token'>=</span> <span class='integer val'>1</span>
  <span class='remainder identifier id'>remainder</span> <span class='assign token'>=</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='window_search identifier id'>window_search</span><span class='lparen token'>(</span><span class='integer val'>10000</span><span class='comma token'>,</span> <span class='integer val'>9000</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='subseq identifier id'>subseq</span><span class='bitor op'>|</span>
    <span class='puts identifier id'>puts</span> <span class='subseq identifier id'>subseq</span><span class='dot token'>.</span><span class='to_fasta identifier id'>to_fasta</span><span class='lparen token'>(</span><span class='dstring node'>&quot;segment #{i}&quot;</span><span class='comma token'>,</span> <span class='integer val'>60</span><span class='rparen token'>)</span>
    <span class='i identifier id'>i</span> <span class='opasgn op'>+=</span> <span class='integer val'>1</span>
  <span class='end end kw'>end</span>
  <span class='puts identifier id'>puts</span> <span class='remainder identifier id'>remainder</span><span class='dot token'>.</span><span class='to_fasta identifier id'>to_fasta</span><span class='lparen token'>(</span><span class='dstring node'>&quot;segment #{i}&quot;</span><span class='comma token'>,</span> <span class='integer val'>60</span><span class='rparen token'>)</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>window_size</em>: Fixnum

</li>
<li>(optional) <em>step_size</em>: Fixnum (default 1)

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA/AA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/common.rb', line 179</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/common.rb#L179">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">179
180
181
182
183
184
185
186</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='window_search identifier id'>window_search</span><span class='lparen token'>(</span><span class='window_size identifier id'>window_size</span><span class='comma token'>,</span> <span class='step_size identifier id'>step_size</span> <span class='assign token'>=</span> <span class='integer val'>1</span><span class='rparen token'>)</span>
  <span class='last_step identifier id'>last_step</span> <span class='assign token'>=</span> <span class='integer val'>0</span>
  <span class='integer val'>0</span><span class='dot token'>.</span><span class='step identifier id'>step</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='dot token'>.</span><span class='length identifier id'>length</span> <span class='minus op'>-</span> <span class='window_size identifier id'>window_size</span><span class='comma token'>,</span> <span class='step_size identifier id'>step_size</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='i identifier id'>i</span><span class='bitor op'>|</span> 
    <span class='yield yield kw'>yield</span> <span class='self self kw'>self</span><span class='lbrack token'>[</span><span class='i identifier id'>i</span><span class='comma token'>,</span> <span class='window_size identifier id'>window_size</span><span class='rbrack token'>]</span>                        
    <span class='last_step identifier id'>last_step</span> <span class='assign token'>=</span> <span class='i identifier id'>i</span>
  <span class='end end kw'>end</span>                          
  <span class='return return kw'>return</span> <span class='self self kw'>self</span><span class='lbrack token'>[</span><span class='last_step identifier id'>last_step</span> <span class='plus op'>+</span> <span class='window_size identifier id'>window_size</span> <span class='dot2 op'>..</span> <span class='integer val'>-1</span><span class='rbrack token'>]</span> 
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div>
  </div>

</div>
</div>
          </div>
        </div>
      </div>                    
    </div>

    <div id="footer">
      <div class="site">
        <div class="info">
          <div class="links">
            <a href="http://github.com/blog/148-github-shirts-now-available">Shirts</a> |
            <a href="http://github.com/blog">Blog</a> |
            <a href="http://support.github.com/">Support</a> |
            <a href="http://github.com/training">Training</a> |
            <a href="http://github.com/contact">Contact</a> |
            <a href="http://groups.google.com/group/github/">Google Group</a> |
            <a href="http://develop.github.com/">API</a> |
            <a href="http://twitter.com/github">Status</a>
          </div>
          <div class="company">
            <span id="_rrt" title="0.64509s from xc88-s00039">GitHub</span>
            is <a href="http://logicalawesome.com/">Logical Awesome</a> 2009 | <a href="http://github.com/site/terms">Terms of Service</a> | <a href="http://github.com/site/privacy">Privacy Policy</a>
          </div>
        </div>
        <div class="sponsor">
          <a href="http://engineyard.com/"><img src="http://github.com/images/modules/footer/engine_yard_logo.png" alt="Engine Yard"></a>
          <div>
            Hosting provided by our<br> partners at Engine Yard
          </div>
        </div>
      </div>
    </div>
    
    <div id="coming_soon" style="display: none;">
      This feature is coming soon.  Sit tight!
    </div>

    <div id="facebox" style="display: none;">       
      <div class="popup">         
        <table>           
          <tbody>             
            <tr>               
              <td class="tl"></td>
              <td class="b"></td>
              <td class="tr"></td>             
            </tr>             
          <tr>               
            <td class="b"></td>               
            <td class="body">                 
              <div class="content"></div>                 
              <div class="footer">                   
                <a href="#" class="close"><img src="http://github.com/facebox/closelabel.gif" title="close" class="close_image"></a>                 
              </div>               
            </td>               
            <td class="b"></td>             
          </tr>             
          <tr>               
            <td class="bl"></td>
            <td class="b"></td>
            <td class="br"></td>             
          </tr>           
        </tbody>         
      </table>       
    </div>     


    <script type="text/javascript">
    var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
    document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
    </script>
    <script type="text/javascript">
    try {
    var pageTracker = _gat._getTracker("UA-8689483-1");
    pageTracker._trackPageview();
    } catch(err) {}</script>

  </body>
</html>