<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xml:lang="en" xmlns="http://www.w3.org/1999/xhtml" lang="en">
  <head>
    <title>bioruby's bioruby - Class: Bio::Sequence::NA - GitHub Documentation</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta name="description" content="">

<link rel="search" type="application/opensearchdescription+xml" href="http://github.com/opensearch.xml" title="GitHub">
<link rel="fluid-icon" href="http://github.com/fluidicon.png" title="GitHub">
<link href="http://assets0.github.com/stylesheets/bundle.css" media="screen" rel="stylesheet" type="text/css">
<link href="http://github.com/feeds/bioruby/commits/bioruby/master" rel="alternate" title="Recent Commits to bioruby:master" type="application/atom+xml">

<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js"></script>
<script type="text/javascript" src="http://assets3.github.com/javascripts/bundle.js"></script>              
<script src="../../app.js" type="text/javascript" charset="utf-8"></script>

<script type="text/javascript" src="http://docs.github.com/javascripts/md5.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/repos.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/commits.js"></script>
<script type="text/javascript">
  $(document).ready(function(){
    if (typeof skipHeaders != 'undefined') return;
    
    GitHubRepo.init("bioruby", "bioruby", function(){ 
      GitHubRepo.links = " <a id='namespaces_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/namespaces_button.png' alt='Modules and Classes'/></a> ";
      GitHubRepo.links += " <a id='methods_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/methods_button.png' alt='Methods'/></a> ";
      $("#repo").html(GitHubRepo.content());
      $('#namespaces_button').click(function(){
        var url="/"+$(this).attr('rel')+"/namespaces/";
        $.gitbox(url);
        return false;
      });    
      $('#methods_button').click(function(){
        var url="/"+$(this).attr('rel')+"/methods/";
        $.gitbox(url);
        return false;
      });        
    });
    GitHubCommit.init("bioruby", "bioruby", "97b9284109c9a4431b92eab208509e1df6069b4b", function(){ 
      $("#commit").html(GitHubCommit.content());
    });
  });
</script>

    <link rel="stylesheet" href="../../style.css" type="text/css" charset="utf-8" />
  </head>
  <body>    
    
    
    
        <div id="main" style="padding-bottom:8em;">
      <div id="header">
        <div class="site">
          <div class="logo">
            <a href="http://github.com/"><img src="http://github.com/images/modules/header/logov3.png" alt="github"></a>
          </div>          
          <div class="topsearch">
            <form action="http://github.com/search" id="top_search_form" method="get">
              <input class="search" name="q" type="search"> <input value="Search" type="submit">
            </form>
            <div class="links">
              <a href="http://github.com/repositories">Browse</a> | <a href="http://github.com/guides">Guides</a> | <a href="http://github.com/search">Advanced</a>
            </div>
          </div>
          <div class="actions">
            <a href="http://github.com">Home</a>
            <a href="http://github.com/plans"><b><u>Pricing and Signup</u></b></a>
            <a href="http://docs.github.com/">Documentation</a>
            <a href="https://github.com/login">Login</a>
          </div>                  
        </div>
      </div>    
                                                  
      <div id="repo_menu">
        <div class="site">
          <ul>          
            <li class=""><a href="http://github.com/bioruby/bioruby/tree/">Source</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/commits/">Commits</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/network">Network</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/issues">Issues</a></li>
            <li class=""><a href="http://wiki.github.com/bioruby/bioruby">Wiki</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/graphs">Graphs</a></li>                    
            <li class="active"><a href="/bioruby/bioruby">Documentation</a></li>
          </ul>
        </div>
      </div>

      <div id="repo_sub_menu">
        <div class="site">
          <div class="joiner"></div>      
        </div>
      </div>

      <div class="site">  
        <div id="repos">
          <div id="repo" class="repo public">
            <div class="title">
              <div class="path">
                <a href="http://github.com/bioruby">bioruby</a> / <b><a href="http://github.com/bioruby/bioruby/tree">bioruby</a></b>        
                <a id="namespaces_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/namespaces_button.png" alt="Modules and Classes"/></a>
                <a id="methods_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/methods_button.png" alt="Methods"/></a>
              </div>
            </div>            
          </div>
          <div id="commit">
          </div> 
          <br/>                                       
          <div id="content">
            <div class="wikistyle">
  <div class="section class Bio_Sequence_NA">
  <h1>Class: Bio::Sequence::NA</h1>
  <div class="section inheritance">
  <ul>
  <li><a href="../../String.html" title="String">String</a></li><ul><li>Bio::Sequence::NA</li>
  </ul></ul>
</div><div class="section mixins">
  <h2>Included Modules</h2>
  <p><a href="Common.html" title="Bio::Sequence::Common">Bio::Sequence::Common</a></p>
</div><div class="section docstring">
  <p>
<b>TODO</b>
</p>
<pre class="code">
  <span class='minus op'>-</span> <span class='add identifier id'>add</span> <span class='string val'>&quot;Ohno&quot;</span> <span class='style identifier id'>style</span>
  <span class='minus op'>-</span> <span class='add identifier id'>add</span> <span class='a identifier id'>a</span> <span class='accessor identifier id'>accessor</span> <span class='to identifier id'>to</span> <span class='drum identifier id'>drum</span> <span class='pattern identifier id'>pattern</span>
  <span class='minus op'>-</span> <span class='add identifier id'>add</span> <span class='a identifier id'>a</span> <span class='new identifier id'>new</span> <span class='feature identifier id'>feature</span> <span class='to identifier id'>to</span> <span class='select identifier id'>select</span> <span class='music identifier id'>music</span> <span class='style identifier id'>style</span> <span class='lparen token'>(</span><span class='pop identifier id'>pop</span><span class='comma token'>,</span> <span class='trans identifier id'>trans</span><span class='comma token'>,</span> <span class='ryukyu identifier id'>ryukyu</span><span class='comma token'>,</span> <span class='dot3 op'>...</span><span class='rparen token'>)</span>
  <span class='minus op'>-</span> <span class='what identifier id'>what</span> <span class='is identifier id'>is</span> <span class='the identifier id'>the</span> <span class='base? fid id'>base?</span>
</pre>
<p>
++
</p>

</div><div class="section constants">
  
</div><div class="section constructor">
<h2>Constructor</h2>
  <div id="initialize-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>initialize</span><span class='args'>(str)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Generate an nucleic acid sequence object from a string.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>&quot;aagcttggaccgttgaagt&quot;</span><span class='rparen token'>)</span>
</pre>
<p>
or maybe (if you have an nucleic acid sequence in a file)
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='symbol val'>:NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='File constant id'>File</span><span class='dot token'>.</span><span class='open identifier id'>open</span><span class='lparen token'>(</span><span class='string val'>'dna.txt'</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='read identifier id'>read</span><span class='rparen token'>)</span>
</pre>
<p>
Nucleic Acid sequences are <b>always</b> all lowercase in bioruby
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>&quot;AAGcTtGG&quot;</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; &quot;aagcttgg&quot;</span>
</pre>
<p>
Whitespace is stripped from the sequence
</p>
<pre class="code">
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>&quot;atg\nggg\ttt\r  gc&quot;</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; &quot;atggggttgc&quot;</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>str</em>: String

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::NA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 77</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L77">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">77
78
79
80
81</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='initialize identifier id'>initialize</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
  <span class='super super kw'>super</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='downcase! fid id'>downcase!</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='tr! fid id'>tr!</span><span class='lparen token'>(</span><span class='string val'>&quot; \t\n\r&quot;</span><span class='comma token'>,</span><span class='string val'>''</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
</div>  <div class="section visibilitygroup">
      <div class="section methodsummary">
    <h2>Public Class Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#randomize-class_method">randomize</span><span class='args'>(*arg, &amp;block)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Generate a new random sequence with the given frequency of bases.
</p>

        
      </td>
    </tr>
  
</table>
  </div>
  <div class="section methodsummary">
    <h2>Public Instance Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#at_content-instance_method">#at_content</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Calculate the ratio of AT / ATGC bases.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#at_skew-instance_method">#at_skew</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Calculate the ratio of (A - T) / (A + T) bases.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#codon_usage-instance_method">#codon_usage</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns counts of each codon in the sequence in a hash.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#cut_with_enzyme-instance_method">#cut_with_enzyme</span><span class='args'>(*args)</span>
        <span class='block'></span>
        
          <span class='alias'>#cut_with_enzymes</span>
        
      </th>
      <td class="docstring">
        <p>
Example:.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#dna-instance_method">#dna</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns a new sequence object with any &#8216;u&#8217; bases changed to
&#8216;t&#8217;.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#dna-21-instance_method">#dna!</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Changes any &#8216;u&#8217; bases in the original sequence to
&#8216;t&#8217;.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#forward_complement-instance_method">#forward_complement</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns a new complementary sequence object (without reversing).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#forward_complement-21-instance_method">#forward_complement!</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Converts the current sequence into its complement (without reversing).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#gc_content-instance_method">#gc_content</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Calculate the ratio of GC / ATGC bases.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#gc_percent-instance_method">#gc_percent</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Calculate the ratio of GC / ATGC bases as a percentage rounded to the
nearest whole number.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#gc_skew-instance_method">#gc_skew</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Calculate the ratio of (G - C) / (G + C) bases.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#illegal_bases-instance_method">#illegal_bases</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns an alphabetically sorted array of any non-standard bases (other
than &#8216;atgcu&#8217;).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#molecular_weight-instance_method">#molecular_weight</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Estimate molecular weight (using the values from BioPerl&#8217;s SeqStats.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#names-instance_method">#names</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Generate the list of the names of each nucleotide along with the sequence
(full name).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#pikachu-instance_method">#pikachu</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        
        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#reverse_complement-instance_method">#reverse_complement</span><span class='args'></span>
        <span class='block'></span>
        
          <span class='alias'>#complement</span>
        
      </th>
      <td class="docstring">
        <p>
Returns a new sequence object with the reverse complement sequence to the
original.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#reverse_complement-21-instance_method">#reverse_complement!</span><span class='args'></span>
        <span class='block'></span>
        
          <span class='alias'>#complement!</span>
        
      </th>
      <td class="docstring">
        <p>
Converts the original sequence into its reverse complement.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#rna-instance_method">#rna</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns a new sequence object with any &#8216;t&#8217; bases changed to
&#8216;u&#8217;.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#rna-21-instance_method">#rna!</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Changes any &#8216;t&#8217; bases in the original sequence to
&#8216;u&#8217;.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#splicing-instance_method">#splicing</span><span class='args'>(position)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Alias of Bio::Sequence::Common splice method, documented there.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#to_midi-instance_method">#to_midi</span><span class='args'>(style = {}, drum = true)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
style:
</p>
<pre class="code">
  <span class='Hash constant id'>Hash</span> <span class='of identifier id'>of</span> <span class='symbol val'>:tempo</span><span class='comma token'>,</span> <span class='symbol val'>:scale</span><span class='comma token'>,</span> <span class='symbol val'>:tones</span>
</pre>
<p>
scale:
</p>
<pre class="code">
  <span class='C constant id'>C</span>  <span class='C constant id'>C</span><span class='comment val'># D  D# E  F  F# G  G# A  A#  B</span>
  <span class='integer val'>0</span>  <span class='integer val'>1</span>  <span class='integer val'>2</span>  <span class='integer val'>3</span>  <span class='integer val'>4</span>  <span class='integer val'>5</span>  <span class='integer val'>6</span>  <span class='integer val'>7</span>  <span class='integer val'>8</span>  <span class='integer val'>9</span>  <span class='integer val'>10</span>  <span class='integer val'>11</span>
</pre>
<p>
tones:
</p>
<pre class="code">
  <span class='Hash constant id'>Hash</span> <span class='of identifier id'>of</span> <span class='symbol val'>:prog</span><span class='comma token'>,</span> <span class='symbol val'>:base</span><span class='comma token'>,</span> <span class='symbol val'>:range</span> <span class='minus op'>-</span><span class='minus op'>-</span> <span class='tone identifier id'>tone</span><span class='comma token'>,</span> <span class='vol? fid id'>vol?</span> <span class='or or kw'>or</span> <span class='len? fid id'>len?</span><span class='comma token'>,</span> <span class='octaves identifier id'>octaves</span>
</pre>
<p>
drum:
</p>
<pre class="code">
  <span class='true true kw'>true</span> <span class='lparen token'>(</span><span class='with identifier id'>with</span> <span class='rhythm identifier id'>rhythm</span> <span class='part identifier id'>part</span><span class='rparen token'>)</span><span class='comma token'>,</span> <span class='false false kw'>false</span> <span class='lparen token'>(</span><span class='without identifier id'>without</span> <span class='rhythm identifier id'>rhythm</span> <span class='part identifier id'>part</span><span class='rparen token'>)</span><span class='dot token'>.</span>
</pre>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#to_re-instance_method">#to_re</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Create a ruby regular expression instance (Regexp)[http://corelib.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#translate-instance_method">#translate</span><span class='args'>(frame = 1, table = 1, unknown = 'X')</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Translate into an amino acid sequence.
</p>

        
      </td>
    </tr>
  
</table><div class="inherited">
  
  <div>
    <h2>Public Instance Methods Inherited from <a href="../../String.html" title="String">String</a></h2>
    <p><span class='name'><a href="../../String.html#fill-instance_method" title="fill">fill</a></span>, <span class='name'><a href="../../String.html#fold-instance_method" title="fold">fold</a></span>, <span class='name'><a href="../../String.html#skip-instance_method" title="skip">skip</a></span>, <span class='name'><a href="../../String.html#step-instance_method" title="step">step</a></span>, <span class='name'><a href="../../String.html#to_aaseq-instance_method" title="to_aaseq">to_aaseq</a></span>, <span class='name'><a href="../../String.html#to_naseq-instance_method" title="to_naseq">to_naseq</a></span></p>
  </div>
  
</div>
<div class="clear"></div><div class="included">
  
  <div>
    <h2>Public Instance Methods Included from <a href="Common.html" title="Bio::Sequence::Common">Bio::Sequence::Common</a></h2>
    <p><span class='name'><a href="Common.html#%2B-instance_method" title="+">+</a></span>, <span class='name'><a href="Common.html#%3C%3C-instance_method" title="&lt;&lt;">&lt;&lt;</a></span>, <span class='name'><a href="Common.html#composition-instance_method" title="composition">composition</a></span>, <span class='name'><a href="Common.html#concat-instance_method" title="concat">concat</a></span>, <span class='name'><a href="Common.html#normalize%21-instance_method" title="normalize!">normalize!</a></span>, <span class='name'><a href="Common.html#randomize-instance_method" title="randomize">randomize</a></span>, <span class='name'><a href="Common.html#seq-instance_method" title="seq">seq</a></span>, <span class='name'><a href="Common.html#splice-instance_method" title="splice">splice</a></span>, <span class='name'><a href="Common.html#subseq-instance_method" title="subseq">subseq</a></span>, <span class='name'><a href="Common.html#to_fasta-instance_method" title="to_fasta">to_fasta</a></span>, <span class='name'><a href="Common.html#to_s-instance_method" title="to_s">to_s</a></span>, <span class='name'><a href="Common.html#total-instance_method" title="total">total</a></span>, <span class='name'><a href="Common.html#window_search-instance_method" title="window_search">window_search</a></span></p>
  </div>
  
</div>
<div class="clear"></div>
  </div>
<div class="section methoddetails">
  <h2>Public Class Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>randomize</h3>
</div><div id="randomize-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>randomize</span><span class='args'>(*arg, &amp;block)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Generate a new random sequence with the given frequency of bases. The
sequence length is determined by their cumulative sum. (See also
Bio::Sequence::Common#randomize which creates a new randomized sequence
object using the base composition of an existing sequence instance).
</p>
<pre class="code">
  <span class='counts identifier id'>counts</span> <span class='assign token'>=</span> <span class='lbrace token'>{</span><span class='string val'>'a'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>1</span><span class='comma token'>,</span><span class='string val'>'c'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>2</span><span class='comma token'>,</span><span class='string val'>'g'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>3</span><span class='comma token'>,</span><span class='string val'>'t'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>4</span><span class='rbrace token'>}</span>
  <span class='puts identifier id'>puts</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span><span class='lparen token'>(</span><span class='counts identifier id'>counts</span><span class='rparen token'>)</span>  <span class='comment val'>#=&gt; &quot;ggcttgttac&quot; (for example)</span>
</pre>
<p>
You may also feed the output of randomize into a block
</p>
<pre class="code">
  <span class='actual_counts identifier id'>actual_counts</span> <span class='assign token'>=</span> <span class='lbrace token'>{</span><span class='string val'>'a'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>0</span><span class='comma token'>,</span> <span class='string val'>'c'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>0</span><span class='comma token'>,</span> <span class='string val'>'g'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>0</span><span class='comma token'>,</span> <span class='string val'>'t'</span><span class='assign token'>=</span><span class='gt op'>&gt;</span><span class='integer val'>0</span><span class='rbrace token'>}</span>
  <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span><span class='lparen token'>(</span><span class='counts identifier id'>counts</span><span class='rparen token'>)</span> <span class='lbrace token'>{</span><span class='bitor op'>|</span><span class='x identifier id'>x</span><span class='bitor op'>|</span> <span class='actual_counts identifier id'>actual_counts</span><span class='lbrack token'>[</span><span class='x identifier id'>x</span><span class='rbrack token'>]</span> <span class='opasgn op'>+=</span> <span class='integer val'>1</span><span class='rbrace token'>}</span>
  <span class='actual_counts identifier id'>actual_counts</span>                     <span class='comment val'>#=&gt; {&quot;a&quot;=&gt;1, &quot;c&quot;=&gt;2, &quot;g&quot;=&gt;3, &quot;t&quot;=&gt;4}</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(optional) <em>hash</em>: Hash object

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::NA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/compat.rb', line 83</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/compat.rb#L83">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">83
84
85</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='comma token'>,</span> <span class='bitand op'>&amp;</span><span class='block identifier id'>block</span><span class='rparen token'>)</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>''</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='randomize identifier id'>randomize</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='arg identifier id'>arg</span><span class='comma token'>,</span> <span class='bitand op'>&amp;</span><span class='block identifier id'>block</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div><div class="section methoddetails">
  <h2>Public Instance Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>at_content</h3>
</div><div id="at_content-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>at_content</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Calculate the ratio of AT / ATGC bases. U is regarded as T.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='at_content identifier id'>at_content</span>                       <span class='comment val'>#=&gt; 0.444444444444444</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Float

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 319</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L319">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">319
320
321
322
323
324
325</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='at_content identifier id'>at_content</span>
  <span class='count identifier id'>count</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>
  <span class='at identifier id'>at</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'a'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'t'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'u'</span><span class='rbrack token'>]</span>
  <span class='gc identifier id'>gc</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'g'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'c'</span><span class='rbrack token'>]</span>
  <span class='return return kw'>return</span> <span class='integer val'>0</span><span class='integer val'>.0</span> <span class='if if_mod kw'>if</span> <span class='at identifier id'>at</span> <span class='plus op'>+</span> <span class='gc identifier id'>gc</span> <span class='eq op'>==</span> <span class='integer val'>0</span>
  <span class='return return kw'>return</span> <span class='at identifier id'>at</span><span class='dot token'>.</span><span class='quo identifier id'>quo</span><span class='lparen token'>(</span><span class='at identifier id'>at</span> <span class='plus op'>+</span> <span class='gc identifier id'>gc</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>at_skew</h3>
</div><div id="at_skew-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>at_skew</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Calculate the ratio of (A - T) / (A + T) bases. U is regarded as T.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgttgttgttc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='at_skew identifier id'>at_skew</span>                          <span class='comment val'>#=&gt; -0.75</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Float

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 347</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L347">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">347
348
349
350
351
352
353</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='at_skew identifier id'>at_skew</span>
  <span class='count identifier id'>count</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>
  <span class='a identifier id'>a</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'a'</span><span class='rbrack token'>]</span>
  <span class='t identifier id'>t</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'t'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'u'</span><span class='rbrack token'>]</span>
  <span class='return return kw'>return</span> <span class='integer val'>0</span><span class='integer val'>.0</span> <span class='if if_mod kw'>if</span> <span class='a identifier id'>a</span> <span class='plus op'>+</span> <span class='t identifier id'>t</span> <span class='eq op'>==</span> <span class='integer val'>0</span>
  <span class='return return kw'>return</span> <span class='lparen token'>(</span><span class='a identifier id'>a</span> <span class='minus op'>-</span> <span class='t identifier id'>t</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='quo identifier id'>quo</span><span class='lparen token'>(</span><span class='a identifier id'>a</span> <span class='plus op'>+</span> <span class='t identifier id'>t</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>codon_usage</h3>
</div><div id="codon_usage-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>codon_usage</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns counts of each codon in the sequence in a hash.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='codon_usage identifier id'>codon_usage</span>                <span class='comment val'>#=&gt; {&quot;gcg&quot;=&gt;1, &quot;tga&quot;=&gt;1, &quot;atg&quot;=&gt;1}</span>
</pre>
<p>
This method does not validate codons! Any three letter group is a
&#8216;codon&#8217;. So,
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggNNtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='codon_usage identifier id'>codon_usage</span>                <span class='comment val'>#=&gt; {&quot;tga&quot;=&gt;1, &quot;gnn&quot;=&gt;1, &quot;atg&quot;=&gt;1}</span>

  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgg--tga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='codon_usage identifier id'>codon_usage</span>                <span class='comment val'>#=&gt; {&quot;tga&quot;=&gt;1, &quot;g--&quot;=&gt;1, &quot;atg&quot;=&gt;1}</span>
</pre>
<p>
Also, there is no option to work in any frame other than the first.
</p>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Hash object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 275</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L275">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">275
276
277
278
279
280
281</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='codon_usage identifier id'>codon_usage</span>
  <span class='hash identifier id'>hash</span> <span class='assign token'>=</span> <span class='Hash constant id'>Hash</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='integer val'>0</span><span class='rparen token'>)</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='window_search identifier id'>window_search</span><span class='lparen token'>(</span><span class='integer val'>3</span><span class='comma token'>,</span> <span class='integer val'>3</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='codon identifier id'>codon</span><span class='bitor op'>|</span>
    <span class='hash identifier id'>hash</span><span class='lbrack token'>[</span><span class='codon identifier id'>codon</span><span class='rbrack token'>]</span> <span class='opasgn op'>+=</span> <span class='integer val'>1</span>
  <span class='end end kw'>end</span>
  <span class='return return kw'>return</span> <span class='hash identifier id'>hash</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>cut_with_enzyme</h3>
</div><div id="cut_with_enzyme-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>cut_with_enzyme</span><span class='args'>(*args)</span>
    <span class='block'></span>
  </tt>
</div>
<p class="aliases">
  <span class="aka">Also known as:</span>
  
    <tt id="cut_with_enzymes-instance_method">cut_with_enzymes</tt>
  
</p>
</div><div class="section docstring">
  <p>
Example:
</p>
<pre class="code">
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'gaattc'</span><span class='rparen token'>)</span>
  <span class='cuts identifier id'>cuts</span> <span class='assign token'>=</span> <span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='cut_with_enzyme identifier id'>cut_with_enzyme</span><span class='lparen token'>(</span><span class='string val'>'EcoRI'</span><span class='rparen token'>)</span>
</pre>
<p>
<em>or</em>
</p>
<pre class="code">
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'gaattc'</span><span class='rparen token'>)</span>
  <span class='cuts identifier id'>cuts</span> <span class='assign token'>=</span> <span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='cut_with_enzyme identifier id'>cut_with_enzyme</span><span class='lparen token'>(</span><span class='string val'>'g^aattc'</span><span class='rparen token'>)</span>
</pre>
<hr size="1"></hr><p>
See Bio::RestrictionEnzyme::Analysis.cut
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 481</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L481">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">481
482
483</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='cut_with_enzyme identifier id'>cut_with_enzyme</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='args identifier id'>args</span><span class='rparen token'>)</span>
  <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='RestrictionEnzyme constant id'>RestrictionEnzyme</span><span class='colon2 op'>::</span><span class='Analysis constant id'>Analysis</span><span class='dot token'>.</span><span class='cut identifier id'>cut</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='comma token'>,</span> <span class='mult op'>*</span><span class='args identifier id'>args</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>dna</h3>
</div><div id="dna-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>dna</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns a new sequence object with any &#8216;u&#8217; bases changed to
&#8216;t&#8217;. The original sequence is not modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'augc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='dna identifier id'>dna</span>                              <span class='comment val'>#=&gt; 'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'augc'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 425</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L425">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">425
426
427</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='dna identifier id'>dna</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='tr identifier id'>tr</span><span class='lparen token'>(</span><span class='string val'>'u'</span><span class='comma token'>,</span> <span class='string val'>'t'</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>dna!</h3>
</div><div id="dna-21-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>dna!</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Changes any &#8216;u&#8217; bases in the original sequence to
&#8216;t&#8217;. The original sequence is modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'augc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='dna! fid id'>dna!</span>                             <span class='comment val'>#=&gt; 'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'atgc'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>current Bio::Sequence::NA object (modified)

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 437</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L437">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">437
438
439</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='dna! fid id'>dna!</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='tr! fid id'>tr!</span><span class='lparen token'>(</span><span class='string val'>'u'</span><span class='comma token'>,</span> <span class='string val'>'t'</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>forward_complement</h3>
</div><div id="forward_complement-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>forward_complement</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns a new complementary sequence object (without reversing). The
original sequence object is not modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='forward_complement identifier id'>forward_complement</span>               <span class='comment val'>#=&gt; 'tacg'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'atgc'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 102</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L102">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">102
103
104
105
106</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='forward_complement identifier id'>forward_complement</span>
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='rparen token'>)</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='forward_complement! fid id'>forward_complement!</span>
  <span class='s identifier id'>s</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>forward_complement!</h3>
</div><div id="forward_complement-21-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>forward_complement!</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Converts the current sequence into its complement (without reversing). The
original sequence object is modified.
</p>
<pre class="code">
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='forward_complement! fid id'>forward_complement!</span>              <span class='comment val'>#=&gt; 'tacg'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'tacg'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>current Bio::Sequence::NA object (modified)

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 116</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L116">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">116
117
118
119
120
121
122
123</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='forward_complement! fid id'>forward_complement!</span>
  <span class='if if kw'>if</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='rna? fid id'>rna?</span>
    <span class='self self kw'>self</span><span class='dot token'>.</span><span class='tr! fid id'>tr!</span><span class='lparen token'>(</span><span class='string val'>'augcrymkdhvbswn'</span><span class='comma token'>,</span> <span class='string val'>'uacgyrkmhdbvswn'</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='self self kw'>self</span><span class='dot token'>.</span><span class='tr! fid id'>tr!</span><span class='lparen token'>(</span><span class='string val'>'atgcrymkdhvbswn'</span><span class='comma token'>,</span> <span class='string val'>'tacgyrkmhdbvswn'</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='self self kw'>self</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>gc_content</h3>
</div><div id="gc_content-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>gc_content</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Calculate the ratio of GC / ATGC bases. U is regarded as T.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='gc_content identifier id'>gc_content</span>                       <span class='comment val'>#=&gt; 0.555555555555556</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Float

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 305</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L305">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">305
306
307
308
309
310
311</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='gc_content identifier id'>gc_content</span>
  <span class='count identifier id'>count</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>
  <span class='at identifier id'>at</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'a'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'t'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'u'</span><span class='rbrack token'>]</span>
  <span class='gc identifier id'>gc</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'g'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'c'</span><span class='rbrack token'>]</span>
  <span class='return return kw'>return</span> <span class='integer val'>0</span><span class='integer val'>.0</span> <span class='if if_mod kw'>if</span> <span class='at identifier id'>at</span> <span class='plus op'>+</span> <span class='gc identifier id'>gc</span> <span class='eq op'>==</span> <span class='integer val'>0</span>
  <span class='return return kw'>return</span> <span class='gc identifier id'>gc</span><span class='dot token'>.</span><span class='quo identifier id'>quo</span><span class='lparen token'>(</span><span class='at identifier id'>at</span> <span class='plus op'>+</span> <span class='gc identifier id'>gc</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>gc_percent</h3>
</div><div id="gc_percent-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>gc_percent</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Calculate the ratio of GC / ATGC bases as a percentage rounded to the
nearest whole number. U is regarded as T.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='gc_percent identifier id'>gc_percent</span>                       <span class='comment val'>#=&gt; 55</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Fixnum

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 290</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L290">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">290
291
292
293
294
295
296
297</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='gc_percent identifier id'>gc_percent</span>
  <span class='count identifier id'>count</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>
  <span class='at identifier id'>at</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'a'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'t'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'u'</span><span class='rbrack token'>]</span>
  <span class='gc identifier id'>gc</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'g'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'c'</span><span class='rbrack token'>]</span>
  <span class='return return kw'>return</span> <span class='integer val'>0</span> <span class='if if_mod kw'>if</span> <span class='at identifier id'>at</span> <span class='plus op'>+</span> <span class='gc identifier id'>gc</span> <span class='eq op'>==</span> <span class='integer val'>0</span>
  <span class='gc identifier id'>gc</span> <span class='assign token'>=</span> <span class='integer val'>100</span> <span class='mult op'>*</span> <span class='gc identifier id'>gc</span> <span class='div op'>/</span> <span class='lparen token'>(</span><span class='at identifier id'>at</span> <span class='plus op'>+</span> <span class='gc identifier id'>gc</span><span class='rparen token'>)</span>
  <span class='return return kw'>return</span> <span class='gc identifier id'>gc</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>gc_skew</h3>
</div><div id="gc_skew-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>gc_skew</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Calculate the ratio of (G - C) / (G + C) bases.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='gc_skew identifier id'>gc_skew</span>                          <span class='comment val'>#=&gt; 0.6</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Float

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 333</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L333">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">333
334
335
336
337
338
339</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='gc_skew identifier id'>gc_skew</span>
  <span class='count identifier id'>count</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>
  <span class='g identifier id'>g</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'g'</span><span class='rbrack token'>]</span>
  <span class='c identifier id'>c</span> <span class='assign token'>=</span> <span class='count identifier id'>count</span><span class='lbrack token'>[</span><span class='string val'>'c'</span><span class='rbrack token'>]</span>
  <span class='return return kw'>return</span> <span class='integer val'>0</span><span class='integer val'>.0</span> <span class='if if_mod kw'>if</span> <span class='g identifier id'>g</span> <span class='plus op'>+</span> <span class='c identifier id'>c</span> <span class='eq op'>==</span> <span class='integer val'>0</span>
  <span class='return return kw'>return</span> <span class='lparen token'>(</span><span class='g identifier id'>g</span> <span class='minus op'>-</span> <span class='c identifier id'>c</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='quo identifier id'>quo</span><span class='lparen token'>(</span><span class='g identifier id'>g</span> <span class='plus op'>+</span> <span class='c identifier id'>c</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>illegal_bases</h3>
</div><div id="illegal_bases-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>illegal_bases</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns an alphabetically sorted array of any non-standard bases (other
than &#8216;atgcu&#8217;).
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgStgQccR'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='illegal_bases identifier id'>illegal_bases</span>                    <span class='comment val'>#=&gt; [&quot;q&quot;, &quot;r&quot;, &quot;s&quot;]</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Array object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 362</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L362">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">362
363
364</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='illegal_bases identifier id'>illegal_bases</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='scan identifier id'>scan</span><span class='lparen token'>(</span><span class='regexp val'>/[^atgcu]/</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='sort identifier id'>sort</span><span class='dot token'>.</span><span class='uniq identifier id'>uniq</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>molecular_weight</h3>
</div><div id="molecular_weight-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>molecular_weight</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Estimate molecular weight (using the values from BioPerl&#8217;s
SeqStats.pm[http://doc.bioperl.org/releases/bioperl-1.0.1/Bio/Tools/SeqStats.html]
module).
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='molecular_weight identifier id'>molecular_weight</span>                 <span class='comment val'>#=&gt; 2841.00708</span>
</pre>
<p>
RNA and DNA do not have the same molecular weights,
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'auggcguga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='molecular_weight identifier id'>molecular_weight</span>                 <span class='comment val'>#=&gt; 2956.94708</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Float object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 378</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L378">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">378
379
380
381
382
383
384</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='molecular_weight identifier id'>molecular_weight</span>
  <span class='if if kw'>if</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='rna? fid id'>rna?</span>
    <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='NucleicAcid constant id'>NucleicAcid</span><span class='dot token'>.</span><span class='weight identifier id'>weight</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='comma token'>,</span> <span class='true true kw'>true</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='NucleicAcid constant id'>NucleicAcid</span><span class='dot token'>.</span><span class='weight identifier id'>weight</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>names</h3>
</div><div id="names-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>names</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Generate the list of the names of each nucleotide along with the sequence
(full name). Names used in bioruby are found in the Bio::AminoAcid::NAMES
hash.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atg'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='names identifier id'>names</span>                    <span class='comment val'>#=&gt; [&quot;Adenine&quot;, &quot;Thymine&quot;, &quot;Guanine&quot;]</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Array object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 409</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L409">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">409
410
411
412
413
414
415</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='names identifier id'>names</span>
  <span class='array identifier id'>array</span> <span class='assign token'>=</span> <span class='lbrack token'>[</span><span class='rbrack token'>]</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='each_byte identifier id'>each_byte</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='x identifier id'>x</span><span class='bitor op'>|</span>
    <span class='array identifier id'>array</span><span class='dot token'>.</span><span class='push identifier id'>push</span><span class='lparen token'>(</span><span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='NucleicAcid constant id'>NucleicAcid</span><span class='dot token'>.</span><span class='names identifier id'>names</span><span class='lbrack token'>[</span><span class='x identifier id'>x</span><span class='dot token'>.</span><span class='chr identifier id'>chr</span><span class='dot token'>.</span><span class='upcase identifier id'>upcase</span><span class='rbrack token'>]</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='return return kw'>return</span> <span class='array identifier id'>array</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>pikachu</h3>
</div><div id="pikachu-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>pikachu</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/compat.rb', line 87</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/compat.rb#L87">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">87
88
89</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='pikachu identifier id'>pikachu</span> 
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='tr identifier id'>tr</span><span class='lparen token'>(</span><span class='string val'>&quot;atgc&quot;</span><span class='comma token'>,</span> <span class='string val'>&quot;pika&quot;</span><span class='rparen token'>)</span> <span class='comment val'># joke, of course :-)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>reverse_complement</h3>
</div><div id="reverse_complement-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>reverse_complement</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>
<p class="aliases">
  <span class="aka">Also known as:</span>
  
    <tt id="complement-instance_method">complement</tt>
  
</p>
</div><div class="section docstring">
  <p>
Returns a new sequence object with the reverse complement sequence to the
original. The original sequence is not modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='reverse_complement identifier id'>reverse_complement</span>               <span class='comment val'>#=&gt; 'gcat'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'atgc'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 133</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L133">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">133
134
135
136
137</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='reverse_complement identifier id'>reverse_complement</span>
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='rparen token'>)</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='reverse_complement! fid id'>reverse_complement!</span>
  <span class='s identifier id'>s</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>reverse_complement!</h3>
</div><div id="reverse_complement-21-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>reverse_complement!</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>
<p class="aliases">
  <span class="aka">Also known as:</span>
  
    <tt id="complement!-instance_method">complement!</tt>
  
</p>
</div><div class="section docstring">
  <p>
Converts the original sequence into its reverse complement. The original
sequence is modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='reverse_complement identifier id'>reverse_complement</span>               <span class='comment val'>#=&gt; 'gcat'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'gcat'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>current Bio::Sequence::NA object (modified)

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 147</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L147">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">147
148
149
150</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='reverse_complement! fid id'>reverse_complement!</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='reverse! fid id'>reverse!</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='forward_complement! fid id'>forward_complement!</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>rna</h3>
</div><div id="rna-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>rna</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns a new sequence object with any &#8216;t&#8217; bases changed to
&#8216;u&#8217;. The original sequence is not modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='dna identifier id'>dna</span>                              <span class='comment val'>#=&gt; 'augc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'atgc'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>new Bio::Sequence::NA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 449</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L449">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">449
450
451</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='rna identifier id'>rna</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='tr identifier id'>tr</span><span class='lparen token'>(</span><span class='string val'>'t'</span><span class='comma token'>,</span> <span class='string val'>'u'</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>rna!</h3>
</div><div id="rna-21-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>rna!</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Changes any &#8216;t&#8217; bases in the original sequence to
&#8216;u&#8217;. The original sequence is modified.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='dna! fid id'>dna!</span>                             <span class='comment val'>#=&gt; 'augc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'augc'</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>current Bio::Sequence::NA object (modified)

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 461</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L461">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">461
462
463</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='rna! fid id'>rna!</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='tr! fid id'>tr!</span><span class='lparen token'>(</span><span class='string val'>'t'</span><span class='comma token'>,</span> <span class='string val'>'u'</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>splicing</h3>
</div><div id="splicing-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>splicing</span><span class='args'>(position)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Alias of Bio::Sequence::Common splice method, documented there.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 84</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L84">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">84
85
86
87
88
89
90
91
92</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='splicing identifier id'>splicing</span><span class='lparen token'>(</span><span class='position identifier id'>position</span><span class='rparen token'>)</span> <span class='comment val'>#:nodoc:</span>
  <span class='mRNA identifier id'>mRNA</span> <span class='assign token'>=</span> <span class='super super kw'>super</span>
  <span class='if if kw'>if</span> <span class='mRNA identifier id'>mRNA</span><span class='dot token'>.</span><span class='rna? fid id'>rna?</span>
    <span class='mRNA identifier id'>mRNA</span><span class='dot token'>.</span><span class='tr! fid id'>tr!</span><span class='lparen token'>(</span><span class='string val'>'t'</span><span class='comma token'>,</span> <span class='string val'>'u'</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='mRNA identifier id'>mRNA</span><span class='dot token'>.</span><span class='tr! fid id'>tr!</span><span class='lparen token'>(</span><span class='string val'>'u'</span><span class='comma token'>,</span> <span class='string val'>'t'</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='mRNA identifier id'>mRNA</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>to_midi</h3>
</div><div id="to_midi-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>to_midi</span><span class='args'>(style = {}, drum = true)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
style:
</p>
<pre class="code">
  <span class='Hash constant id'>Hash</span> <span class='of identifier id'>of</span> <span class='symbol val'>:tempo</span><span class='comma token'>,</span> <span class='symbol val'>:scale</span><span class='comma token'>,</span> <span class='symbol val'>:tones</span>
</pre>
<p>
scale:
</p>
<pre class="code">
  <span class='C constant id'>C</span>  <span class='C constant id'>C</span><span class='comment val'># D  D# E  F  F# G  G# A  A#  B</span>
  <span class='integer val'>0</span>  <span class='integer val'>1</span>  <span class='integer val'>2</span>  <span class='integer val'>3</span>  <span class='integer val'>4</span>  <span class='integer val'>5</span>  <span class='integer val'>6</span>  <span class='integer val'>7</span>  <span class='integer val'>8</span>  <span class='integer val'>9</span>  <span class='integer val'>10</span>  <span class='integer val'>11</span>
</pre>
<p>
tones:
</p>
<pre class="code">
  <span class='Hash constant id'>Hash</span> <span class='of identifier id'>of</span> <span class='symbol val'>:prog</span><span class='comma token'>,</span> <span class='symbol val'>:base</span><span class='comma token'>,</span> <span class='symbol val'>:range</span> <span class='minus op'>-</span><span class='minus op'>-</span> <span class='tone identifier id'>tone</span><span class='comma token'>,</span> <span class='vol? fid id'>vol?</span> <span class='or or kw'>or</span> <span class='len? fid id'>len?</span><span class='comma token'>,</span> <span class='octaves identifier id'>octaves</span>
</pre>
<p>
drum:
</p>
<pre class="code">
  <span class='true true kw'>true</span> <span class='lparen token'>(</span><span class='with identifier id'>with</span> <span class='rhythm identifier id'>rhythm</span> <span class='part identifier id'>part</span><span class='rparen token'>)</span><span class='comma token'>,</span> <span class='false false kw'>false</span> <span class='lparen token'>(</span><span class='without identifier id'>without</span> <span class='rhythm identifier id'>rhythm</span> <span class='part identifier id'>part</span><span class='rparen token'>)</span>
</pre>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/shell/plugin/midi.rb', line 351</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/shell/plugin/midi.rb#L351">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">351
352
353
354
355
356
357
358
359
360
361
362
363
364
365
366
367
368
369
370
371
372
373
374
375
376
377
378
379
380
381
382
383
384
385
386
387
388</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='to_midi identifier id'>to_midi</span><span class='lparen token'>(</span><span class='style identifier id'>style</span> <span class='assign token'>=</span> <span class='lbrace token'>{</span><span class='rbrace token'>}</span><span class='comma token'>,</span> <span class='drum identifier id'>drum</span> <span class='assign token'>=</span> <span class='true true kw'>true</span><span class='rparen token'>)</span>
  <span class='default identifier id'>default</span> <span class='assign token'>=</span> <span class='MidiTrack constant id'>MidiTrack</span><span class='colon2 op'>::</span><span class='Styles constant id'>Styles</span><span class='lbrack token'>[</span><span class='string val'>&quot;Ichinose&quot;</span><span class='rbrack token'>]</span>
  <span class='if if kw'>if</span> <span class='style identifier id'>style</span><span class='dot token'>.</span><span class='is_a? fid id'>is_a?</span><span class='lparen token'>(</span><span class='String constant id'>String</span><span class='rparen token'>)</span>
    <span class='style identifier id'>style</span> <span class='assign token'>=</span> <span class='MidiTrack constant id'>MidiTrack</span><span class='colon2 op'>::</span><span class='Styles constant id'>Styles</span><span class='lbrack token'>[</span><span class='style identifier id'>style</span><span class='rbrack token'>]</span> <span class='orop op'>||</span> <span class='default identifier id'>default</span>
  <span class='end end kw'>end</span>
  <span class='tempo identifier id'>tempo</span> <span class='assign token'>=</span> <span class='style identifier id'>style</span><span class='lbrack token'>[</span><span class='symbol val'>:tempo</span><span class='rbrack token'>]</span> <span class='orop op'>||</span> <span class='default identifier id'>default</span><span class='lbrack token'>[</span><span class='symbol val'>:tempo</span><span class='rbrack token'>]</span>
  <span class='scale identifier id'>scale</span> <span class='assign token'>=</span> <span class='style identifier id'>style</span><span class='lbrack token'>[</span><span class='symbol val'>:scale</span><span class='rbrack token'>]</span> <span class='orop op'>||</span> <span class='default identifier id'>default</span><span class='lbrack token'>[</span><span class='symbol val'>:scale</span><span class='rbrack token'>]</span>
  <span class='tones identifier id'>tones</span> <span class='assign token'>=</span> <span class='style identifier id'>style</span><span class='lbrack token'>[</span><span class='symbol val'>:tones</span><span class='rbrack token'>]</span> <span class='orop op'>||</span> <span class='default identifier id'>default</span><span class='lbrack token'>[</span><span class='symbol val'>:tones</span><span class='rbrack token'>]</span>

  <span class='track identifier id'>track</span> <span class='assign token'>=</span> <span class='lbrack token'>[</span><span class='rbrack token'>]</span>

  <span class='tones identifier id'>tones</span><span class='dot token'>.</span><span class='each_with_index identifier id'>each_with_index</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='tone identifier id'>tone</span><span class='comma token'>,</span> <span class='i identifier id'>i</span><span class='bitor op'>|</span>
    <span class='ch identifier id'>ch</span> <span class='assign token'>=</span> <span class='i identifier id'>i</span>
    <span class='ch identifier id'>ch</span> <span class='opasgn op'>+=</span> <span class='integer val'>1</span> <span class='if if_mod kw'>if</span> <span class='i identifier id'>i</span> <span class='geq op'>&gt;=</span> <span class='integer val'>9</span>         <span class='comment val'># skip rythm track</span>
    <span class='track identifier id'>track</span><span class='dot token'>.</span><span class='push identifier id'>push</span> <span class='MidiTrack constant id'>MidiTrack</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='ch identifier id'>ch</span><span class='comma token'>,</span> <span class='tone identifier id'>tone</span><span class='lbrack token'>[</span><span class='symbol val'>:prog</span><span class='rbrack token'>]</span><span class='comma token'>,</span> <span class='tone identifier id'>tone</span><span class='lbrack token'>[</span><span class='symbol val'>:base</span><span class='rbrack token'>]</span><span class='comma token'>,</span> <span class='tone identifier id'>tone</span><span class='lbrack token'>[</span><span class='symbol val'>:range</span><span class='rbrack token'>]</span><span class='comma token'>,</span> <span class='scale identifier id'>scale</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>

  <span class='if if kw'>if</span> <span class='drum identifier id'>drum</span>
    <span class='rhythm identifier id'>rhythm</span> <span class='assign token'>=</span> <span class='lbrack token'>[</span><span class='integer val'>0</span><span class='comma token'>,</span> <span class='integer val'>1</span><span class='comma token'>,</span> <span class='integer val'>2</span><span class='comma token'>,</span> <span class='integer val'>3</span><span class='comma token'>,</span> <span class='integer val'>4</span><span class='comma token'>,</span> <span class='integer val'>5</span><span class='comma token'>,</span> <span class='integer val'>6</span><span class='comma token'>,</span> <span class='integer val'>7</span><span class='comma token'>,</span> <span class='integer val'>8</span><span class='comma token'>,</span> <span class='integer val'>9</span><span class='comma token'>,</span> <span class='integer val'>10</span><span class='comma token'>,</span> <span class='integer val'>11</span><span class='rbrack token'>]</span>
    <span class='track identifier id'>track</span><span class='dot token'>.</span><span class='push identifier id'>push</span><span class='lparen token'>(</span><span class='MidiTrack constant id'>MidiTrack</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='integer val'>9</span><span class='comma token'>,</span> <span class='integer val'>0</span><span class='comma token'>,</span> <span class='integer val'>35</span><span class='comma token'>,</span> <span class='integer val'>2</span><span class='comma token'>,</span> <span class='rhythm identifier id'>rhythm</span><span class='rparen token'>)</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>

  <span class='cur identifier id'>cur</span> <span class='assign token'>=</span> <span class='integer val'>0</span>
  <span class='window_search identifier id'>window_search</span><span class='lparen token'>(</span><span class='integer val'>4</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='s identifier id'>s</span><span class='bitor op'>|</span>
    <span class='track identifier id'>track</span><span class='lbrack token'>[</span><span class='cur identifier id'>cur</span> <span class='mod op'>%</span> <span class='track identifier id'>track</span><span class='dot token'>.</span><span class='length identifier id'>length</span><span class='rbrack token'>]</span><span class='dot token'>.</span><span class='push identifier id'>push</span><span class='lparen token'>(</span><span class='s identifier id'>s</span><span class='rparen token'>)</span>
    <span class='cur identifier id'>cur</span> <span class='opasgn op'>+=</span> <span class='integer val'>1</span>
  <span class='end end kw'>end</span>

  <span class='track identifier id'>track</span><span class='dot token'>.</span><span class='each identifier id'>each</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='t identifier id'>t</span><span class='bitor op'>|</span>
    <span class='t identifier id'>t</span><span class='dot token'>.</span><span class='push_silent identifier id'>push_silent</span><span class='lparen token'>(</span><span class='integer val'>12</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>

  <span class='ans identifier id'>ans</span> <span class='assign token'>=</span> <span class='track identifier id'>track</span><span class='lbrack token'>[</span><span class='integer val'>0</span><span class='rbrack token'>]</span><span class='dot token'>.</span><span class='header identifier id'>header</span><span class='lparen token'>(</span><span class='track identifier id'>track</span><span class='dot token'>.</span><span class='length identifier id'>length</span><span class='comma token'>,</span> <span class='tempo identifier id'>tempo</span><span class='rparen token'>)</span>
  <span class='track identifier id'>track</span><span class='dot token'>.</span><span class='each identifier id'>each</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='t identifier id'>t</span><span class='bitor op'>|</span>
    <span class='ans identifier id'>ans</span> <span class='opasgn op'>+=</span> <span class='t identifier id'>t</span><span class='dot token'>.</span><span class='encode identifier id'>encode</span>
  <span class='end end kw'>end</span>
  <span class='return return kw'>return</span> <span class='ans identifier id'>ans</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>to_re</h3>
</div><div id="to_re-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>to_re</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Create a ruby regular expression instance
(Regexp)[http://corelib.rubyonrails.org/classes/Regexp.html]
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='to_re identifier id'>to_re</span>                            <span class='comment val'>#=&gt; /atggcgtga/</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Regexp object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 393</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L393">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">393
394
395
396
397
398
399</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='to_re identifier id'>to_re</span>
  <span class='if if kw'>if</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='rna? fid id'>rna?</span>
    <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='NucleicAcid constant id'>NucleicAcid</span><span class='dot token'>.</span><span class='to_re identifier id'>to_re</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='dot token'>.</span><span class='dna identifier id'>dna</span><span class='comma token'>,</span> <span class='true true kw'>true</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='NucleicAcid constant id'>NucleicAcid</span><span class='dot token'>.</span><span class='to_re identifier id'>to_re</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>translate</h3>
</div><div id="translate-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>translate</span><span class='args'>(frame = 1, table = 1, unknown = 'X')</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Translate into an amino acid sequence.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atggcgtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span>                        <span class='comment val'>#=&gt; &quot;MA*&quot;</span>
</pre>
<p>
By default, translate starts in reading frame position 1, but you can start
in either 2 or 3 as well,
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>2</span><span class='rparen token'>)</span>                     <span class='comment val'>#=&gt; &quot;WR&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>3</span><span class='rparen token'>)</span>                     <span class='comment val'>#=&gt; &quot;GV&quot;</span>
</pre>
<p>
You may also translate the reverse complement in one step by using frame
values of -1, -2, and -3 (or 4, 5, and 6)
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>-1</span><span class='rparen token'>)</span>                    <span class='comment val'>#=&gt; &quot;SRH&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>4</span><span class='rparen token'>)</span>                     <span class='comment val'>#=&gt; &quot;SRH&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='reverse_complement identifier id'>reverse_complement</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>1</span><span class='rparen token'>)</span>  <span class='comment val'>#=&gt; &quot;SRH&quot;</span>
</pre>
<p>
The default codon table in the translate function is the Standard
Eukaryotic codon table. The translate function takes either a number or a
Bio::CodonTable object for its table argument. The available tables are
(NCBI[http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=t]):
</p>
<pre class="code">
  <span class='float val'>1</span><span class='dot token'>.</span> <span class='string val'>&quot;Standard (Eukaryote)&quot;</span>
  <span class='float val'>2</span><span class='dot token'>.</span> <span class='string val'>&quot;Vertebrate Mitochondrial&quot;</span>
  <span class='float val'>3</span><span class='dot token'>.</span> <span class='string val'>&quot;Yeast Mitochondorial&quot;</span>
  <span class='float val'>4</span><span class='dot token'>.</span> <span class='string val'>&quot;Mold, Protozoan, Coelenterate Mitochondrial and Mycoplasma/Spiroplasma&quot;</span>
  <span class='float val'>5</span><span class='dot token'>.</span> <span class='string val'>&quot;Invertebrate Mitochondrial&quot;</span>
  <span class='float val'>6</span><span class='dot token'>.</span> <span class='string val'>&quot;Ciliate Macronuclear and Dasycladacean&quot;</span>
  <span class='float val'>9</span><span class='dot token'>.</span> <span class='string val'>&quot;Echinoderm Mitochondrial&quot;</span>
  <span class='float val'>10</span><span class='dot token'>.</span> <span class='string val'>&quot;Euplotid Nuclear&quot;</span>
  <span class='float val'>11</span><span class='dot token'>.</span> <span class='string val'>&quot;Bacteria&quot;</span>
  <span class='float val'>12</span><span class='dot token'>.</span> <span class='string val'>&quot;Alternative Yeast Nuclear&quot;</span>
  <span class='float val'>13</span><span class='dot token'>.</span> <span class='string val'>&quot;Ascidian Mitochondrial&quot;</span>
  <span class='float val'>14</span><span class='dot token'>.</span> <span class='string val'>&quot;Flatworm Mitochondrial&quot;</span>
  <span class='float val'>15</span><span class='dot token'>.</span> <span class='string val'>&quot;Blepharisma Macronuclear&quot;</span>
  <span class='float val'>16</span><span class='dot token'>.</span> <span class='string val'>&quot;Chlorophycean Mitochondrial&quot;</span>
  <span class='float val'>21</span><span class='dot token'>.</span> <span class='string val'>&quot;Trematode Mitochondrial&quot;</span>
  <span class='float val'>22</span><span class='dot token'>.</span> <span class='string val'>&quot;Scenedesmus obliquus mitochondrial&quot;</span>
  <span class='float val'>23</span><span class='dot token'>.</span> <span class='string val'>&quot;Thraustochytrium Mitochondrial&quot;</span>
</pre>
<p>
If you are using anything other than the default table, you must specify
frame in the translate method call,
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span>                <span class='comment val'>#=&gt; &quot;MA*&quot;  (using defaults)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>1</span><span class='comma token'>,</span><span class='integer val'>1</span><span class='rparen token'>)</span>           <span class='comment val'>#=&gt; &quot;MA*&quot;  (same as above, but explicit)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>1</span><span class='comma token'>,</span><span class='integer val'>2</span><span class='rparen token'>)</span>           <span class='comment val'>#=&gt; &quot;MAW&quot;  (different codon table)</span>
</pre>
<p>
and using a Bio::CodonTable instance in the translate method call,
</p>
<pre class="code">
  <span class='mt_table identifier id'>mt_table</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='CodonTable constant id'>CodonTable</span><span class='lbrack token'>[</span><span class='integer val'>2</span><span class='rbrack token'>]</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>1</span><span class='comma token'>,</span> <span class='mt_table identifier id'>mt_table</span><span class='rparen token'>)</span>           <span class='comment val'>#=&gt; &quot;MAW&quot;</span>
</pre>
<p>
By default, any invalid or unknown codons (as could happen if the sequence
contains ambiguities) will be represented by &#8216;X&#8217; in the
translated sequence. You may change this to any character of your choice.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgcNNtga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span>                        <span class='comment val'>#=&gt; &quot;MX*&quot;</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='integer val'>1</span><span class='comma token'>,</span><span class='integer val'>1</span><span class='comma token'>,</span><span class='string val'>'9'</span><span class='rparen token'>)</span>               <span class='comment val'>#=&gt; &quot;M9*&quot;</span>
</pre>
<p>
The translate method considers gaps to be unknown characters and treats
them as such (i.e. does not collapse sequences prior to translation), so
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc--tga'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span>                        <span class='comment val'>#=&gt; &quot;MX*&quot;</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(optional) <em>frame</em>: one of 1,2,3,4,5,6,-1,-2,-3 (default 1)

</li>
<li>(optional) <em>table</em>: Fixnum in range 1,23 or Bio::CodonTable object
(default 1)

</li>
<li>(optional) <em>unknown</em>: Character (default &#8216;X&#8217;)

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::AA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/na.rb', line 234</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/na.rb#L234">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='translate identifier id'>translate</span><span class='lparen token'>(</span><span class='frame identifier id'>frame</span> <span class='assign token'>=</span> <span class='integer val'>1</span><span class='comma token'>,</span> <span class='table identifier id'>table</span> <span class='assign token'>=</span> <span class='integer val'>1</span><span class='comma token'>,</span> <span class='unknown identifier id'>unknown</span> <span class='assign token'>=</span> <span class='string val'>'X'</span><span class='rparen token'>)</span>
  <span class='if if kw'>if</span> <span class='table identifier id'>table</span><span class='dot token'>.</span><span class='is_a? fid id'>is_a?</span><span class='lparen token'>(</span><span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='CodonTable constant id'>CodonTable</span><span class='rparen token'>)</span>
    <span class='ct identifier id'>ct</span> <span class='assign token'>=</span> <span class='table identifier id'>table</span>
  <span class='else else kw'>else</span>
    <span class='ct identifier id'>ct</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='CodonTable constant id'>CodonTable</span><span class='lbrack token'>[</span><span class='table identifier id'>table</span><span class='rbrack token'>]</span>
  <span class='end end kw'>end</span>
  <span class='naseq identifier id'>naseq</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='dna identifier id'>dna</span>
  <span class='case case kw'>case</span> <span class='frame identifier id'>frame</span>
  <span class='when when kw'>when</span> <span class='integer val'>1</span><span class='comma token'>,</span> <span class='integer val'>2</span><span class='comma token'>,</span> <span class='integer val'>3</span>
    <span class='from identifier id'>from</span> <span class='assign token'>=</span> <span class='frame identifier id'>frame</span> <span class='minus op'>-</span> <span class='integer val'>1</span>
  <span class='when when kw'>when</span> <span class='integer val'>4</span><span class='comma token'>,</span> <span class='integer val'>5</span><span class='comma token'>,</span> <span class='integer val'>6</span>
    <span class='from identifier id'>from</span> <span class='assign token'>=</span> <span class='frame identifier id'>frame</span> <span class='minus op'>-</span> <span class='integer val'>4</span>
    <span class='naseq identifier id'>naseq</span><span class='dot token'>.</span><span class='complement! fid id'>complement!</span>
  <span class='when when kw'>when</span> <span class='integer val'>-1</span><span class='comma token'>,</span> <span class='integer val'>-2</span><span class='comma token'>,</span> <span class='integer val'>-3</span>
    <span class='from identifier id'>from</span> <span class='assign token'>=</span> <span class='integer val'>-1</span> <span class='minus op'>-</span> <span class='frame identifier id'>frame</span>
    <span class='naseq identifier id'>naseq</span><span class='dot token'>.</span><span class='complement! fid id'>complement!</span>
  <span class='else else kw'>else</span>
    <span class='from identifier id'>from</span> <span class='assign token'>=</span> <span class='integer val'>0</span>
  <span class='end end kw'>end</span>
  <span class='nalen identifier id'>nalen</span> <span class='assign token'>=</span> <span class='naseq identifier id'>naseq</span><span class='dot token'>.</span><span class='length identifier id'>length</span> <span class='minus op'>-</span> <span class='from identifier id'>from</span>
  <span class='nalen identifier id'>nalen</span> <span class='opasgn op'>-=</span> <span class='nalen identifier id'>nalen</span> <span class='mod op'>%</span> <span class='integer val'>3</span>
  <span class='aaseq identifier id'>aaseq</span> <span class='assign token'>=</span> <span class='naseq identifier id'>naseq</span><span class='lbrack token'>[</span><span class='from identifier id'>from</span><span class='comma token'>,</span> <span class='nalen identifier id'>nalen</span><span class='rbrack token'>]</span><span class='dot token'>.</span><span class='gsub identifier id'>gsub</span><span class='lparen token'>(</span><span class='regexp val'>/.{3}/</span><span class='rparen token'>)</span> <span class='lbrace token'>{</span><span class='bitor op'>|</span><span class='codon identifier id'>codon</span><span class='bitor op'>|</span> <span class='ct identifier id'>ct</span><span class='lbrack token'>[</span><span class='codon identifier id'>codon</span><span class='rbrack token'>]</span> <span class='or or kw'>or</span> <span class='unknown identifier id'>unknown</span><span class='rbrace token'>}</span>
  <span class='return return kw'>return</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='AA constant id'>AA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='aaseq identifier id'>aaseq</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div>
  </div>

</div>
</div>
          </div>
        </div>
      </div>                    
    </div>

    <div id="footer">
      <div class="site">
        <div class="info">
          <div class="links">
            <a href="http://github.com/blog/148-github-shirts-now-available">Shirts</a> |
            <a href="http://github.com/blog">Blog</a> |
            <a href="http://support.github.com/">Support</a> |
            <a href="http://github.com/training">Training</a> |
            <a href="http://github.com/contact">Contact</a> |
            <a href="http://groups.google.com/group/github/">Google Group</a> |
            <a href="http://develop.github.com/">API</a> |
            <a href="http://twitter.com/github">Status</a>
          </div>
          <div class="company">
            <span id="_rrt" title="0.64509s from xc88-s00039">GitHub</span>
            is <a href="http://logicalawesome.com/">Logical Awesome</a> ©2009 | <a href="http://github.com/site/terms">Terms of Service</a> | <a href="http://github.com/site/privacy">Privacy Policy</a>
          </div>
        </div>
        <div class="sponsor">
          <a href="http://engineyard.com/"><img src="http://github.com/images/modules/footer/engine_yard_logo.png" alt="Engine Yard"></a>
          <div>
            Hosting provided by our<br> partners at Engine Yard
          </div>
        </div>
      </div>
    </div>
    
    <div id="coming_soon" style="display: none;">
      This feature is coming soon.  Sit tight!
    </div>

    <div id="facebox" style="display: none;">       
      <div class="popup">         
        <table>           
          <tbody>             
            <tr>               
              <td class="tl"></td>
              <td class="b"></td>
              <td class="tr"></td>             
            </tr>             
          <tr>               
            <td class="b"></td>               
            <td class="body">                 
              <div class="content"></div>                 
              <div class="footer">                   
                <a href="#" class="close"><img src="http://github.com/facebox/closelabel.gif" title="close" class="close_image"></a>                 
              </div>               
            </td>               
            <td class="b"></td>             
          </tr>             
          <tr>               
            <td class="bl"></td>
            <td class="b"></td>
            <td class="br"></td>             
          </tr>           
        </tbody>         
      </table>       
    </div>     


    <script type="text/javascript">
    var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
    document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
    </script>
    <script type="text/javascript">
    try {
    var pageTracker = _gat._getTracker("UA-8689483-1");
    pageTracker._trackPageview();
    } catch(err) {}</script>

  </body>
</html>