<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xml:lang="en" xmlns="http://www.w3.org/1999/xhtml" lang="en">
  <head>
    <title>bioruby's bioruby - GitHub Documentation</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta name="description" content="">

<link rel="search" type="application/opensearchdescription+xml" href="http://github.com/opensearch.xml" title="GitHub">
<link rel="fluid-icon" href="http://github.com/fluidicon.png" title="GitHub">
<link href="http://assets0.github.com/stylesheets/bundle.css" media="screen" rel="stylesheet" type="text/css">
<link href="http://github.com/feeds/bioruby/commits/bioruby/master" rel="alternate" title="Recent Commits to bioruby:master" type="application/atom+xml">

<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js"></script>
<script type="text/javascript" src="http://assets3.github.com/javascripts/bundle.js"></script>              
<script src="app.js" type="text/javascript" charset="utf-8"></script>

<script type="text/javascript" src="http://docs.github.com/javascripts/md5.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/repos.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/commits.js"></script>
<script type="text/javascript">
  $(document).ready(function(){
    if (typeof skipHeaders != 'undefined') return;
    
    GitHubRepo.init("bioruby", "bioruby", function(){ 
      GitHubRepo.links = " <a id='namespaces_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/namespaces_button.png' alt='Modules and Classes'/></a> ";
      GitHubRepo.links += " <a id='methods_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/methods_button.png' alt='Methods'/></a> ";
      $("#repo").html(GitHubRepo.content());
      $('#namespaces_button').click(function(){
        var url="/"+$(this).attr('rel')+"/namespaces/";
        $.gitbox(url);
        return false;
      });    
      $('#methods_button').click(function(){
        var url="/"+$(this).attr('rel')+"/methods/";
        $.gitbox(url);
        return false;
      });        
    });
    GitHubCommit.init("bioruby", "bioruby", "97b9284109c9a4431b92eab208509e1df6069b4b", function(){ 
      $("#commit").html(GitHubCommit.content());
    });
  });
</script>

    <link rel="stylesheet" href="style.css" type="text/css" charset="utf-8" />
  </head>
  <body>    
    
    
        <div id="main" style="padding-bottom:8em;">
      <div id="header">
        <div class="site">
          <div class="logo">
            <a href="http://github.com/"><img src="http://github.com/images/modules/header/logov3.png" alt="github"></a>
          </div>          
          <div class="topsearch">
            <form action="http://github.com/search" id="top_search_form" method="get">
              <input class="search" name="q" type="search"> <input value="Search" type="submit">
            </form>
            <div class="links">
              <a href="http://github.com/repositories">Browse</a> | <a href="http://github.com/guides">Guides</a> | <a href="http://github.com/search">Advanced</a>
            </div>
          </div>
          <div class="actions">
            <a href="http://github.com">Home</a>
            <a href="http://github.com/plans"><b><u>Pricing and Signup</u></b></a>
            <a href="http://docs.github.com/">Documentation</a>
            <a href="https://github.com/login">Login</a>
          </div>                  
        </div>
      </div>    
                                                  
      <div id="repo_menu">
        <div class="site">
          <ul>          
            <li class=""><a href="http://github.com/bioruby/bioruby/tree/">Source</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/commits/">Commits</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/network">Network</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/issues">Issues</a></li>
            <li class=""><a href="http://wiki.github.com/bioruby/bioruby">Wiki</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/graphs">Graphs</a></li>                    
            <li class="active"><a href="/bioruby/bioruby">Documentation</a></li>
          </ul>
        </div>
      </div>

      <div id="repo_sub_menu">
        <div class="site">
          <div class="joiner"></div>      
        </div>
      </div>

      <div class="site">  
        <div id="repos">
          <div id="repo" class="repo public">
            <div class="title">
              <div class="path">
                <a href="http://github.com/bioruby">bioruby</a> / <b><a href="http://github.com/bioruby/bioruby/tree">bioruby</a></b>        
                <a id="namespaces_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/namespaces_button.png" alt="Modules and Classes"/></a>
                <a id="methods_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/methods_button.png" alt="Methods"/></a>
              </div>
            </div>            
          </div>
          <div id="commit">
          </div> 
          <br/>                                       
          <div id="content">
            <div class="section docstring readme wikistyle">
  <h1>README.DEV</h1>
<table>
<tr><td valign="top">Copyright:</td><td>Copyright (C) 2005, 2006 Toshiaki Katayama &lt;k@bioruby.org&gt;

</td></tr>
<tr><td valign="top">Copyright:</td><td>Copyright (C) 2006, 2008 Jan Aerts &lt;jandot@bioruby.org&gt;

</td></tr>
</table>
<h1>HOW TO CONTRIBUTE TO THE BIORUBY PROJECT?</h1>
<p>
There are many possible ways to contribute to the BioRuby project, such as:
</p>
<ul>
<li>Join the discussion on the BioRuby mailing list

</li>
<li>Send a bug report or write a bug fix patch

</li>
<li>Add and correct documentation

</li>
<li>Develop code for new features, etc.

</li>
</ul>
<p>
All of these are welcome! However, this document describes the last option,
how to contribute your code to the BioRuby distribution.
</p>
<p>
We would like to include your contribution as long as the scope of your
module meets the field of bioinformatics.
</p>
<h2>Git</h2>
<p>
Bioruby is now under git source control at
http://github.com/bioruby/bioruby. There are two basic ways to contribute:
with patches or pull requests. Both are explained on the bioruby wiki at
http://bioruby.open-bio.org/wiki.
</p>
<h1>LICENSE</h1>
<p>
If you would like your module to be included in the BioRuby distribution,
you need to give us right to change the license of your module to make it
compatible with other modules in BioRuby.
</p>
<p>
BioRuby was previously distributed under the LGPL license, but now is
distributed under the same terms as Ruby.
</p>
<h1>CODING STYLE</h1>
<p>
You will need to follow the typical coding styles of the BioRuby modules:
</p>
<h2>Use the following naming conventions</h2>
<ul>
<li>CamelCase for module and class names

</li>
<li>&#8216;_&#8217;-separated_lowercase for method names

</li>
<li>&#8216;_&#8217;-separated_lowercase for variable names

</li>
<li>all UPPERCASE for constants

</li>
</ul>
<h2>Indentation must not include tabs</h2>
<ul>
<li>Use 2 spaces for indentation.

</li>
<li>Don&#8217;t replace spaces to tabs.

</li>
</ul>
<h2>Comments</h2>
<p>
Don&#8217;t use <tt>=begin</tt> and <tt>=end</tt> blocks for comments. If
you need to add comments, include it in the RDoc documentation.
</p>
<h2>Documentation should be written in the RDoc format in the source code</h2>
<p>
The RDoc format is becoming the popular standard for Ruby documentation. We
are now in transition from the previously used RD format to the RDoc format
in API documentation.
</p>
<p>
Additional tutorial documentation and working examples are encouraged with
your contribution. You may use the header part of the file for this purpose
as demonstrated in the previous section.
</p>
<h2>Standard documentation</h2>
<h3>of files</h3>
<p>
Each file should start with a header, which covers the following topics:
</p>
<ul>
<li>copyright

</li>
<li>license

</li>
<li>description of the file (<em>not</em> the classes; see below)

</li>
<li>any references, if appropriate

</li>
</ul>
<p>
The header should be formatted as follows:
</p>
<pre class="code">
  <span class='comment val'>#</span>
  <span class='comment val'># = bio/db/hoge.rb - Hoge database parser classes</span>
  <span class='comment val'>#</span>
  <span class='comment val'># Copyright::  Copyright (C) 2001, 2003-2005 Bio R. Hacker &lt;brh@example.org&gt;,</span>
  <span class='comment val'># Copyright::  Copyright (C) 2006 Chem R. Hacker &lt;crh@example.org&gt;</span>
  <span class='comment val'>#</span>
  <span class='comment val'># License::    The Ruby License</span>
  <span class='comment val'>#</span>
  <span class='comment val'># == Description</span>
  <span class='comment val'>#</span>
  <span class='comment val'># This file contains classes that implement an interface to the Hoge database.</span>
  <span class='comment val'>#</span>
  <span class='comment val'># == References</span>
  <span class='comment val'>#</span>
  <span class='comment val'># * Hoge F. et al., The Hoge database, Nucleic. Acid. Res. 123:100--123 (2030)</span>
  <span class='comment val'># * http://hoge.db/</span>
  <span class='comment val'>#</span>

  <span class='require identifier id'>require</span> <span class='string val'>'foo'</span>

  <span class='module module kw'>module</span> <span class='Bio constant id'>Bio</span>

    <span class='autoload identifier id'>autoload</span> <span class='symbol val'>:Bar</span><span class='comma token'>,</span> <span class='string val'>'bio/bar'</span>

  <span class='class class kw'>class</span> <span class='Hoge constant id'>Hoge</span>
    <span class='colon op'>:</span>
  <span class='end end kw'>end</span>  <span class='comment val'># Hoge</span>

  <span class='end end kw'>end</span>  <span class='comment val'># Bio</span>
</pre>
<h3>of classes and methods within those files</h3>
<p>
Classes and methods should be documented in a standardized format, as in
the following example (from lib/bio/sequence.rb):
</p>
<pre class="code">
  <span class='comment val'># == Description</span>
  <span class='comment val'>#</span>
  <span class='comment val'># Bio::Sequence objects represent annotated sequences in bioruby.</span>
  <span class='comment val'># A Bio::Sequence object is a wrapper around the actual sequence,</span>
  <span class='comment val'># represented as either a Bio::Sequence::NA or a Bio::Sequence::AA object.</span>
  <span class='comment val'># For most users, this encapsulation will be completely transparent.</span>
  <span class='comment val'># Bio::Sequence responds to all methods defined for Bio::Sequence::NA/AA</span>
  <span class='comment val'># objects using the same arguments and returning the same values (even though</span>
  <span class='comment val'># these methods are not documented specifically for Bio::Sequence).</span>
  <span class='comment val'>#</span>
  <span class='comment val'># == Usage</span>
  <span class='comment val'>#</span>
  <span class='comment val'>#   require 'bio'</span>
  <span class='comment val'>#</span>
  <span class='comment val'>#   # Create a nucleic or amino acid sequence</span>
  <span class='comment val'>#   dna = Bio::Sequence.auto('atgcatgcATGCATGCAAAA')</span>
  <span class='comment val'>#   rna = Bio::Sequence.auto('augcaugcaugcaugcaaaa')</span>
  <span class='comment val'>#   aa = Bio::Sequence.auto('ACDEFGHIKLMNPQRSTVWYU')</span>
  <span class='comment val'>#</span>
  <span class='comment val'>#   # Print in FASTA format</span>
  <span class='comment val'>#   puts dna.output(:fasta)</span>
  <span class='comment val'>#</span>
  <span class='comment val'>#   # Print all codons</span>
  <span class='comment val'>#   dna.window_search(3,3) do |codon|</span>
  <span class='comment val'>#     puts codon</span>
  <span class='comment val'>#   end</span>
  <span class='comment val'>#</span>
  <span class='class class kw'>class</span> <span class='Sequence constant id'>Sequence</span>

    <span class='comment val'># Create a new Bio::Sequence object</span>
    <span class='comment val'>#</span>
    <span class='comment val'>#   s = Bio::Sequence.new('atgc')</span>
    <span class='comment val'>#   puts s                                  # =&gt; 'atgc'</span>
    <span class='comment val'>#</span>
    <span class='comment val'># Note that this method does not intialize the contained sequence</span>
    <span class='comment val'># as any kind of bioruby object, only as a simple string</span>
    <span class='comment val'>#</span>
    <span class='comment val'>#   puts s.seq.class                        # =&gt; String</span>
    <span class='comment val'>#</span>
    <span class='comment val'># See Bio::Sequence#na, Bio::Sequence#aa, and Bio::Sequence#auto</span>
    <span class='comment val'># for methods to transform the basic String of a just created</span>
    <span class='comment val'># Bio::Sequence object to a proper bioruby object</span>
    <span class='comment val'># ---</span>
    <span class='comment val'># *Arguments*:</span>
    <span class='comment val'># * (required) _str_: String or Bio::Sequence::NA/AA object</span>
    <span class='comment val'># *Returns*:: Bio::Sequence object</span>
    <span class='def def kw'>def</span> <span class='initialize identifier id'>initialize</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
      <span class='@seq ivar id'>@seq</span> <span class='assign token'>=</span> <span class='str identifier id'>str</span>
    <span class='end end kw'>end</span>

    <span class='comment val'># The sequence identifier.  For example, for a sequence</span>
    <span class='comment val'># of Genbank origin, this is the accession number.</span>
    <span class='attr_accessor identifier id'>attr_accessor</span> <span class='symbol val'>:entry_id</span>

    <span class='comment val'># An Array of Bio::Feature objects</span>
    <span class='attr_accessor identifier id'>attr_accessor</span> <span class='symbol val'>:features</span>
  <span class='end end kw'>end</span> <span class='comment val'># Sequence</span>
</pre>
<p>
Preceding the class definition (<tt>class Sequence</tt>), there is at least
a description and a usage example. Please use the <tt>Description</tt> and
<tt>Usage</tt> headings. If appropriate, refer to other classes that
interact with or are related to the class.
</p>
<p>
The code in the usage example should, if possible, be in a format that a
user can copy-and-paste into a new script to run. It should illustrate the
most important uses of the class. If possible and if it would not clutter
up the example too much, try to provide any input data directly into the
usage example, instead of refering to ARGV or ARGF for input.
</p>
<pre class="code">
  <span class='dna identifier id'>dna</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcATGCATGCAAAA'</span><span class='rparen token'>)</span>
</pre>
<p>
Otherwise, describe the input shortly, for example:
</p>
<pre class="code">
  <span class='comment val'># input should be string consisting of nucleotides</span>
  <span class='dna identifier id'>dna</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='ARGF constant id'>ARGF</span><span class='dot token'>.</span><span class='read identifier id'>read</span><span class='rparen token'>)</span>
</pre>
<p>
Methods should be preceded by a comment that describes what the method
does, including any relevant usage examples. (In contrast to the
documentation for the class itself, headings are not required.) In
addition, any arguments should be listed, as well as the type of thing that
is returned by the method. The format of this information is as follows:
</p>
<pre class="code">
  <span class='comment val'># ---</span>
  <span class='comment val'># *Arguments*:</span>
  <span class='comment val'># * (required) _str_: String or Bio::Sequence::NA</span>
  <span class='comment val'># * (optional) _nr_: a number that means something</span>
  <span class='comment val'># *Returns*:: true or false</span>
</pre>
<p>
Attribute accessors can be preceded by a short description.
</p>
<h2>Exception handling</h2>
<p>
Don&#8217;t use
</p>
<pre class="code">
  <span class='$stderr gvar id'>$stderr</span><span class='dot token'>.</span><span class='puts identifier id'>puts</span> <span class='string val'>&quot;WARNING&quot;</span>
</pre>
<p>
in your code. Instead, try to avoid printing error messages. For fatal
errors, use <tt>raise</tt> with an appropriate message.
</p>
<h2>Testing code should use &#8216;test/unit&#8217;</h2>
<p>
Unit tests should come with your modules by which you can assure what you
meant to do with each method. The test code is useful to make maintenance
easy and ensure stability. The use of
</p>
<pre class="code">
  <span class='if if kw'>if</span> <span class='__FILE__ __file__ kw'>__FILE__</span> <span class='eq op'>==</span> <span class='$0 gvar id'>$0</span>
</pre>
<p>
is deprecated.
</p>
<h2>Using autoload</h2>
<p>
To quicken the initial load time we have replaced most of
&#8216;require&#8217; to &#8216;autoload&#8217; since BioRuby version 0.7.
During this change, we have found some tips:
</p>
<p>
You should not separate the same namespace into several files.
</p>
<ul>
<li>For example, if you have separated definitions of the Bio::Foo class into
two files (e.g. &#8216;bio/foo.rb&#8217; and &#8216;bio/bar.rb&#8217;), you
need to resolve the dependencies (including the load order) yourself.

</li>
<li>If you have a defined Bio::Foo in &#8216;bio/foo.rb&#8217; and a defined
Bio::Foo::Bar in &#8216;bio/foo/bar.rb&#8217; add the following line in the
&#8216;bio/foo.rb&#8217; file:

<pre class="code">
  <span class='autoload identifier id'>autoload</span> <span class='symbol val'>:Bar</span><span class='comma token'>,</span> <span class='string val'>'bio/foo/bar'</span>
</pre>
</li>
</ul>
<p>
You should not put several top level namespaces in one file.
</p>
<ul>
<li>For example, if you have Bio::A, Bio::B and Bio::C in the file
&#8216;bio/foo.rb&#8217;, you need

<pre class="code">
  <span class='autoload identifier id'>autoload</span> <span class='symbol val'>:A</span><span class='comma token'>,</span> <span class='string val'>'bio/foo'</span>
  <span class='autoload identifier id'>autoload</span> <span class='symbol val'>:B</span><span class='comma token'>,</span> <span class='string val'>'bio/foo'</span>
  <span class='autoload identifier id'>autoload</span> <span class='symbol val'>:C</span><span class='comma token'>,</span> <span class='string val'>'bio/foo'</span>
</pre>
<p>
to load the module automatically (instead of require
&#8216;bio/foo&#8217;). In this case, you should put them under the new
namespace like Bio::Foo::A, Bio::Foo::B and Bio::Foo::C in the file
&#8216;bio/foo&#8217;, then use
</p>
<pre class="code">
  <span class='autoload identifier id'>autoload</span> <span class='symbol val'>:Foo</span><span class='comma token'>,</span> <span class='string val'>'bio/foo'</span>
</pre>
<p>
so autoload can be written in 1 line.
</p>
</li>
</ul>
<h1>NAMESPACE</h1>
<p>
Your module should be located under the top-level module Bio and put under
the &#8216;bioruby/lib/bio&#8217; directory. The class/module names and the
file names should be short and descriptive.
</p>
<p>
There are already several sub directories in &#8216;bioruby/lib&#8217;:
</p>
<pre class="code">
  <span class='bio identifier id'>bio</span><span class='div op'>/</span><span class='mult op'>*</span><span class='dot token'>.</span><span class='rb identifier id'>rb</span>   <span class='minus op'>-</span><span class='minus op'>-</span> <span class='general identifier id'>general</span> <span class='and and kw'>and</span> <span class='widely identifier id'>widely</span> <span class='used identifier id'>used</span> <span class='basic identifier id'>basic</span> <span class='classes identifier id'>classes</span>
  <span class='bio identifier id'>bio</span><span class='div op'>/</span><span class='appl identifier id'>appl</span><span class='div op'>/</span>  <span class='minus op'>-</span><span class='minus op'>-</span> <span class='wrapper identifier id'>wrapper</span> <span class='and and kw'>and</span> <span class='parser identifier id'>parser</span> <span class='for for kw'>for</span> <span class='the identifier id'>the</span> <span class='external identifier id'>external</span> <span class='applications identifier id'>applications</span>
  <span class='bio identifier id'>bio</span><span class='div op'>/</span><span class='data identifier id'>data</span><span class='div op'>/</span>  <span class='minus op'>-</span><span class='minus op'>-</span> <span class='basic identifier id'>basic</span> <span class='biological identifier id'>biological</span> <span class='data identifier id'>data</span>
  <span class='bio identifier id'>bio</span><span class='div op'>/</span><span class='db identifier id'>db</span><span class='div op'>/</span>    <span class='minus op'>-</span><span class='minus op'>-</span> <span class='flatfile identifier id'>flatfile</span> <span class='database identifier id'>database</span> <span class='entry identifier id'>entry</span> <span class='parsers identifier id'>parsers</span>
  <span class='bio identifier id'>bio</span><span class='div op'>/</span><span class='io identifier id'>io</span><span class='div op'>/</span>    <span class='minus op'>-</span><span class='minus op'>-</span> <span class='I constant id'>I</span><span class='div op'>/</span><span class='O constant id'>O</span> <span class='interfaces identifier id'>interfaces</span> <span class='for for kw'>for</span> <span class='files identifier id'>files</span><span class='comma token'>,</span> <span class='RDB constant id'>RDB</span><span class='comma token'>,</span> <span class='web identifier id'>web</span> <span class='services identifier id'>services</span> <span class='etc identifier id'>etc</span><span class='dot token'>.</span>
  <span class='bio identifier id'>bio</span><span class='div op'>/</span><span class='util identifier id'>util</span><span class='div op'>/</span>  <span class='minus op'>-</span><span class='minus op'>-</span> <span class='utilities identifier id'>utilities</span> <span class='and and kw'>and</span> <span class='algorithms identifier id'>algorithms</span> <span class='for for kw'>for</span> <span class='bioinformatics identifier id'>bioinformatics</span>
</pre>
<p>
If your module doesn&#8217;t match any of the above, please propose an
appropriate directory name when you contribute. Please let the staff
discuss on namespaces (class names), API (method names) before commiting a
new module or making changes on existing modules.
</p>
<h1>MAINTENANCE</h1>
<p>
Finally, please maintain the code you&#8217;ve contributed. Please let us
know (on the bioruby list) before you commit, so that users can discuss on
the change.
</p>

</div>
          </div>
        </div>
      </div>                    
    </div>

    <div id="footer">
      <div class="site">
        <div class="info">
          <div class="links">
            <a href="http://github.com/blog/148-github-shirts-now-available">Shirts</a> |
            <a href="http://github.com/blog">Blog</a> |
            <a href="http://support.github.com/">Support</a> |
            <a href="http://github.com/training">Training</a> |
            <a href="http://github.com/contact">Contact</a> |
            <a href="http://groups.google.com/group/github/">Google Group</a> |
            <a href="http://develop.github.com/">API</a> |
            <a href="http://twitter.com/github">Status</a>
          </div>
          <div class="company">
            <span id="_rrt" title="0.64509s from xc88-s00039">GitHub</span>
            is <a href="http://logicalawesome.com/">Logical Awesome</a> ©2009 | <a href="http://github.com/site/terms">Terms of Service</a> | <a href="http://github.com/site/privacy">Privacy Policy</a>
          </div>
        </div>
        <div class="sponsor">
          <a href="http://engineyard.com/"><img src="http://github.com/images/modules/footer/engine_yard_logo.png" alt="Engine Yard"></a>
          <div>
            Hosting provided by our<br> partners at Engine Yard
          </div>
        </div>
      </div>
    </div>
    
    <div id="coming_soon" style="display: none;">
      This feature is coming soon.  Sit tight!
    </div>

    <div id="facebox" style="display: none;">       
      <div class="popup">         
        <table>           
          <tbody>             
            <tr>               
              <td class="tl"></td>
              <td class="b"></td>
              <td class="tr"></td>             
            </tr>             
          <tr>               
            <td class="b"></td>               
            <td class="body">                 
              <div class="content"></div>                 
              <div class="footer">                   
                <a href="#" class="close"><img src="http://github.com/facebox/closelabel.gif" title="close" class="close_image"></a>                 
              </div>               
            </td>               
            <td class="b"></td>             
          </tr>             
          <tr>               
            <td class="bl"></td>
            <td class="b"></td>
            <td class="br"></td>             
          </tr>           
        </tbody>         
      </table>       
    </div>     


    <script type="text/javascript">
    var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
    document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
    </script>
    <script type="text/javascript">
    try {
    var pageTracker = _gat._getTracker("UA-8689483-1");
    pageTracker._trackPageview();
    } catch(err) {}</script>

  </body>
</html>