<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xml:lang="en" xmlns="http://www.w3.org/1999/xhtml" lang="en">
  <head>
    <title>bioruby's bioruby - Class: Bio::Spidey::Report::SegmentPair - GitHub Documentation</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta name="description" content="">

<link rel="search" type="application/opensearchdescription+xml" href="http://github.com/opensearch.xml" title="GitHub">
<link rel="fluid-icon" href="http://github.com/fluidicon.png" title="GitHub">
<link href="http://assets0.github.com/stylesheets/bundle.css" media="screen" rel="stylesheet" type="text/css">
<link href="http://github.com/feeds/bioruby/commits/bioruby/master" rel="alternate" title="Recent Commits to bioruby:master" type="application/atom+xml">

<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js"></script>
<script type="text/javascript" src="http://assets3.github.com/javascripts/bundle.js"></script>              
<script src="../../../app.js" type="text/javascript" charset="utf-8"></script>

<script type="text/javascript" src="http://docs.github.com/javascripts/md5.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/repos.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/commits.js"></script>
<script type="text/javascript">
  $(document).ready(function(){
    if (typeof skipHeaders != 'undefined') return;
    
    GitHubRepo.init("bioruby", "bioruby", function(){ 
      GitHubRepo.links = " <a id='namespaces_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/namespaces_button.png' alt='Modules and Classes'/></a> ";
      GitHubRepo.links += " <a id='methods_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/methods_button.png' alt='Methods'/></a> ";
      $("#repo").html(GitHubRepo.content());
      $('#namespaces_button').click(function(){
        var url="/"+$(this).attr('rel')+"/namespaces/";
        $.gitbox(url);
        return false;
      });    
      $('#methods_button').click(function(){
        var url="/"+$(this).attr('rel')+"/methods/";
        $.gitbox(url);
        return false;
      });        
    });
    GitHubCommit.init("bioruby", "bioruby", "97b9284109c9a4431b92eab208509e1df6069b4b", function(){ 
      $("#commit").html(GitHubCommit.content());
    });
  });
</script>

    <link rel="stylesheet" href="../../../style.css" type="text/css" charset="utf-8" />
  </head>
  <body>    
    
    
    
        <div id="main" style="padding-bottom:8em;">
      <div id="header">
        <div class="site">
          <div class="logo">
            <a href="http://github.com/"><img src="http://github.com/images/modules/header/logov3.png" alt="github"></a>
          </div>          
          <div class="topsearch">
            <form action="http://github.com/search" id="top_search_form" method="get">
              <input class="search" name="q" type="search"> <input value="Search" type="submit">
            </form>
            <div class="links">
              <a href="http://github.com/repositories">Browse</a> | <a href="http://github.com/guides">Guides</a> | <a href="http://github.com/search">Advanced</a>
            </div>
          </div>
          <div class="actions">
            <a href="http://github.com">Home</a>
            <a href="http://github.com/plans"><b><u>Pricing and Signup</u></b></a>
            <a href="http://docs.github.com/">Documentation</a>
            <a href="https://github.com/login">Login</a>
          </div>                  
        </div>
      </div>    
                                                  
      <div id="repo_menu">
        <div class="site">
          <ul>          
            <li class=""><a href="http://github.com/bioruby/bioruby/tree/">Source</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/commits/">Commits</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/network">Network</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/issues">Issues</a></li>
            <li class=""><a href="http://wiki.github.com/bioruby/bioruby">Wiki</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/graphs">Graphs</a></li>                    
            <li class="active"><a href="/bioruby/bioruby">Documentation</a></li>
          </ul>
        </div>
      </div>

      <div id="repo_sub_menu">
        <div class="site">
          <div class="joiner"></div>      
        </div>
      </div>

      <div class="site">  
        <div id="repos">
          <div id="repo" class="repo public">
            <div class="title">
              <div class="path">
                <a href="http://github.com/bioruby">bioruby</a> / <b><a href="http://github.com/bioruby/bioruby/tree">bioruby</a></b>        
                <a id="namespaces_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/namespaces_button.png" alt="Modules and Classes"/></a>
                <a id="methods_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/methods_button.png" alt="Methods"/></a>
              </div>
            </div>            
          </div>
          <div id="commit">
          </div> 
          <br/>                                       
          <div id="content">
            <div class="wikistyle">
  <div class="section class Bio_Spidey_Report_SegmentPair">
  <h1>Class: Bio::Spidey::Report::SegmentPair</h1>
  <div class="section docstring">
  <p>
Sequence segment pair of Spidey result. Similar to Bio::Blast::Report::Hsp
but lacks many methods. For mRNA-genome mapping programs, unlike other
homology search programs, the class is used not only for exons but also for
introns. (Note that intron data would not be available according to
run-time options of the program.)
</p>

</div><div class="section attributes">
  
    <div class="">
      <h2>Instance Attributes</h2>
      <table>
      
        <tr>
          <th class="name">aaseqline</td>
          <td class="readwrite">
            [<span id='aaseqline-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns amino acide sequence in alignment.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">align_len</td>
          <td class="readwrite">
            [<span id='align_len-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns alignment length of the segment pair.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">gaps</td>
          <td class="readwrite">
            [<span id='gaps-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns gaps.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">genomic</td>
          <td class="readwrite">
            [<span id='genomic-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns segment informations of the &#8216;Genomic&#8217;.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">midline</td>
          <td class="readwrite">
            [<span id='midline-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns the middle line of the alignment of the segment pair.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">mismatches</td>
          <td class="readwrite">
            [<span id='mismatches-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns mismatches.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">mrna</td>
          <td class="readwrite">
            [<span id='mrna-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns segment informations of the &#8216;mRNA&#8217;.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">percent_identity</td>
          <td class="readwrite">
            [<span id='percent_identity-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns percent identity of the segment pair.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">splice_site</td>
          <td class="readwrite">
            [<span id='splice_site-instance_method'>R</span>W] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Returns splice site information.
</p>

            
          </td>
        </tr>
      
      </table>
    </div>
  
</div><div class="section constants">
  
</div><div class="section constructor">
<h2>Constructor</h2>
  <div id="initialize-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>initialize</span><span class='args'>(genomic, mrna, midline, aaseqline, percent_identity, mismatches, gaps, splice_site, align_len)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
&#8212; segment pair (like Bio::BLAST::*::Report::Hsp) ++ Creates a new
SegmentPair object. It is designed to be called from Bio::Spidey::Report::*
classes. Users shall not call it directly.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 125</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L125">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">125
126
127
128
129
130
131
132
133
134
135
136
137</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='initialize identifier id'>initialize</span><span class='lparen token'>(</span><span class='genomic identifier id'>genomic</span><span class='comma token'>,</span> <span class='mrna identifier id'>mrna</span><span class='comma token'>,</span> <span class='midline identifier id'>midline</span><span class='comma token'>,</span> <span class='aaseqline identifier id'>aaseqline</span><span class='comma token'>,</span>
               <span class='percent_identity identifier id'>percent_identity</span><span class='comma token'>,</span> <span class='mismatches identifier id'>mismatches</span><span class='comma token'>,</span> <span class='gaps identifier id'>gaps</span><span class='comma token'>,</span> <span class='splice_site identifier id'>splice_site</span><span class='comma token'>,</span>
               <span class='align_len identifier id'>align_len</span><span class='rparen token'>)</span>
  <span class='@genomic ivar id'>@genomic</span>   <span class='assign token'>=</span> <span class='genomic identifier id'>genomic</span>
  <span class='@mrna ivar id'>@mrna</span>      <span class='assign token'>=</span> <span class='mrna identifier id'>mrna</span>
  <span class='@midline ivar id'>@midline</span>   <span class='assign token'>=</span> <span class='midline identifier id'>midline</span>
  <span class='@aaseqline ivar id'>@aaseqline</span> <span class='assign token'>=</span> <span class='aaseqline identifier id'>aaseqline</span>
  <span class='@percent_identity ivar id'>@percent_identity</span> <span class='assign token'>=</span> <span class='percent_identity identifier id'>percent_identity</span>
  <span class='@mismaches ivar id'>@mismaches</span>        <span class='assign token'>=</span> <span class='mismatches identifier id'>mismatches</span>
  <span class='@gaps ivar id'>@gaps</span>             <span class='assign token'>=</span> <span class='gaps identifier id'>gaps</span>
  <span class='@splice_site ivar id'>@splice_site</span>      <span class='assign token'>=</span> <span class='splice_site identifier id'>splice_site</span>
  <span class='@align_len ivar id'>@align_len</span>        <span class='assign token'>=</span> <span class='align_len identifier id'>align_len</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
</div>  <div class="section visibilitygroup">
      <div class="section methodsummary">
    <h2>Public Class Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#new_intron-class_method">new_intron</span><span class='args'>(from, to, strand, aln)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Creates a new SegmentPair object when the segment pair is an intron.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#parse-class_method">parse</span><span class='args'>(str, strand, complement, aln)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Parses a piece of Spidey result text and creates a new SegmentPair object.
</p>

        
      </td>
    </tr>
  
</table>
  </div>
  <div class="section methodsummary">
    <h2>Public Instance Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#hit_from-instance_method">#hit_from</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns start position of the genomic (target, hit) (the first position is
1).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#hit_strand-instance_method">#hit_strand</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns strand information of the genomic (target, hit).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#hit_to-instance_method">#hit_to</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns end position (including its position) of the genomic (target, hit).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#hseq-instance_method">#hseq</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns the sequence (with gaps) of the genomic (target, hit).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#qseq-instance_method">#qseq</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns the sequence (with gaps) of the mRNA (query).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#query_from-instance_method">#query_from</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
&#8212; Bio::BLAST::*::Report::Hsp compatible methods
</p>
<pre class="code">
  <span class='Methods constant id'>Methods</span> <span class='already identifier id'>already</span> <span class='defined identifier id'>defined</span><span class='colon op'>:</span> <span class='midline identifier id'>midline</span><span class='comma token'>,</span> <span class='percent_identity identifier id'>percent_identity</span><span class='comma token'>,</span>
    <span class='gaps identifier id'>gaps</span><span class='comma token'>,</span> <span class='align_len identifier id'>align_len</span><span class='comma token'>,</span> <span class='mismatch_count identifier id'>mismatch_count</span>
</pre>
<p>
++ Returns start position of the mRNA (query) (the first position is 1).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#query_strand-instance_method">#query_strand</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns strand information of the mRNA (query).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#query_to-instance_method">#query_to</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Returns end position (including its position) of the mRNA (query).
</p>

        
      </td>
    </tr>
  
</table>
  </div>
<div class="section methoddetails">
  <h2>Public Class Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>new_intron</h3>
</div><div id="new_intron-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>new_intron</span><span class='args'>(from, to, strand, aln)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Creates a new SegmentPair object when the segment pair is an intron. It is
designed to be called internally from Bio::Spidey::Report::* classes. Users
shall not call it directly.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 182</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L182">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">182
183
184
185
186
187
188
189</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='new_intron identifier id'>new_intron</span><span class='lparen token'>(</span><span class='from identifier id'>from</span><span class='comma token'>,</span> <span class='to identifier id'>to</span><span class='comma token'>,</span> <span class='strand identifier id'>strand</span><span class='comma token'>,</span> <span class='aln identifier id'>aln</span><span class='rparen token'>)</span>
  <span class='genomic identifier id'>genomic</span>   <span class='assign token'>=</span> <span class='Segment constant id'>Segment</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='from identifier id'>from</span><span class='comma token'>,</span> <span class='to identifier id'>to</span><span class='comma token'>,</span> <span class='strand identifier id'>strand</span><span class='comma token'>,</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>0</span><span class='rbrack token'>]</span><span class='rparen token'>)</span>
  <span class='mrna identifier id'>mrna</span>      <span class='assign token'>=</span> <span class='Segment constant id'>Segment</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span>    <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>2</span><span class='rbrack token'>]</span><span class='rparen token'>)</span>
  <span class='midline identifier id'>midline</span>   <span class='assign token'>=</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>1</span><span class='rbrack token'>]</span>
  <span class='aaseqline identifier id'>aaseqline</span> <span class='assign token'>=</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>3</span><span class='rbrack token'>]</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='genomic identifier id'>genomic</span><span class='comma token'>,</span> <span class='mrna identifier id'>mrna</span><span class='comma token'>,</span> <span class='midline identifier id'>midline</span><span class='comma token'>,</span> <span class='aaseqline identifier id'>aaseqline</span><span class='comma token'>,</span>
           <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='nil nil kw'>nil</span><span class='comma token'>,</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>parse</h3>
</div><div id="parse-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>parse</span><span class='args'>(str, strand, complement, aln)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Parses a piece of Spidey result text and creates a new SegmentPair object.
It is designed to be called internally from Bio::Spidey::Report::* classes.
Users shall not call it directly.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 196</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L196">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='parse identifier id'>parse</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='comma token'>,</span> <span class='strand identifier id'>strand</span><span class='comma token'>,</span> <span class='complement identifier id'>complement</span><span class='comma token'>,</span> <span class='aln identifier id'>aln</span><span class='rparen token'>)</span>
  <span class='regexp val'>/\AExon\s*\d+(\(\-\))?\:\s*(\d+)\-(\d+)\s*\(gen\)\s+(\d+)\-(\d+)\s*\(mRNA\)\s+id\s*([\d\.]+)\s*\%\s+mismatches\s+(\d+)\s+gaps\s+(\d+)\s+splice site\s*\(d +a\)\s*\:\s*(\d+)\s+(\d+)/</span> <span class='match op'>=~</span> <span class='str identifier id'>str</span>
  <span class='if if kw'>if</span> <span class='strand identifier id'>strand</span> <span class='eq op'>==</span> <span class='string val'>'minus'</span> <span class='then then kw'>then</span>
    <span class='genomic identifier id'>genomic</span> <span class='assign token'>=</span> <span class='Segment constant id'>Segment</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='$3 nth_ref id'>$3</span><span class='comma token'>,</span> <span class='$2 nth_ref id'>$2</span><span class='comma token'>,</span> <span class='strand identifier id'>strand</span><span class='comma token'>,</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>0</span><span class='rbrack token'>]</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='genomic identifier id'>genomic</span> <span class='assign token'>=</span> <span class='Segment constant id'>Segment</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='$2 nth_ref id'>$2</span><span class='comma token'>,</span> <span class='$3 nth_ref id'>$3</span><span class='comma token'>,</span> <span class='string val'>'plus'</span><span class='comma token'>,</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>0</span><span class='rbrack token'>]</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='if if kw'>if</span> <span class='complement identifier id'>complement</span> <span class='then then kw'>then</span>
    <span class='mrna identifier id'>mrna</span>    <span class='assign token'>=</span> <span class='Segment constant id'>Segment</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='$4 nth_ref id'>$4</span><span class='comma token'>,</span> <span class='$5 nth_ref id'>$5</span><span class='comma token'>,</span> <span class='string val'>'minus'</span><span class='comma token'>,</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>2</span><span class='rbrack token'>]</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='mrna identifier id'>mrna</span>    <span class='assign token'>=</span> <span class='Segment constant id'>Segment</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='$4 nth_ref id'>$4</span><span class='comma token'>,</span> <span class='$5 nth_ref id'>$5</span><span class='comma token'>,</span> <span class='string val'>'plus'</span><span class='comma token'>,</span>  <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>2</span><span class='rbrack token'>]</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='percent_identity identifier id'>percent_identity</span> <span class='assign token'>=</span> <span class='$6 nth_ref id'>$6</span>
  <span class='mismatches identifier id'>mismatches</span> <span class='assign token'>=</span> <span class='lparen token'>(</span><span class='$7 nth_ref id'>$7</span> <span class='question op'>?</span> <span class='$7 nth_ref id'>$7</span><span class='dot token'>.</span><span class='to_i identifier id'>to_i</span> <span class='colon op'>:</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='gaps identifier id'>gaps</span> <span class='assign token'>=</span> <span class='lparen token'>(</span><span class='$8 nth_ref id'>$8</span> <span class='question op'>?</span> <span class='$8 nth_ref id'>$8</span><span class='dot token'>.</span><span class='to_i identifier id'>to_i</span> <span class='colon op'>:</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='splice_site identifier id'>splice_site</span> <span class='assign token'>=</span> <span class='lbrace token'>{</span>
    <span class='symbol val'>:d</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='lparen token'>(</span><span class='$9 nth_ref id'>$9</span>  <span class='question op'>?</span> <span class='$9 nth_ref id'>$9</span><span class='dot token'>.</span><span class='to_i identifier id'>to_i</span>  <span class='colon op'>:</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span><span class='comma token'>,</span>
    <span class='symbol val'>:a</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='lparen token'>(</span><span class='$10 nth_ref id'>$10</span> <span class='question op'>?</span> <span class='$10 nth_ref id'>$10</span><span class='dot token'>.</span><span class='to_i identifier id'>to_i</span> <span class='colon op'>:</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='rbrace token'>}</span>
  <span class='midline identifier id'>midline</span>   <span class='assign token'>=</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>1</span><span class='rbrack token'>]</span>
  <span class='aaseqline identifier id'>aaseqline</span> <span class='assign token'>=</span> <span class='aln identifier id'>aln</span><span class='lbrack token'>[</span><span class='integer val'>3</span><span class='rbrack token'>]</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='genomic identifier id'>genomic</span><span class='comma token'>,</span> <span class='mrna identifier id'>mrna</span><span class='comma token'>,</span> <span class='midline identifier id'>midline</span><span class='comma token'>,</span> <span class='aaseqline identifier id'>aaseqline</span><span class='comma token'>,</span>
           <span class='percent_identity identifier id'>percent_identity</span><span class='comma token'>,</span> <span class='mismatches identifier id'>mismatches</span><span class='comma token'>,</span> <span class='gaps identifier id'>gaps</span><span class='comma token'>,</span> <span class='splice_site identifier id'>splice_site</span><span class='comma token'>,</span>
           <span class='lparen token'>(</span><span class='midline identifier id'>midline</span> <span class='integer val'>? </span><span class='midline identifier id'>midline</span><span class='dot token'>.</span><span class='length identifier id'>length</span> <span class='colon op'>:</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div><div class="section methoddetails">
  <h2>Public Instance Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>hit_from</h3>
</div><div id="hit_from-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>hit_from</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns start position of the genomic (target, hit) (the first position is
1).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 243</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L243">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">243</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='hit_from identifier id'>hit_from</span><span class='semicolon token'>;</span>     <span class='@genomic ivar id'>@genomic</span><span class='dot token'>.</span><span class='from identifier id'>from</span><span class='semicolon token'>;</span>    <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>hit_strand</h3>
</div><div id="hit_strand-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>hit_strand</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns strand information of the genomic (target, hit). Returns
&#8216;plus&#8217;, &#8216;minus&#8217;, or nil.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 254</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L254">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">254</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='hit_strand identifier id'>hit_strand</span><span class='semicolon token'>;</span>   <span class='@genomic ivar id'>@genomic</span><span class='dot token'>.</span><span class='strand identifier id'>strand</span><span class='semicolon token'>;</span>  <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>hit_to</h3>
</div><div id="hit_to-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>hit_to</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns end position (including its position) of the genomic (target, hit).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 247</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L247">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">247</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='hit_to identifier id'>hit_to</span><span class='semicolon token'>;</span>       <span class='@genomic ivar id'>@genomic</span><span class='dot token'>.</span><span class='to identifier id'>to</span><span class='semicolon token'>;</span>      <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>hseq</h3>
</div><div id="hseq-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>hseq</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns the sequence (with gaps) of the genomic (target, hit).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 250</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L250">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">250</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='hseq identifier id'>hseq</span><span class='semicolon token'>;</span>         <span class='@genomic ivar id'>@genomic</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='semicolon token'>;</span>     <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>qseq</h3>
</div><div id="qseq-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>qseq</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns the sequence (with gaps) of the mRNA (query).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 235</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L235">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">235</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='qseq identifier id'>qseq</span><span class='semicolon token'>;</span>         <span class='@mrna ivar id'>@mrna</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='semicolon token'>;</span>        <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>query_from</h3>
</div><div id="query_from-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>query_from</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
&#8212; Bio::BLAST::*::Report::Hsp compatible methods
</p>
<pre class="code">
  <span class='Methods constant id'>Methods</span> <span class='already identifier id'>already</span> <span class='defined identifier id'>defined</span><span class='colon op'>:</span> <span class='midline identifier id'>midline</span><span class='comma token'>,</span> <span class='percent_identity identifier id'>percent_identity</span><span class='comma token'>,</span>
    <span class='gaps identifier id'>gaps</span><span class='comma token'>,</span> <span class='align_len identifier id'>align_len</span><span class='comma token'>,</span> <span class='mismatch_count identifier id'>mismatch_count</span>
</pre>
<p>
++ Returns start position of the mRNA (query) (the first position is 1).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 229</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L229">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">229</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='query_from identifier id'>query_from</span><span class='semicolon token'>;</span>   <span class='@mrna ivar id'>@mrna</span><span class='dot token'>.</span><span class='from identifier id'>from</span><span class='semicolon token'>;</span>       <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>query_strand</h3>
</div><div id="query_strand-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>query_strand</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns strand information of the mRNA (query). Returns &#8216;plus&#8217;,
&#8216;minus&#8217;, or nil.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 239</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L239">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">239</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='query_strand identifier id'>query_strand</span><span class='semicolon token'>;</span> <span class='@mrna ivar id'>@mrna</span><span class='dot token'>.</span><span class='strand identifier id'>strand</span><span class='semicolon token'>;</span>     <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>query_to</h3>
</div><div id="query_to-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>query_to</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Returns end position (including its position) of the mRNA (query).
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/appl/spidey/report.rb', line 232</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/appl/spidey/report.rb#L232">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">232</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='query_to identifier id'>query_to</span><span class='semicolon token'>;</span>     <span class='@mrna ivar id'>@mrna</span><span class='dot token'>.</span><span class='to identifier id'>to</span><span class='semicolon token'>;</span>         <span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div>
  </div>

</div>
</div>
          </div>
        </div>
      </div>                    
    </div>

    <div id="footer">
      <div class="site">
        <div class="info">
          <div class="links">
            <a href="http://github.com/blog/148-github-shirts-now-available">Shirts</a> |
            <a href="http://github.com/blog">Blog</a> |
            <a href="http://support.github.com/">Support</a> |
            <a href="http://github.com/training">Training</a> |
            <a href="http://github.com/contact">Contact</a> |
            <a href="http://groups.google.com/group/github/">Google Group</a> |
            <a href="http://develop.github.com/">API</a> |
            <a href="http://twitter.com/github">Status</a>
          </div>
          <div class="company">
            <span id="_rrt" title="0.64509s from xc88-s00039">GitHub</span>
            is <a href="http://logicalawesome.com/">Logical Awesome</a> ©2009 | <a href="http://github.com/site/terms">Terms of Service</a> | <a href="http://github.com/site/privacy">Privacy Policy</a>
          </div>
        </div>
        <div class="sponsor">
          <a href="http://engineyard.com/"><img src="http://github.com/images/modules/footer/engine_yard_logo.png" alt="Engine Yard"></a>
          <div>
            Hosting provided by our<br> partners at Engine Yard
          </div>
        </div>
      </div>
    </div>
    
    <div id="coming_soon" style="display: none;">
      This feature is coming soon.  Sit tight!
    </div>

    <div id="facebox" style="display: none;">       
      <div class="popup">         
        <table>           
          <tbody>             
            <tr>               
              <td class="tl"></td>
              <td class="b"></td>
              <td class="tr"></td>             
            </tr>             
          <tr>               
            <td class="b"></td>               
            <td class="body">                 
              <div class="content"></div>                 
              <div class="footer">                   
                <a href="#" class="close"><img src="http://github.com/facebox/closelabel.gif" title="close" class="close_image"></a>                 
              </div>               
            </td>               
            <td class="b"></td>             
          </tr>             
          <tr>               
            <td class="bl"></td>
            <td class="b"></td>
            <td class="br"></td>             
          </tr>           
        </tbody>         
      </table>       
    </div>     


    <script type="text/javascript">
    var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
    document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
    </script>
    <script type="text/javascript">
    try {
    var pageTracker = _gat._getTracker("UA-8689483-1");
    pageTracker._trackPageview();
    } catch(err) {}</script>

  </body>
</html>