<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xml:lang="en" xmlns="http://www.w3.org/1999/xhtml" lang="en">
  <head>
    <title>bioruby's bioruby - Class: Bio::Sequence - GitHub Documentation</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
<meta name="description" content="">

<link rel="search" type="application/opensearchdescription+xml" href="http://github.com/opensearch.xml" title="GitHub">
<link rel="fluid-icon" href="http://github.com/fluidicon.png" title="GitHub">
<link href="http://assets0.github.com/stylesheets/bundle.css" media="screen" rel="stylesheet" type="text/css">
<link href="http://github.com/feeds/bioruby/commits/bioruby/master" rel="alternate" title="Recent Commits to bioruby:master" type="application/atom+xml">

<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.min.js"></script>
<script type="text/javascript" src="http://assets3.github.com/javascripts/bundle.js"></script>              
<script src="../app.js" type="text/javascript" charset="utf-8"></script>

<script type="text/javascript" src="http://docs.github.com/javascripts/md5.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/repos.js"></script>
<script type="text/javascript" src="http://docs.github.com/javascripts/commits.js"></script>
<script type="text/javascript">
  $(document).ready(function(){
    if (typeof skipHeaders != 'undefined') return;
    
    GitHubRepo.init("bioruby", "bioruby", function(){ 
      GitHubRepo.links = " <a id='namespaces_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/namespaces_button.png' alt='Modules and Classes'/></a> ";
      GitHubRepo.links += " <a id='methods_button' rel='bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b' href='#'><img class='button' src='http://docs.github.com/methods_button.png' alt='Methods'/></a> ";
      $("#repo").html(GitHubRepo.content());
      $('#namespaces_button').click(function(){
        var url="/"+$(this).attr('rel')+"/namespaces/";
        $.gitbox(url);
        return false;
      });    
      $('#methods_button').click(function(){
        var url="/"+$(this).attr('rel')+"/methods/";
        $.gitbox(url);
        return false;
      });        
    });
    GitHubCommit.init("bioruby", "bioruby", "97b9284109c9a4431b92eab208509e1df6069b4b", function(){ 
      $("#commit").html(GitHubCommit.content());
    });
  });
</script>

    <link rel="stylesheet" href="../style.css" type="text/css" charset="utf-8" />
  </head>
  <body>    
    
    
    
        <div id="main" style="padding-bottom:8em;">
      <div id="header">
        <div class="site">
          <div class="logo">
            <a href="http://github.com/"><img src="http://github.com/images/modules/header/logov3.png" alt="github"></a>
          </div>          
          <div class="topsearch">
            <form action="http://github.com/search" id="top_search_form" method="get">
              <input class="search" name="q" type="search"> <input value="Search" type="submit">
            </form>
            <div class="links">
              <a href="http://github.com/repositories">Browse</a> | <a href="http://github.com/guides">Guides</a> | <a href="http://github.com/search">Advanced</a>
            </div>
          </div>
          <div class="actions">
            <a href="http://github.com">Home</a>
            <a href="http://github.com/plans"><b><u>Pricing and Signup</u></b></a>
            <a href="http://docs.github.com/">Documentation</a>
            <a href="https://github.com/login">Login</a>
          </div>                  
        </div>
      </div>    
                                                  
      <div id="repo_menu">
        <div class="site">
          <ul>          
            <li class=""><a href="http://github.com/bioruby/bioruby/tree/">Source</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/commits/">Commits</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/network">Network</a></li>            
            <li class=""><a href="http://github.com/bioruby/bioruby/issues">Issues</a></li>
            <li class=""><a href="http://wiki.github.com/bioruby/bioruby">Wiki</a></li>
            <li class=""><a href="http://github.com/bioruby/bioruby/graphs">Graphs</a></li>                    
            <li class="active"><a href="/bioruby/bioruby">Documentation</a></li>
          </ul>
        </div>
      </div>

      <div id="repo_sub_menu">
        <div class="site">
          <div class="joiner"></div>      
        </div>
      </div>

      <div class="site">  
        <div id="repos">
          <div id="repo" class="repo public">
            <div class="title">
              <div class="path">
                <a href="http://github.com/bioruby">bioruby</a> / <b><a href="http://github.com/bioruby/bioruby/tree">bioruby</a></b>        
                <a id="namespaces_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/namespaces_button.png" alt="Modules and Classes"/></a>
                <a id="methods_button" rel="bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b" href="#"><img class="button" src="http://docs.github.com/methods_button.png" alt="Methods"/></a>
              </div>
            </div>            
          </div>
          <div id="commit">
          </div> 
          <br/>                                       
          <div id="content">
            <div class="wikistyle">
  <div class="section class Bio_Sequence">
  <h1>Class: Bio::Sequence</h1>
  <div class="section mixins">
  <h2>Included Modules</h2>
  <p><a href="Sequence/Format.html" title="Bio::Sequence::Format">Bio::Sequence::Format</a></p>
</div><div class="section docstring">
  <p>
Bio::Sequence objects represent annotated sequences in bioruby. A
Bio::Sequence object is a wrapper around the actual sequence, represented
as either a Bio::Sequence::NA or a Bio::Sequence::AA object. For most
users, this encapsulation will be completely transparent. Bio::Sequence
responds to all methods defined for Bio::Sequence::NA/AA objects using the
same arguments and returning the same values (even though these methods are
not documented specifically for Bio::Sequence).
</p>
<h1>USAGE</h1>
<pre class="code">
  <span class='comment val'># Create a nucleic or amino acid sequence</span>
  <span class='dna identifier id'>dna</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcATGCATGCAAAA'</span><span class='rparen token'>)</span>
  <span class='rna identifier id'>rna</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='string val'>'augcaugcaugcaugcaaaa'</span><span class='rparen token'>)</span>
  <span class='aa identifier id'>aa</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='string val'>'ACDEFGHIKLMNPQRSTVWYU'</span><span class='rparen token'>)</span>

  <span class='comment val'># Print it out</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span>
  <span class='puts identifier id'>puts</span> <span class='aa identifier id'>aa</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span>

  <span class='comment val'># Get a subsequence, bioinformatics style (first nucleotide is '1')</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='subseq identifier id'>subseq</span><span class='lparen token'>(</span><span class='integer val'>2</span><span class='comma token'>,</span><span class='integer val'>6</span><span class='rparen token'>)</span>

  <span class='comment val'># Get a subsequence, informatics style (first nucleotide is '0')</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='lbrack token'>[</span><span class='integer val'>2</span><span class='comma token'>,</span><span class='integer val'>6</span><span class='rbrack token'>]</span>

  <span class='comment val'># Print in FASTA format</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='output identifier id'>output</span><span class='lparen token'>(</span><span class='symbol val'>:fasta</span><span class='rparen token'>)</span>

  <span class='comment val'># Print all codons</span>
  <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='window_search identifier id'>window_search</span><span class='lparen token'>(</span><span class='integer val'>3</span><span class='comma token'>,</span><span class='integer val'>3</span><span class='rparen token'>)</span> <span class='do do kw'>do</span> <span class='bitor op'>|</span><span class='codon identifier id'>codon</span><span class='bitor op'>|</span>
    <span class='puts identifier id'>puts</span> <span class='codon identifier id'>codon</span>
  <span class='end end kw'>end</span>

  <span class='comment val'># Splice or otherwise mangle your sequence</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='splicing identifier id'>splicing</span><span class='lparen token'>(</span><span class='string val'>&quot;complement(join(1..5,16..20))&quot;</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='rna identifier id'>rna</span><span class='dot token'>.</span><span class='splicing identifier id'>splicing</span><span class='lparen token'>(</span><span class='string val'>&quot;complement(join(1..5,16..20))&quot;</span><span class='rparen token'>)</span>

  <span class='comment val'># Convert a sequence containing ambiguity codes into a</span>
  <span class='comment val'># regular expression you can use for subsequent searching</span>
  <span class='puts identifier id'>puts</span> <span class='aa identifier id'>aa</span><span class='dot token'>.</span><span class='to_re identifier id'>to_re</span>

  <span class='comment val'># These should speak for themselves</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='complement identifier id'>complement</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='molecular_weight identifier id'>molecular_weight</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='translate identifier id'>translate</span>
  <span class='puts identifier id'>puts</span> <span class='dna identifier id'>dna</span><span class='dot token'>.</span><span class='gc_percent identifier id'>gc_percent</span>
</pre>

</div><div class="section attributes">
  
    <div class="">
      <h2>Instance Attributes</h2>
      <table>
      
        <tr>
          <th class="name">classification</td>
          <td class="readwrite">
            [<span id='classification-instance_method'>R</span><span id='classification=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Organism classification, taxonomic classification of the source organism.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">comments</td>
          <td class="readwrite">
            [<span id='comments-instance_method'>R</span><span id='comments=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Comments (String or an Array of String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">data_class</td>
          <td class="readwrite">
            [<span id='data_class-instance_method'>R</span><span id='data_class=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Data Class defined by EMBL (String) See http://www.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">date_created</td>
          <td class="readwrite">
            [<span id='date_created-instance_method'>R</span><span id='date_created=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Created date of the sequence entry (Date, DateTime, Time, or String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">date_modified</td>
          <td class="readwrite">
            [<span id='date_modified-instance_method'>R</span><span id='date_modified=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Last modified date of the sequence entry (Date, DateTime, Time, or String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">dblinks</td>
          <td class="readwrite">
            [<span id='dblinks-instance_method'>R</span><span id='dblinks=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Links to other database entries.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">definition</td>
          <td class="readwrite">
            [<span id='definition-instance_method'>R</span><span id='definition=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
A String with a description of the sequence (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">division</td>
          <td class="readwrite">
            [<span id='division-instance_method'>R</span><span id='division=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Taxonomic Division defined by EMBL/GenBank/DDBJ (String) See http://www.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">entry_id</td>
          <td class="readwrite">
            [<span id='entry_id-instance_method'>R</span><span id='entry_id=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
The sequence identifier (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">entry_version</td>
          <td class="readwrite">
            [<span id='entry_version-instance_method'>R</span><span id='entry_version=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Version of the entry (String or Integer).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">features</td>
          <td class="readwrite">
            [<span id='features-instance_method'>R</span><span id='features=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Features (An Array of Bio::Feature objects).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">id_namespace</td>
          <td class="readwrite">
            [<span id='id_namespace-instance_method'>R</span><span id='id_namespace=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Namespace of the sequence IDs described in entry_id, primary_accession, and
secondary_accessions methods (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">keywords</td>
          <td class="readwrite">
            [<span id='keywords-instance_method'>R</span><span id='keywords=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Keywords (An Array of String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">molecule_type</td>
          <td class="readwrite">
            [<span id='molecule_type-instance_method'>R</span><span id='molecule_type=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
molecular type (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">moltype</td>
          <td class="readwrite">
            [<span id='moltype-instance_method'>R</span><span id='moltype=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Bio::Sequence::NA/AA.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">organelle</td>
          <td class="readwrite">
            [<span id='organelle-instance_method'>R</span><span id='organelle=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
(not well supported) Organelle information (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">other_seqids</td>
          <td class="readwrite">
            [<span id='other_seqids-instance_method'>R</span><span id='other_seqids=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Sequence identifiers which are not described in entry_id,
primary_accession,and secondary_accessions methods (Array of
Bio::Sequence::DBLink objects).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">primary_accession</td>
          <td class="readwrite">
            [<span id='primary_accession-instance_method'>R</span><span id='primary_accession=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Primary accession number (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">references</td>
          <td class="readwrite">
            [<span id='references-instance_method'>R</span><span id='references=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
References (An Array of Bio::Reference objects).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">release_created</td>
          <td class="readwrite">
            [<span id='release_created-instance_method'>R</span><span id='release_created=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Release information when created (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">release_modified</td>
          <td class="readwrite">
            [<span id='release_modified-instance_method'>R</span><span id='release_modified=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Release information when last-modified (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">secondary_accessions</td>
          <td class="readwrite">
            [<span id='secondary_accessions-instance_method'>R</span><span id='secondary_accessions=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Secondary accession numbers (Array of String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">sequence_version</td>
          <td class="readwrite">
            [<span id='sequence_version-instance_method'>R</span><span id='sequence_version=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <hr size="1"></hr><p>
Attributes below have been added during BioHackathon2008 +++ Version number
of the sequence (String or Integer).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">seq</td>
          <td class="readwrite">
            [<span id='seq-instance_method'>R</span><span id='seq=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
The sequence object, usually Bio::Sequence::NA/AA, but could be a simple
String.
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">species</td>
          <td class="readwrite">
            [<span id='species-instance_method'>R</span><span id='species=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Organism species (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">strandedness</td>
          <td class="readwrite">
            [<span id='strandedness-instance_method'>R</span><span id='strandedness=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Strandedness (String).
</p>

            
          </td>
        </tr>
      
        <tr>
          <th class="name">topology</td>
          <td class="readwrite">
            [<span id='topology-instance_method'>R</span><span id='topology=-instance_method'>W</span>] 
          </td>
          <td class="visibility">
            public
          </td>
          <td class="docstring">
            <p>
Topology (String).
</p>

            
          </td>
        </tr>
      
      </table>
    </div>
  
</div><div class="section constants">
  
</div><div class="section constructor">
<h2>Constructor</h2>
  <div id="initialize-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>initialize</span><span class='args'>(str)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Create a new Bio::Sequence object
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'atgc'</span>
</pre>
<p>
Note that this method does not intialize the contained sequence as any kind
of bioruby object, only as a simple string
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; String</span>
</pre>
<p>
See Bio::Sequence#na, Bio::Sequence#aa, and Bio::Sequence#auto for methods
to transform the basic String of a just created Bio::Sequence object to a
proper bioruby object
</p>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>str</em>: String or Bio::Sequence::NA/AA object

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 94</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L94">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">94
95
96</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='initialize identifier id'>initialize</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
  <span class='@seq ivar id'>@seq</span> <span class='assign token'>=</span> <span class='str identifier id'>str</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
</div><div class="section methodmissing">
  <h2>Dynamic Method Handling</h2>

  <p class="note">
    This class handles dynamic methods through the <tt>method_missing</tt> method
    
  </p>

  <div id="method_missing-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>method_missing</span><span class='args'>(sym, *args, &amp;block)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Pass any unknown method calls to the wrapped sequence object. see
http://www.rubycentral.com/book/ref_c_object.html#Object.method_missing
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 100</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L100">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='method_missing identifier id'>method_missing</span><span class='lparen token'>(</span><span class='sym identifier id'>sym</span><span class='comma token'>,</span> <span class='mult op'>*</span><span class='args identifier id'>args</span><span class='comma token'>,</span> <span class='bitand op'>&amp;</span><span class='block identifier id'>block</span><span class='rparen token'>)</span> <span class='comment val'>#:nodoc:</span>
  <span class='begin begin kw'>begin</span>
    <span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='__send__ identifier id'>__send__</span><span class='lparen token'>(</span><span class='sym identifier id'>sym</span><span class='comma token'>,</span> <span class='mult op'>*</span><span class='args identifier id'>args</span><span class='comma token'>,</span> <span class='bitand op'>&amp;</span><span class='block identifier id'>block</span><span class='rparen token'>)</span>
  <span class='rescue rescue kw'>rescue</span> <span class='NoMethodError constant id'>NoMethodError</span> <span class='assign token'>=</span><span class='gt op'>&gt;</span> <span class='evar identifier id'>evar</span>
    <span class='lineno identifier id'>lineno</span> <span class='assign token'>=</span> <span class='__LINE__ __line__ kw'>__LINE__</span> <span class='minus op'>-</span> <span class='integer val'>2</span>
    <span class='file identifier id'>file</span> <span class='assign token'>=</span> <span class='__FILE__ __file__ kw'>__FILE__</span>
    <span class='bt_here identifier id'>bt_here</span> <span class='assign token'>=</span> <span class='lbrack token'>[</span> <span class='dstring node'>&quot;#{file}:#{lineno}:in \`__send__\'&quot;</span><span class='comma token'>,</span>
                <span class='dstring node'>&quot;#{file}:#{lineno}:in \`method_missing\'&quot;</span>
              <span class='rbrack token'>]</span>
    <span class='if if kw'>if</span> <span class='bt_here identifier id'>bt_here</span> <span class='eq op'>==</span> <span class='evar identifier id'>evar</span><span class='dot token'>.</span><span class='backtrace identifier id'>backtrace</span><span class='lbrack token'>[</span><span class='integer val'>0</span><span class='comma token'>,</span> <span class='integer val'>2</span><span class='rbrack token'>]</span> <span class='then then kw'>then</span>
      <span class='bt identifier id'>bt</span> <span class='assign token'>=</span> <span class='evar identifier id'>evar</span><span class='dot token'>.</span><span class='backtrace identifier id'>backtrace</span><span class='lbrack token'>[</span><span class='float val'>2</span><span class='dot2 op'>..</span><span class='integer val'>-1</span><span class='rbrack token'>]</span>
      <span class='evar identifier id'>evar</span> <span class='assign token'>=</span> <span class='evar identifier id'>evar</span><span class='dot token'>.</span><span class='class identifier id'>class</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='dstring node'>&quot;undefined method \`#{sym.to_s}\' for #{self.inspect}&quot;</span><span class='rparen token'>)</span>
      <span class='evar identifier id'>evar</span><span class='dot token'>.</span><span class='set_backtrace identifier id'>set_backtrace</span><span class='lparen token'>(</span><span class='bt identifier id'>bt</span><span class='rparen token'>)</span>
    <span class='end end kw'>end</span>
    <span class='comment val'>#p lineno</span>
    <span class='comment val'>#p file</span>
    <span class='comment val'>#p bt_here</span>
    <span class='comment val'>#p evar.backtrace</span>
    <span class='raise identifier id'>raise</span><span class='lparen token'>(</span><span class='evar identifier id'>evar</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
</div>  <div class="section visibilitygroup">
      <div class="section methodsummary">
    <h2>Public Class Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#adapter-class_method">adapter</span><span class='args'>(source_data, adapter_module)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Normally, users should not call this method directly.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#auto-class_method">auto</span><span class='args'>(str)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Given a sequence String, guess its type, Amino Acid or Nucleic Acid, and
return a new Bio::Sequence object wrapping a sequence of the guessed type
(either Bio::Sequence::AA or Bio::Sequence::NA).
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#guess-class_method">guess</span><span class='args'>(str, *args)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Guess the class of a given sequence.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#input-class_method">input</span><span class='args'>(str, format = nil)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Create a new Bio::Sequence object from a formatted string (GenBank, EMBL,
fasta format, etc.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#read-class_method">read</span><span class='args'>(str, format = nil)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
alias of Bio::Sequence.
</p>

        
      </td>
    </tr>
  
</table>
  </div>
  <div class="section methodsummary">
    <h2>Public Instance Method Summary</h2>
<table class="summary">
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#aa-instance_method">#aa</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Transform the sequence wrapped in the current Bio::Sequence object into a
Bio::Sequence::NA object.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#accessions-instance_method">#accessions</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
accession numbers of the sequence.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#auto-instance_method">#auto</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Guess the type of sequence, Amino Acid or Nucleic Acid, and create a new
sequence object (Bio::Sequence::AA or Bio::Sequence::NA) on the basis of
this guess.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#guess-instance_method">#guess</span><span class='args'>(threshold = 0.9, length = 10000, index = 0)</span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Guess the class of the current sequence.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#na-instance_method">#na</span><span class='args'></span>
        <span class='block'></span>
        
      </th>
      <td class="docstring">
        <p>
Transform the sequence wrapped in the current Bio::Sequence object into a
Bio::Sequence::NA object.
</p>

        
      </td>
    </tr>
  
    <tr>
      <th class="signature">
        <span class='name'><a href="#to_s-instance_method">#to_s</span><span class='args'></span>
        <span class='block'></span>
        
          <span class='alias'>#to_str</span>
        
      </th>
      <td class="docstring">
        <p>
Return sequence as String[http://corelib.
</p>

        
      </td>
    </tr>
  
</table><div class="included">
  
  <div>
    <h2>Public Instance Methods Included from <a href="Sequence/Format.html" title="Bio::Sequence::Format">Bio::Sequence::Format</a></h2>
    <p><span class='name'><a href="Sequence/Format.html#list_output_formats-instance_method" title="list_output_formats">list_output_formats</a></span>, <span class='name'><a href="Sequence/Format.html#output-instance_method" title="output">output</a></span></p>
  </div>
  
</div>
<div class="clear"></div>
  </div>
<div class="section methoddetails">
  <h2>Public Class Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>adapter</h3>
</div><div id="adapter-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>adapter</span><span class='args'>(source_data, adapter_module)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Normally, users should not call this method directly. Use
Bio::*#to_biosequence (e.g. Bio::GenBank#to_biosequence).
</p>
<p>
Creates a new Bio::Sequence object from database data with an adapter
module.
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 442</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L442">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">442
443
444
445
446
447
448
449
450</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='adapter identifier id'>adapter</span><span class='lparen token'>(</span><span class='source_data identifier id'>source_data</span><span class='comma token'>,</span> <span class='adapter_module identifier id'>adapter_module</span><span class='rparen token'>)</span>
  <span class='biosequence identifier id'>biosequence</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='biosequence identifier id'>biosequence</span><span class='dot token'>.</span><span class='instance_eval identifier id'>instance_eval</span> <span class='lbrace token'>{</span>
    <span class='remove_instance_variable identifier id'>remove_instance_variable</span><span class='lparen token'>(</span><span class='symbol val'>:@seq</span><span class='rparen token'>)</span>
    <span class='@source_data ivar id'>@source_data</span> <span class='assign token'>=</span> <span class='source_data identifier id'>source_data</span>
  <span class='rbrace token'>}</span>
  <span class='biosequence identifier id'>biosequence</span><span class='dot token'>.</span><span class='extend identifier id'>extend</span><span class='lparen token'>(</span><span class='adapter_module identifier id'>adapter_module</span><span class='rparen token'>)</span>
  <span class='biosequence identifier id'>biosequence</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>auto</h3>
</div><div id="auto-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>auto</span><span class='args'>(str)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Given a sequence String, guess its type, Amino Acid or Nucleic Acid, and
return a new Bio::Sequence object wrapping a sequence of the guessed type
(either Bio::Sequence::AA or Bio::Sequence::NA)
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>str</em>: String <b>or</b> Bio::Sequence::NA/AA object

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 262</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L262">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">262
263
264
265
266</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
  <span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span>
  <span class='return return kw'>return</span> <span class='seq identifier id'>seq</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>guess</h3>
</div><div id="guess-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>guess</span><span class='args'>(str, *args)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Guess the class of a given sequence. Returns the class (Bio::Sequence::AA
or Bio::Sequence::NA) guessed. In general, used by developers only, but if
you know what you are doing, feel free.
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>        <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<p>
There are three optional parameters: `threshold`, `length`, and `index`.
</p>
<p>
The `threshold` value (defaults to 0.9) is the frequency of nucleic acid
bases [AGCTUagctu] required in the sequence for this method to produce a
Bio::Sequence::NA &quot;guess&quot;. In the default case, if less than 90%
of the bases (after excluding [Nn]) are in the set [AGCTUagctu], then the
guess is Bio::Sequence::AA.
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcqq'</span><span class='rparen token'>)</span>      <span class='comment val'>#=&gt; Bio::Sequence::AA</span>
  <span class='puts identifier id'>puts</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcqq'</span><span class='comma token'>,</span> <span class='integer val'>0</span><span class='integer val'>.8</span><span class='rparen token'>)</span> <span class='comment val'>#=&gt; Bio::Sequence::AA</span>
  <span class='puts identifier id'>puts</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcqq'</span><span class='comma token'>,</span> <span class='integer val'>0</span><span class='integer val'>.7</span><span class='rparen token'>)</span> <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<p>
The `length` value is how much of the total sequence to use in the guess
(default 10000). If your sequence is very long, you may want to use a
smaller amount to reduce the computational burden.
</p>
<pre class="code">
  <span class='comment val'># limit the guess to the first 1000 positions</span>
  <span class='puts identifier id'>puts</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='string val'>'A VERY LONG SEQUENCE'</span><span class='comma token'>,</span> <span class='integer val'>0</span><span class='integer val'>.9</span><span class='comma token'>,</span> <span class='integer val'>1000</span><span class='rparen token'>)</span>
</pre>
<p>
The `index` value is where to start the guess. Perhaps you know there are a
lot of gaps at the start&#8230;
</p>
<pre class="code">
  <span class='puts identifier id'>puts</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='string val'>'-----atgcc'</span><span class='rparen token'>)</span>             <span class='comment val'>#=&gt; Bio::Sequence::AA</span>
  <span class='puts identifier id'>puts</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='string val'>'-----atgcc'</span><span class='comma token'>,</span><span class='integer val'>0</span><span class='integer val'>.9</span><span class='comma token'>,</span><span class='integer val'>10000</span><span class='comma token'>,</span><span class='integer val'>5</span><span class='rparen token'>)</span> <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>str</em>: String <b>or</b> Bio::Sequence::NA/AA object

</li>
<li>(optional) <em>threshold</em>: Float in range 0,1 (default 0.9)

</li>
<li>(optional) <em>length</em>: Fixnum (default 10000)

</li>
<li>(optional) <em>index</em>: Fixnum (default 1)

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::NA/AA

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 360</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L360">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">360
361
362</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='comma token'>,</span> <span class='mult op'>*</span><span class='args identifier id'>args</span><span class='rparen token'>)</span>
  <span class='self self kw'>self</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='mult op'>*</span><span class='args identifier id'>args</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>input</h3>
</div><div id="input-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>input</span><span class='args'>(str, format = nil)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Create a new Bio::Sequence object from a formatted string (GenBank, EMBL,
fasta format, etc.)
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='input identifier id'>input</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(required) <em>str</em>: string

</li>
<li>(optional) <em>format</em>: format specification (class or nil)

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 415</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L415">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">415
416
417
418
419
420
421
422
423</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='input identifier id'>input</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='comma token'>,</span> <span class='format identifier id'>format</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='if if kw'>if</span> <span class='format identifier id'>format</span> <span class='then then kw'>then</span>
    <span class='klass identifier id'>klass</span> <span class='assign token'>=</span> <span class='format identifier id'>format</span>
  <span class='else else kw'>else</span>
    <span class='klass identifier id'>klass</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='FlatFile constant id'>FlatFile</span><span class='colon2 op'>::</span><span class='AutoDetect constant id'>AutoDetect</span><span class='dot token'>.</span><span class='default identifier id'>default</span><span class='dot token'>.</span><span class='autodetect identifier id'>autodetect</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
  <span class='obj identifier id'>obj</span> <span class='assign token'>=</span> <span class='klass identifier id'>klass</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='rparen token'>)</span>
  <span class='obj identifier id'>obj</span><span class='dot token'>.</span><span class='to_biosequence identifier id'>to_biosequence</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>read</h3>
</div><div id="read-class_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>read</span><span class='args'>(str, format = nil)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
alias of Bio::Sequence.input
</p>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 426</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L426">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">426
427
428</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='self self kw'>self</span><span class='dot token'>.</span><span class='read identifier id'>read</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='comma token'>,</span> <span class='format identifier id'>format</span> <span class='assign token'>=</span> <span class='nil nil kw'>nil</span><span class='rparen token'>)</span>
  <span class='input identifier id'>input</span><span class='lparen token'>(</span><span class='str identifier id'>str</span><span class='comma token'>,</span> <span class='format identifier id'>format</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div><div class="section methoddetails">
  <h2>Public Instance Method Details</h2>
  
    <div class="method">
      <div class="method_header">
  <h3>aa</h3>
</div><div id="aa-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>aa</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Transform the sequence wrapped in the current Bio::Sequence object into a
Bio::Sequence::NA object. This method will change the current object. This
method does not validate your choice, so be careful!
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; String</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='aa identifier id'>aa</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; Bio::Sequence::AA !!!</span>
</pre>
<p>
However, if you know your sequence type, this method may be constructively
used after initialization,
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'RRLE'</span><span class='rparen token'>)</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='aa identifier id'>aa</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::AA

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 401</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L401">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">401
402
403
404</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='aa identifier id'>aa</span>
  <span class='@seq ivar id'>@seq</span> <span class='assign token'>=</span> <span class='AA constant id'>AA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='seq identifier id'>seq</span><span class='rparen token'>)</span>
  <span class='@moltype ivar id'>@moltype</span> <span class='assign token'>=</span> <span class='AA constant id'>AA</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>accessions</h3>
</div><div id="accessions-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>accessions</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
accession numbers of the sequence
</p>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Array of String

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 433</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L433">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">433
434
435</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='accessions identifier id'>accessions</span>
  <span class='lbrack token'>[</span> <span class='primary_accession identifier id'>primary_accession</span><span class='comma token'>,</span> <span class='secondary_accessions identifier id'>secondary_accessions</span> <span class='rbrack token'>]</span><span class='dot token'>.</span><span class='flatten identifier id'>flatten</span><span class='dot token'>.</span><span class='compact identifier id'>compact</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>auto</h3>
</div><div id="auto-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>auto</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Guess the type of sequence, Amino Acid or Nucleic Acid, and create a new
sequence object (Bio::Sequence::AA or Bio::Sequence::NA) on the basis of
this guess. This method will change the current Bio::Sequence object.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; String</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='auto identifier id'>auto</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::NA/AA object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 243</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L243">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">243
244
245
246
247
248
249
250</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='auto identifier id'>auto</span>
  <span class='@moltype ivar id'>@moltype</span> <span class='assign token'>=</span> <span class='guess identifier id'>guess</span>
  <span class='if if kw'>if</span> <span class='@moltype ivar id'>@moltype</span> <span class='eq op'>==</span> <span class='NA constant id'>NA</span>
    <span class='@seq ivar id'>@seq</span> <span class='assign token'>=</span> <span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='seq identifier id'>seq</span><span class='rparen token'>)</span>
  <span class='else else kw'>else</span>
    <span class='@seq ivar id'>@seq</span> <span class='assign token'>=</span> <span class='AA constant id'>AA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='seq identifier id'>seq</span><span class='rparen token'>)</span>
  <span class='end end kw'>end</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>guess</h3>
</div><div id="guess-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>guess</span><span class='args'>(threshold = 0.9, length = 10000, index = 0)</span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Guess the class of the current sequence. Returns the class
(Bio::Sequence::AA or Bio::Sequence::NA) guessed. In general, used by
developers only, but if you know what you are doing, feel free.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span>                            <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<p>
There are three parameters: `threshold`, `length`, and `index`.
</p>
<p>
The `threshold` value (defaults to 0.9) is the frequency of nucleic acid
bases [AGCTUagctu] required in the sequence for this method to produce a
Bio::Sequence::NA &quot;guess&quot;. In the default case, if less than 90%
of the bases (after excluding [Nn]) are in the set [AGCTUagctu], then the
guess is Bio::Sequence::AA.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgcatgcqq'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span>                            <span class='comment val'>#=&gt; Bio::Sequence::AA</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='integer val'>0</span><span class='integer val'>.8</span><span class='rparen token'>)</span>                       <span class='comment val'>#=&gt; Bio::Sequence::AA</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='integer val'>0</span><span class='integer val'>.7</span><span class='rparen token'>)</span>                       <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<p>
The `length` value is how much of the total sequence to use in the guess
(default 10000). If your sequence is very long, you may want to use a
smaller amount to reduce the computational burden.
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='A constant id'>A</span> <span class='VERY constant id'>VERY</span> <span class='LONG constant id'>LONG</span> <span class='SEQUENCE constant id'>SEQUENCE</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='integer val'>0</span><span class='integer val'>.9</span><span class='comma token'>,</span> <span class='integer val'>1000</span><span class='rparen token'>)</span>  <span class='comment val'># limit the guess to the first 1000 positions</span>
</pre>
<p>
The `index` value is where to start the guess. Perhaps you know there are a
lot of gaps at the start&#8230;
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'-----atgcc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span>                            <span class='comment val'>#=&gt; Bio::Sequence::AA</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='integer val'>0</span><span class='integer val'>.9</span><span class='comma token'>,</span><span class='integer val'>10000</span><span class='comma token'>,</span><span class='integer val'>5</span><span class='rparen token'>)</span>               <span class='comment val'>#=&gt; Bio::Sequence::NA</span>
</pre>
<hr size="1"></hr><p>
<b>Arguments</b>:
</p>
<ul>
<li>(optional) <em>threshold</em>: Float in range 0,1 (default 0.9)

</li>
<li>(optional) <em>length</em>: Fixnum (default 10000)

</li>
<li>(optional) <em>index</em>: Fixnum (default 1)

</li>
</ul>
<table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::NA/AA

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 307</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L307">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">307
308
309
310
311
312
313
314
315
316
317
318
319
320
321</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='guess identifier id'>guess</span><span class='lparen token'>(</span><span class='threshold identifier id'>threshold</span> <span class='assign token'>=</span> <span class='integer val'>0</span><span class='integer val'>.9</span><span class='comma token'>,</span> <span class='length identifier id'>length</span> <span class='assign token'>=</span> <span class='integer val'>10000</span><span class='comma token'>,</span> <span class='index identifier id'>index</span> <span class='assign token'>=</span> <span class='integer val'>0</span><span class='rparen token'>)</span>
  <span class='str identifier id'>str</span> <span class='assign token'>=</span> <span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span><span class='lbrack token'>[</span><span class='index identifier id'>index</span><span class='comma token'>,</span><span class='length identifier id'>length</span><span class='rbrack token'>]</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span><span class='dot token'>.</span><span class='extend identifier id'>extend</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='colon2 op'>::</span><span class='Common constant id'>Common</span>
  <span class='cmp identifier id'>cmp</span> <span class='assign token'>=</span> <span class='str identifier id'>str</span><span class='dot token'>.</span><span class='composition identifier id'>composition</span>

  <span class='bases identifier id'>bases</span> <span class='assign token'>=</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'A'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'T'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'G'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'C'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'U'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span>
          <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'a'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'t'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'g'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'c'</span><span class='rbrack token'>]</span> <span class='plus op'>+</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'u'</span><span class='rbrack token'>]</span>

  <span class='total identifier id'>total</span> <span class='assign token'>=</span> <span class='str identifier id'>str</span><span class='dot token'>.</span><span class='length identifier id'>length</span> <span class='minus op'>-</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'N'</span><span class='rbrack token'>]</span> <span class='minus op'>-</span> <span class='cmp identifier id'>cmp</span><span class='lbrack token'>[</span><span class='string val'>'n'</span><span class='rbrack token'>]</span>

  <span class='if if kw'>if</span> <span class='bases identifier id'>bases</span><span class='dot token'>.</span><span class='to_f identifier id'>to_f</span> <span class='div op'>/</span> <span class='total identifier id'>total</span> <span class='gt op'>&gt;</span> <span class='threshold identifier id'>threshold</span>
    <span class='return return kw'>return</span> <span class='NA constant id'>NA</span>
  <span class='else else kw'>else</span>
    <span class='return return kw'>return</span> <span class='AA constant id'>AA</span>
  <span class='end end kw'>end</span>
<span class='end end kw'>end</span> 
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>na</h3>
</div><div id="na-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>na</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>

</div><div class="section docstring">
  <p>
Transform the sequence wrapped in the current Bio::Sequence object into a
Bio::Sequence::NA object. This method will change the current object. This
method does not validate your choice, so be careful!
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'RRLE'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; String</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='na identifier id'>na</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                        <span class='comment val'>#=&gt; Bio::Sequence::NA !!!</span>
</pre>
<p>
However, if you know your sequence type, this method may be constructively
used after initialization,
</p>
<pre class="code">
  <span class='s identifier id'>s</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='na identifier id'>na</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>Bio::Sequence::NA

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence.rb', line 380</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence.rb#L380">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">380
381
382
383</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='na identifier id'>na</span>
  <span class='@seq ivar id'>@seq</span> <span class='assign token'>=</span> <span class='NA constant id'>NA</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='seq identifier id'>seq</span><span class='rparen token'>)</span>
  <span class='@moltype ivar id'>@moltype</span> <span class='assign token'>=</span> <span class='NA constant id'>NA</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
    <div class="method">
      <div class="method_header">
  <h3>to_s</h3>
</div><div id="to_s-instance_method" class="section method">
  <div class="details_title">
  <div class='section methodsignature'>
  <tt class='def'>
    <span class='visibility'>public</span>
    <span class='return_types'></span>
    <span class='name'>to_s</span><span class='args'></span>
    <span class='block'></span>
  </tt>
</div>
<p class="aliases">
  <span class="aka">Also known as:</span>
  
    <tt id="to_str-instance_method">to_str</tt>
  
</p>
</div><div class="section docstring">
  <p>
Return sequence as
String[http://corelib.rubyonrails.org/classes/String.html]. The original
sequence is unchanged.
</p>
<pre class="code">
  <span class='seq identifier id'>seq</span> <span class='assign token'>=</span> <span class='Bio constant id'>Bio</span><span class='colon2 op'>::</span><span class='Sequence constant id'>Sequence</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='string val'>'atgc'</span><span class='rparen token'>)</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span>                             <span class='comment val'>#=&gt; 'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='to_s identifier id'>to_s</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                       <span class='comment val'>#=&gt; String</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span>                                  <span class='comment val'>#=&gt; 'atgc'</span>
  <span class='puts identifier id'>puts</span> <span class='s identifier id'>s</span><span class='dot token'>.</span><span class='class identifier id'>class</span>                            <span class='comment val'>#=&gt; Bio::Sequence</span>
</pre>
<hr size="1"></hr><table>
<tr><td valign="top"><b>Returns</b>:</td><td>String object

</td></tr>
</table>

</div></div>
<div class="section source">
  <span>[<a class="source_link" href="#">View source</a>]</span>
  <div class="files" style="display:none">
    <div class="file">
      <div class="meta">
        <div class="info">
          <span>File 'lib/bio/sequence/compat.rb', line 32</span>
        </div>
        <div class="actions">      
          <a id="raw-url" href="http://github.com/bioruby/bioruby/blob/97b9284109c9a4431b92eab208509e1df6069b4b/lib/bio/sequence/compat.rb#L32">raw</a>      
        </div>
      </div>
            
      <div class="data syntax">    
        <table cellspacing="0" cellpadding="0">
          <tbody>
            <tr>
              <td>            
                <pre class="line_numbers">32
33
34</pre>
              </td>
              <td width="100%">                    
                <div class="highlight"><pre class="code"><span class='def def kw'>def</span> <span class='to_s identifier id'>to_s</span>
  <span class='String constant id'>String</span><span class='dot token'>.</span><span class='new identifier id'>new</span><span class='lparen token'>(</span><span class='self self kw'>self</span><span class='dot token'>.</span><span class='seq identifier id'>seq</span><span class='rparen token'>)</span>
<span class='end end kw'>end</span>
</pre></div>          
              </td>
            </tr>
          </tbody>
        </table>    
      </div>
    </div>  
  </div>
</div>
<div>
</div>
    </div>
  
</div>
  </div>

</div>
</div>
          </div>
        </div>
      </div>                    
    </div>

    <div id="footer">
      <div class="site">
        <div class="info">
          <div class="links">
            <a href="http://github.com/blog/148-github-shirts-now-available">Shirts</a> |
            <a href="http://github.com/blog">Blog</a> |
            <a href="http://support.github.com/">Support</a> |
            <a href="http://github.com/training">Training</a> |
            <a href="http://github.com/contact">Contact</a> |
            <a href="http://groups.google.com/group/github/">Google Group</a> |
            <a href="http://develop.github.com/">API</a> |
            <a href="http://twitter.com/github">Status</a>
          </div>
          <div class="company">
            <span id="_rrt" title="0.64509s from xc88-s00039">GitHub</span>
            is <a href="http://logicalawesome.com/">Logical Awesome</a> ©2009 | <a href="http://github.com/site/terms">Terms of Service</a> | <a href="http://github.com/site/privacy">Privacy Policy</a>
          </div>
        </div>
        <div class="sponsor">
          <a href="http://engineyard.com/"><img src="http://github.com/images/modules/footer/engine_yard_logo.png" alt="Engine Yard"></a>
          <div>
            Hosting provided by our<br> partners at Engine Yard
          </div>
        </div>
      </div>
    </div>
    
    <div id="coming_soon" style="display: none;">
      This feature is coming soon.  Sit tight!
    </div>

    <div id="facebox" style="display: none;">       
      <div class="popup">         
        <table>           
          <tbody>             
            <tr>               
              <td class="tl"></td>
              <td class="b"></td>
              <td class="tr"></td>             
            </tr>             
          <tr>               
            <td class="b"></td>               
            <td class="body">                 
              <div class="content"></div>                 
              <div class="footer">                   
                <a href="#" class="close"><img src="http://github.com/facebox/closelabel.gif" title="close" class="close_image"></a>                 
              </div>               
            </td>               
            <td class="b"></td>             
          </tr>             
          <tr>               
            <td class="bl"></td>
            <td class="b"></td>
            <td class="br"></td>             
          </tr>           
        </tbody>         
      </table>       
    </div>     


    <script type="text/javascript">
    var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
    document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
    </script>
    <script type="text/javascript">
    try {
    var pageTracker = _gat._getTracker("UA-8689483-1");
    pageTracker._trackPageview();
    } catch(err) {}</script>

  </body>
</html>